error[E0506]: cannot assign to `v[..]` because it is borrowed
  --> $DIR/drop-no-may-dangle.rs:31:9
   |
LL |     let p: WrapMayNotDangle<&usize> = WrapMayNotDangle { value: &v[0] };
   |                                                                 ----- borrow of `v[..]` occurs here
...
LL |         v[0] += 1; //~ ERROR cannot assign to `v[..]` because it is borrowed
   |         ^^^^^^^^^ assignment to borrowed `v[..]` occurs here
...
LL | }
   | - borrow later used here, when `p` is dropped

error[E0506]: cannot assign to `v[..]` because it is borrowed
  --> $DIR/drop-no-may-dangle.rs:34:5
   |
LL |     let p: WrapMayNotDangle<&usize> = WrapMayNotDangle { value: &v[0] };
   |                                                                 ----- borrow of `v[..]` occurs here
...
LL |     v[0] += 1; //~ ERROR cannot assign to `v[..]` because it is borrowed
   |     ^^^^^^^^^ assignment to borrowed `v[..]` occurs here
LL | }
   | - borrow later used here, when `p` is dropped

error: aborting due to 2 previous errors

If you want more information on this error, try using "rustc --explain E0506"
