error[E0506]: cannot assign to `*s` because it is borrowed
  --> $DIR/guarantor-issue-46974.rs:19:5
   |
17 |     let t = &mut *s; // this borrow should last for the entire function
   |             ------- borrow of `*s` occurs here
18 |     let x = &t.0;
19 |     *s = (2,); //~ ERROR cannot assign to `*s`
   |     ^^^^^^^^^ assignment to borrowed `*s` occurs here

error: free region `` does not outlive free region `'static`
  --> $DIR/guarantor-issue-46974.rs:25:5
   |
25 |     &s.0 //~ ERROR free region `` does not outlive free region `'static`
   |     ^^^^

error: aborting due to 2 previous errors

