error[E0506]: cannot assign to `*s` because it is borrowed
  --> $DIR/guarantor-issue-46974.rs:19:5
   |
17 |     let t = &mut *s; // this borrow should last for the entire function
   |             ------- borrow of `*s` occurs here
18 |     let x = &t.0;
19 |     *s = (2,); //~ ERROR cannot assign to `*s`
   |     ^^^^^^^^^ assignment to borrowed `*s` occurs here

error[E0621]: explicit lifetime required in the type of `s`
  --> $DIR/guarantor-issue-46974.rs:25:5
   |
23 | fn bar(s: &Box<(i32,)>) -> &'static i32 {
   |        - consider changing the type of `s` to `&'static std::boxed::Box<(i32,)>`
24 |     // FIXME(#46983): error message should be better
25 |     &s.0 //~ ERROR explicit lifetime required in the type of `s` [E0621]
   |     ^^^^ lifetime `'static` required

error: aborting due to 2 previous errors

