error: lifetime may not live long enough
  --> $DIR/issue-58299.rs:17:9
   |
LL | fn foo<'a>(x: i32) {
   |        -- lifetime `'a` defined here
...
LL |         A::<'a>::X..=A::<'static>::X => (), //~ ERROR lifetime may not live long enough
   |         ^^^^^^^^^^ requires that `'a` must outlive `'static`

error: lifetime may not live long enough
  --> $DIR/issue-58299.rs:25:27
   |
LL | fn bar<'a>(x: i32) {
   |        -- lifetime `'a` defined here
...
LL |         A::<'static>::X..=A::<'a>::X => (), //~ ERROR lifetime may not live long enough
   |                           ^^^^^^^^^^ requires that `'a` must outlive `'static`

error: aborting due to 2 previous errors

