error[E0510]: cannot assign `q` in match guard
  --> $DIR/match-guards-partially-borrow.rs:59:13
   |
LL |     match q {
   |           - value is immutable in match guard
...
LL |             q = true; //~ ERROR
   |             ^^^^^^^^ cannot assign

error[E0510]: cannot assign `r` in match guard
  --> $DIR/match-guards-partially-borrow.rs:71:13
   |
LL |     match r {
   |           - value is immutable in match guard
...
LL |             r = true; //~ ERROR
   |             ^^^^^^^^ cannot assign

error[E0510]: cannot assign `t` in match guard
  --> $DIR/match-guards-partially-borrow.rs:95:13
   |
LL |     match t {
   |           - value is immutable in match guard
...
LL |             t = true; //~ ERROR
   |             ^^^^^^^^ cannot assign

error[E0510]: cannot mutably borrow `x.0` in match guard
  --> $DIR/match-guards-partially-borrow.rs:109:22
   |
LL |     match x {
   |           - value is immutable in match guard
...
LL |                 Some(ref mut r) => *r = None, //~ ERROR
   |                      ^^^^^^^^^ cannot mutably borrow

error[E0506]: cannot assign to `t` because it is borrowed
  --> $DIR/match-guards-partially-borrow.rs:121:13
   |
LL |         s if {
   |         - borrow of `t` occurs here
LL |             t = !t; //~ ERROR
   |             ^^^^^^ assignment to borrowed `t` occurs here
LL |             false
LL |         } => (), // What value should `s` have in the arm?
   |         - borrow later used here

error[E0510]: cannot assign `y` in match guard
  --> $DIR/match-guards-partially-borrow.rs:132:13
   |
LL |     match *y {
   |           -- value is immutable in match guard
...
LL |             y = &true; //~ ERROR
   |             ^^^^^^^^^ cannot assign

error[E0510]: cannot assign `z` in match guard
  --> $DIR/match-guards-partially-borrow.rs:143:13
   |
LL |     match z {
   |           - value is immutable in match guard
...
LL |             z = &true; //~ ERROR
   |             ^^^^^^^^^ cannot assign

error[E0510]: cannot assign `a` in match guard
  --> $DIR/match-guards-partially-borrow.rs:155:13
   |
LL |     match a {
   |           - value is immutable in match guard
...
LL |             a = &true; //~ ERROR
   |             ^^^^^^^^^ cannot assign

error[E0510]: cannot assign `b` in match guard
  --> $DIR/match-guards-partially-borrow.rs:166:13
   |
LL |     match b {
   |           - value is immutable in match guard
...
LL |             b = &true; //~ ERROR
   |             ^^^^^^^^^ cannot assign

error: aborting due to 9 previous errors

Some errors occurred: E0506, E0510.
For more information about an error, try `rustc --explain E0506`.
