warning: not reporting region error due to -Znll
  --> $DIR/named-region-basic.rs:19:5
   |
19 |     &*x //~ ERROR free region `'a` does not outlive `'b`
   |     ^^^

error[E0597]: `*x` does not live long enough
  --> $DIR/named-region-basic.rs:19:6
   |
19 |     &*x //~ ERROR free region `'a` does not outlive `'b`
   |      ^^ does not live long enough
   |
   = note: borrowed value must be valid for the static lifetime...
note: ...but borrowed value is only valid for the lifetime 'a as defined on the function body at 18:1
  --> $DIR/named-region-basic.rs:18:1
   |
18 | / fn foo<'a, 'b>(x: &'a u32, y: &'b u32) -> &'b u32 {
19 | |     &*x //~ ERROR free region `'a` does not outlive `'b`
20 | |     //~^ ERROR `*x` does not live long enough
21 | |     //~| WARN not reporting region error due to -Znll
22 | | }
   | |_^

error: free region `'a` does not outlive `'b`
  --> $DIR/named-region-basic.rs:19:5
   |
19 |     &*x //~ ERROR free region `'a` does not outlive `'b`
   |     ^^^

error: aborting due to 2 previous errors

