error: unsatisfied lifetime constraints
  --> $DIR/type-alias-free-regions.rs:19:9
   |
LL | impl<'a> FromBox<'a> for c<'a> {
   |      -- lifetime `'a` defined here
LL |     fn from_box(b: Box<b>) -> Self {
   |                 - has type `std::boxed::Box<std::boxed::Box<&'1 isize>>`
LL |         c { f: b } //~ ERROR
   |         ^^^^^^^^^^ returning this value requires that `'1` must outlive `'a`

error: unsatisfied lifetime constraints
  --> $DIR/type-alias-free-regions.rs:29:9
   |
LL | impl<'a> FromTuple<'a> for c<'a> {
   |      -- lifetime `'a` defined here
LL |     fn from_tuple(b: (b,)) -> Self {
   |                   - has type `(std::boxed::Box<&'1 isize>,)`
LL |         c { f: Box::new(b.0) } //~ ERROR
   |         ^^^^^^^^^^^^^^^^^^^^^^ returning this value requires that `'1` must outlive `'a`

error: aborting due to 2 previous errors

