error: rustc_outlives
  --> $DIR/enum.rs:19:1
   |
LL | / enum Foo<'a, T> { //~ ERROR rustc_outlives
LL | |     One(Bar<'a, T>)
LL | | }
   | |_^
   |
   = note: T : 'a

error: rustc_outlives
  --> $DIR/enum.rs:25:1
   |
LL | / struct Bar<'b, U> { //~ ERROR rustc_outlives
LL | |     field2: &'b U
LL | | }
   | |_^
   |
   = note: U : 'b

error: rustc_outlives
  --> $DIR/enum.rs:31:1
   |
LL | / enum Ying<'c, K> { //~ ERROR rustc_outlives
LL | |     One(&'c Yang<K>)
LL | | }
   | |_^
   |
   = note: K : 'c

error: aborting due to 3 previous errors

