error[E0309]: the parameter type `U` may not live long enough
  --> $DIR/explicit-where.rs:15:5
   |
LL | struct Foo<'b, U> {
   |                - help: consider adding an explicit lifetime bound `U: 'b`...
LL |     bar: Bar<'b, U> //~ Error the parameter type `U` may not live long enough [E0309]
   |     ^^^^^^^^^^^^^^^
   |
note: ...so that the type `U` will meet its required lifetime bounds
  --> $DIR/explicit-where.rs:15:5
   |
LL |     bar: Bar<'b, U> //~ Error the parameter type `U` may not live long enough [E0309]
   |     ^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0309`.
