error[E0309]: the parameter type `T` may not live long enough
  --> $DIR/explicit-impl.rs:27:5
   |
LL | struct Foo<'a, T> {
   |                - help: consider adding an explicit lifetime bound `T: 'a`...
LL |     foo: <Vec<T> as MakeRef<'a>>::Type //~ Error the parameter type `T` may not live long enough [E0309]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: ...so that the type `T` will meet its required lifetime bounds
  --> $DIR/explicit-impl.rs:27:5
   |
LL |     foo: <Vec<T> as MakeRef<'a>>::Type //~ Error the parameter type `T` may not live long enough [E0309]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0309`.
