error[E0309]: the parameter type `U` may not live long enough
  --> $DIR/nested-structs.rs:22:5
   |
LL | struct Bar<'b, U> {
   |                - help: consider adding an explicit lifetime bound `U: 'b`...
LL |     field2: &'b U //~ ERROR the parameter type `U` may not live long enough [E0309]
   |     ^^^^^^^^^^^^^
   |
note: ...so that the reference type `&'b U` does not outlive the data it points at
  --> $DIR/nested-structs.rs:22:5
   |
LL |     field2: &'b U //~ ERROR the parameter type `U` may not live long enough [E0309]
   |     ^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0309`.
