error[E0309]: the associated type `<T as std::iter::Iterator>::Item` may not live long enough
  --> $DIR/projections.rs:17:5
   |
LL |     bar: &'a T::Item //~ Error the associated type `<T as std::iter::Iterator>::Item` may not live long enough [E0309]
   |     ^^^^^^^^^^^^^^^^
   |
   = help: consider adding an explicit lifetime bound `<T as std::iter::Iterator>::Item: 'a`...
note: ...so that the reference type `&'a <T as std::iter::Iterator>::Item` does not outlive the data it points at
  --> $DIR/projections.rs:17:5
   |
LL |     bar: &'a T::Item //~ Error the associated type `<T as std::iter::Iterator>::Item` may not live long enough [E0309]
   |     ^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0309`.
