error[E0309]: the parameter type `U` may not live long enough
  --> $DIR/union.rs:25:5
   |
LL | union Bar<'b, U> {
   |               - help: consider adding an explicit lifetime bound `U: 'b`...
LL |     field2: &'b U //~ ERROR 25:5: 25:18: the parameter type `U` may not live long enough [E0309]
   |     ^^^^^^^^^^^^^
   |
note: ...so that the reference type `&'b U` does not outlive the data it points at
  --> $DIR/union.rs:25:5
   |
LL |     field2: &'b U //~ ERROR 25:5: 25:18: the parameter type `U` may not live long enough [E0309]
   |     ^^^^^^^^^^^^^

error[E0309]: the parameter type `K` may not live long enough
  --> $DIR/union.rs:31:5
   |
LL | union Ying<'c, K> {
   |                - help: consider adding an explicit lifetime bound `K: 'c`...
LL |     field1: &'c Yang<K> //~ ERROR 31:5: 31:24: the parameter type `K` may not live long enough [E0309]
   |     ^^^^^^^^^^^^^^^^^^^
   |
note: ...so that the reference type `&'c Yang<K>` does not outlive the data it points at
  --> $DIR/union.rs:31:5
   |
LL |     field1: &'c Yang<K> //~ ERROR 31:5: 31:24: the parameter type `K` may not live long enough [E0309]
   |     ^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0309`.
