warning: not reporting region error due to nll
  --> $DIR/ex1-return-one-existing-name-self-is-anon.rs:18:30
   |
LL |         if true { x } else { self } //~ ERROR lifetime mismatch
   |                              ^^^^

error: unsatisfied lifetime constraints
  --> $DIR/ex1-return-one-existing-name-self-is-anon.rs:18:30
   |
LL |     fn foo<'a>(&self, x: &'a Foo) -> &'a Foo {
   |            --  - let's call the lifetime of this reference `'1`
   |            |
   |            lifetime `'a` defined here
LL | 
LL |         if true { x } else { self } //~ ERROR lifetime mismatch
   |                              ^^^^ function was supposed to return data with lifetime `'a` but it is returning data with lifetime `'1`

error: aborting due to previous error

