warning: not reporting region error due to nll
  --> $DIR/ex3-both-anon-regions-both-are-structs-latebound-regions.rs:15:12
   |
LL |     x.push(y); //~ ERROR lifetime mismatch
   |            ^

error: unsatisfied lifetime constraints
  --> $DIR/ex3-both-anon-regions-both-are-structs-latebound-regions.rs:15:5
   |
LL | fn foo<'a, 'b>(mut x: Vec<Ref<'a>>, y: Ref<'b>) {
   |        --  -- lifetime `'b` defined here
   |        |
   |        lifetime `'a` defined here
LL |     x.push(y); //~ ERROR lifetime mismatch
   |     ^^^^^^^^^ argument requires that `'b` must outlive `'a`

error: aborting due to previous error

