warning: not reporting region error due to nll
  --> $DIR/ex3-both-anon-regions-both-are-structs-3.rs:16:11
   |
LL |     x.a = x.b; //~ ERROR lifetime mismatch
   |           ^^^

error: unsatisfied lifetime constraints
  --> $DIR/ex3-both-anon-regions-both-are-structs-3.rs:16:5
   |
LL | fn foo(mut x: Ref) {
   |        -----
   |        |
   |        has type `Ref<'_, '1>`
   |        has type `Ref<'2, '_>`
LL |     x.a = x.b; //~ ERROR lifetime mismatch
   |     ^^^^^^^^^ requires that `'1` must outlive `'2`

error: aborting due to previous error

