warning: not reporting region error due to nll
  --> $DIR/ex3-both-anon-regions-3.rs:12:13
   |
LL |     z.push((x,y)); //~ ERROR lifetime mismatch
   |             ^

warning: not reporting region error due to nll
  --> $DIR/ex3-both-anon-regions-3.rs:12:15
   |
LL |     z.push((x,y)); //~ ERROR lifetime mismatch
   |               ^

error: unsatisfied lifetime constraints
  --> $DIR/ex3-both-anon-regions-3.rs:12:5
   |
LL | fn foo(z: &mut Vec<(&u8,&u8)>, (x, y): (&u8, &u8)) {
   |                     -                   - let's call the lifetime of this reference `'1`
   |                     |
   |                     let's call the lifetime of this reference `'2`
LL |     z.push((x,y)); //~ ERROR lifetime mismatch
   |     ^^^^^^^^^^^^^ argument requires that `'1` must outlive `'2`

error: unsatisfied lifetime constraints
  --> $DIR/ex3-both-anon-regions-3.rs:12:5
   |
LL | fn foo(z: &mut Vec<(&u8,&u8)>, (x, y): (&u8, &u8)) {
   |                         -                    - let's call the lifetime of this reference `'1`
   |                         |
   |                         let's call the lifetime of this reference `'2`
LL |     z.push((x,y)); //~ ERROR lifetime mismatch
   |     ^^^^^^^^^^^^^ argument requires that `'1` must outlive `'2`

error: aborting due to 2 previous errors

