warning: not reporting region error due to nll
  --> $DIR/ex3-both-anon-regions-latebound-regions.rs:12:12
   |
LL |     x.push(y); //~ ERROR lifetime mismatch
   |            ^

error: unsatisfied lifetime constraints
  --> $DIR/ex3-both-anon-regions-latebound-regions.rs:12:5
   |
LL | fn foo<'a,'b>(x: &mut Vec<&'a u8>, y: &'b u8) {
   |        -- -- lifetime `'b` defined here
   |        |
   |        lifetime `'a` defined here
LL |     x.push(y); //~ ERROR lifetime mismatch
   |     ^^^^^^^^^ argument requires that `'b` must outlive `'a`

error: aborting due to previous error

