error: unsatisfied lifetime constraints
  --> $DIR/ex3-both-anon-regions-one-is-struct-2.rs:14:5
   |
LL | fn foo(mut x: Ref, y: &u32) {
   |        -----          - let's call the lifetime of this reference `'2`
   |        |
   |        has type `Ref<'_, '1>`
LL |     y = x.b; //~ ERROR lifetime mismatch
   |     ^^^^^^^ requires that `'1` must outlive `'2`

error[E0384]: cannot assign to immutable argument `y`
  --> $DIR/ex3-both-anon-regions-one-is-struct-2.rs:14:5
   |
LL | fn foo(mut x: Ref, y: &u32) {
   |                    - consider changing this to `mut y`
LL |     y = x.b; //~ ERROR lifetime mismatch
   |     ^^^^^^^ cannot assign to immutable argument

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0384`.
