error[E0596]: cannot borrow `y` as mutable, as it is not declared as mutable
  --> $DIR/ex3-both-anon-regions-using-fn-items.rs:11:3
   |
LL | fn foo(x:fn(&u8, &u8), y: Vec<&u8>, z: &u8) {
   |                        - help: consider changing this to be mutable: `mut y`
LL |   y.push(z); //~ ERROR lifetime mismatch
   |   ^ cannot borrow as mutable

error: unsatisfied lifetime constraints
  --> $DIR/ex3-both-anon-regions-using-fn-items.rs:11:3
   |
LL | fn foo(x:fn(&u8, &u8), y: Vec<&u8>, z: &u8) {
   |                               -        - let's call the lifetime of this reference `'1`
   |                               |
   |                               let's call the lifetime of this reference `'2`
LL |   y.push(z); //~ ERROR lifetime mismatch
   |   ^^^^^^^^^ argument requires that `'1` must outlive `'2`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0596`.
