error[E0716]: temporary value dropped while borrowed
  --> $DIR/promotion.rs:13:27
   |
LL |     let x: &'static () = &foo1(); //~ ERROR does not live long enough
   |                           ^^^^^^ creates a temporary which is freed while still in use
...
LL | }
   | - temporary value is freed at the end of this statement
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0716]: temporary value dropped while borrowed
  --> $DIR/promotion.rs:14:28
   |
LL |     let y: &'static i32 = &foo2(42); //~ ERROR does not live long enough
   |                            ^^^^^^^^ creates a temporary which is freed while still in use
...
LL | }
   | - temporary value is freed at the end of this statement
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0716]: temporary value dropped while borrowed
  --> $DIR/promotion.rs:15:28
   |
LL |     let z: &'static i32 = &foo3(); //~ ERROR does not live long enough
   |                            ^^^^^^ creates a temporary which is freed while still in use
...
LL | }
   | - temporary value is freed at the end of this statement
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0716]: temporary value dropped while borrowed
  --> $DIR/promotion.rs:16:34
   |
LL |     let a: &'static Cell<i32> = &foo4();  //~ ERROR does not live long enough
   |                                  ^^^^^^ creates a temporary which is freed while still in use
...
LL | }
   | - temporary value is freed at the end of this statement
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0716]: temporary value dropped while borrowed
  --> $DIR/promotion.rs:17:42
   |
LL |     let a: &'static Option<Cell<i32>> = &foo5(); //~ ERROR does not live long enough
   |                                          ^^^^^^ creates a temporary which is freed while still in use
LL |     let a: &'static Option<Cell<i32>> = &foo6(); //~ ERROR does not live long enough
LL | }
   | - temporary value is freed at the end of this statement
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0716]: temporary value dropped while borrowed
  --> $DIR/promotion.rs:18:42
   |
LL |     let a: &'static Option<Cell<i32>> = &foo6(); //~ ERROR does not live long enough
   |                                          ^^^^^^ creates a temporary which is freed while still in use
LL | }
   | - temporary value is freed at the end of this statement
   |
   = note: borrowed value must be valid for the static lifetime...

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0716`.
