warning: unused return value of `need_to_use_this_value` which must be used: it's important
  --> $DIR/fn_must_use.rs:61:5
   |
61 |     need_to_use_this_value(); //~ WARN unused return value
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/fn_must_use.rs:14:9
   |
14 | #![warn(unused_must_use)]
   |         ^^^^^^^^^^^^^^^

warning: unused return value of `MyStruct::need_to_use_this_method_value` which must be used
  --> $DIR/fn_must_use.rs:66:5
   |
66 |     m.need_to_use_this_method_value(); //~ WARN unused return value
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused return value of `EvenNature::is_even` which must be used: no side effects
  --> $DIR/fn_must_use.rs:67:5
   |
67 |     m.is_even(); // trait method!
   |     ^^^^^^^^^^^^

warning: unused return value of `std::cmp::PartialEq::eq` which must be used
  --> $DIR/fn_must_use.rs:73:5
   |
73 |     2.eq(&3); //~ WARN unused return value
   |     ^^^^^^^^^

warning: unused return value of `std::cmp::PartialEq::eq` which must be used
  --> $DIR/fn_must_use.rs:74:5
   |
74 |     m.eq(&n); //~ WARN unused return value
   |     ^^^^^^^^^

warning: unused comparison which must be used
  --> $DIR/fn_must_use.rs:77:5
   |
77 |     2 == 3; //~ WARN unused comparison
   |     ^^^^^^

warning: unused comparison which must be used
  --> $DIR/fn_must_use.rs:78:5
   |
78 |     m == n; //~ WARN unused comparison
   |     ^^^^^^

