error[E0506]: cannot assign to `vec[..]` because it is borrowed
  --> $DIR/borrowck-vec-pattern-nesting.rs:20:13
   |
LL |         [box ref _a, _, _] => {
   |              ------ borrow of `vec[..]` occurs here
LL |         //~^ borrow of `vec[..]` occurs here
LL |             vec[0] = box 4; //~ ERROR cannot assign
   |             ^^^^^^ assignment to borrowed `vec[..]` occurs here
LL |             //~^ assignment to borrowed `vec[..]` occurs here
LL |             _a.use_ref();
   |             -- borrow later used here

error[E0506]: cannot assign to `vec[..]` because it is borrowed
  --> $DIR/borrowck-vec-pattern-nesting.rs:33:13
   |
LL |         &mut [ref _b..] => {
   |               ------ borrow of `vec[..]` occurs here
LL |         //~^ borrow of `vec[..]` occurs here
LL |             vec[0] = box 4; //~ ERROR cannot assign
   |             ^^^^^^ assignment to borrowed `vec[..]` occurs here
LL |             //~^ assignment to borrowed `vec[..]` occurs here
LL |             _b.use_ref();
   |             -- borrow later used here

error[E0508]: cannot move out of type `[std::boxed::Box<isize>]`, a non-copy slice
  --> $DIR/borrowck-vec-pattern-nesting.rs:43:11
   |
LL |     match vec {
   |           ^^^ cannot move out of here
LL |         &mut [_a, //~ ERROR cannot move out
   |               -- data moved here
   |
note: move occurs because `_a` has type `std::boxed::Box<isize>`, which does not implement the `Copy` trait
  --> $DIR/borrowck-vec-pattern-nesting.rs:44:15
   |
LL |         &mut [_a, //~ ERROR cannot move out
   |               ^^
help: consider removing the `&mut`
   |
LL |         [_a, //~ ERROR cannot move out
LL |             //~| cannot move out
LL |             //~| to prevent move
LL |             ..
LL |         ] => {
   |

error[E0508]: cannot move out of type `[std::boxed::Box<isize>]`, a non-copy slice
  --> $DIR/borrowck-vec-pattern-nesting.rs:57:13
   |
LL |     let a = vec[0]; //~ ERROR cannot move out
   |             ^^^^^^
   |             |
   |             cannot move out of here
   |             help: consider borrowing here: `&vec[0]`

error[E0508]: cannot move out of type `[std::boxed::Box<isize>]`, a non-copy slice
  --> $DIR/borrowck-vec-pattern-nesting.rs:64:11
   |
LL |     match vec {
   |           ^^^ cannot move out of here
...
LL |          _b] => {}
   |          -- data moved here
   |
note: move occurs because `_b` has type `std::boxed::Box<isize>`, which does not implement the `Copy` trait
  --> $DIR/borrowck-vec-pattern-nesting.rs:67:10
   |
LL |          _b] => {}
   |          ^^
help: consider removing the `&mut`
   |
LL |         [ //~ ERROR cannot move out
LL |         //~^ cannot move out
LL |          _b] => {}
   |

error[E0508]: cannot move out of type `[std::boxed::Box<isize>]`, a non-copy slice
  --> $DIR/borrowck-vec-pattern-nesting.rs:70:13
   |
LL |     let a = vec[0]; //~ ERROR cannot move out
   |             ^^^^^^
   |             |
   |             cannot move out of here
   |             help: consider borrowing here: `&vec[0]`

error[E0508]: cannot move out of type `[std::boxed::Box<isize>]`, a non-copy slice
  --> $DIR/borrowck-vec-pattern-nesting.rs:77:11
   |
LL |     match vec {
   |           ^^^ cannot move out of here
LL |         &mut [_a, _b, _c] => {}  //~ ERROR cannot move out
   |         -----------------
   |         |     |   |   |
   |         |     |   |   ...and here
   |         |     |   ...and here
   |         |     data moved here
   |         help: consider removing the `&mut`: `[_a, _b, _c]`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/borrowck-vec-pattern-nesting.rs:78:15
   |
LL |         &mut [_a, _b, _c] => {}  //~ ERROR cannot move out
   |               ^^  ^^  ^^

error[E0508]: cannot move out of type `[std::boxed::Box<isize>]`, a non-copy slice
  --> $DIR/borrowck-vec-pattern-nesting.rs:82:13
   |
LL |     let a = vec[0]; //~ ERROR cannot move out
   |             ^^^^^^
   |             |
   |             cannot move out of here
   |             help: consider borrowing here: `&vec[0]`

error: aborting due to 8 previous errors

Some errors occurred: E0506, E0508.
For more information about an error, try `rustc --explain E0506`.
