error[E0596]: cannot borrow data in a `&` reference as mutable
  --> $DIR/borrowck-borrow-overloaded-auto-deref.rs:57:19
   |
LL |     let __isize = &mut x.y; //~ ERROR cannot borrow
   |                   ^^^^^^^^ cannot borrow as mutable

error[E0596]: cannot borrow data in a `&` reference as mutable
  --> $DIR/borrowck-borrow-overloaded-auto-deref.rs:61:19
   |
LL |     let __isize = &mut x.y; //~ ERROR cannot borrow
   |                   ^^^^^^^^ cannot borrow as mutable

error[E0596]: cannot borrow data in a `&` reference as mutable
  --> $DIR/borrowck-borrow-overloaded-auto-deref.rs:69:5
   |
LL |     &mut x.y //~ ERROR cannot borrow
   |     ^^^^^^^^ cannot borrow as mutable

error[E0596]: cannot borrow data in a `&` reference as mutable
  --> $DIR/borrowck-borrow-overloaded-auto-deref.rs:73:5
   |
LL |     &mut x.y //~ ERROR cannot borrow
   |     ^^^^^^^^ cannot borrow as mutable

error[E0594]: cannot assign to data in a `&` reference
  --> $DIR/borrowck-borrow-overloaded-auto-deref.rs:77:5
   |
LL |     x.y = 3; //~ ERROR cannot assign
   |     ^^^^^^^ cannot assign

error[E0594]: cannot assign to data in a `&` reference
  --> $DIR/borrowck-borrow-overloaded-auto-deref.rs:81:5
   |
LL |     x.y = 3; //~ ERROR cannot assign
   |     ^^^^^^^ cannot assign

error[E0594]: cannot assign to data in a `&` reference
  --> $DIR/borrowck-borrow-overloaded-auto-deref.rs:85:5
   |
LL |     x.y = 3; //~ ERROR cannot assign
   |     ^^^^^^^ cannot assign

error[E0596]: cannot borrow data in a `&` reference as mutable
  --> $DIR/borrowck-borrow-overloaded-auto-deref.rs:93:5
   |
LL |     x.set(0, 0); //~ ERROR cannot borrow
   |     ^ cannot borrow as mutable

error[E0596]: cannot borrow data in a `&` reference as mutable
  --> $DIR/borrowck-borrow-overloaded-auto-deref.rs:97:5
   |
LL |     x.set(0, 0); //~ ERROR cannot borrow
   |     ^ cannot borrow as mutable

error[E0596]: cannot borrow data in a `&` reference as mutable
  --> $DIR/borrowck-borrow-overloaded-auto-deref.rs:105:5
   |
LL |     x.y_mut() //~ ERROR cannot borrow
   |     ^ cannot borrow as mutable

error[E0596]: cannot borrow data in a `&` reference as mutable
  --> $DIR/borrowck-borrow-overloaded-auto-deref.rs:109:5
   |
LL |     x.y_mut() //~ ERROR cannot borrow
   |     ^ cannot borrow as mutable

error[E0596]: cannot borrow data in a `&` reference as mutable
  --> $DIR/borrowck-borrow-overloaded-auto-deref.rs:113:6
   |
LL |     *x.y_mut() = 3; //~ ERROR cannot borrow
   |      ^ cannot borrow as mutable

error[E0596]: cannot borrow data in a `&` reference as mutable
  --> $DIR/borrowck-borrow-overloaded-auto-deref.rs:117:6
   |
LL |     *x.y_mut() = 3; //~ ERROR cannot borrow
   |      ^ cannot borrow as mutable

error[E0596]: cannot borrow data in a `&` reference as mutable
  --> $DIR/borrowck-borrow-overloaded-auto-deref.rs:121:6
   |
LL |     *x.y_mut() = 3; //~ ERROR cannot borrow
   |      ^ cannot borrow as mutable

error: aborting due to 14 previous errors

Some errors occurred: E0594, E0596.
For more information about an error, try `rustc --explain E0594`.
