error[E0382]: use of moved value: `x` (Ast)
  --> $DIR/borrowck-reinit.rs:18:16
   |
17 |     drop(x);
   |          - value moved here
18 |     let _ = (1,x); //~ ERROR use of moved value: `x` (Ast)
   |                ^ value used here after move
   |
   = note: move occurs because `x` has type `std::boxed::Box<i32>`, which does not implement the `Copy` trait

error[E0382]: use of moved value: `x` (Mir)
  --> $DIR/borrowck-reinit.rs:18:16
   |
17 |     drop(x);
   |          - value moved here
18 |     let _ = (1,x); //~ ERROR use of moved value: `x` (Ast)
   |                ^ value used here after move

error: aborting due to 2 previous errors

