error[E0506]: cannot assign to `vec[..]` because it is borrowed
  --> $DIR/borrowck-vec-pattern-nesting.rs:20:13
   |
LL |         [box ref _a, _, _] => {
   |              ------ borrow of `vec[..]` occurs here
LL |         //~^ borrow of `vec[..]` occurs here
LL |             vec[0] = box 4; //~ ERROR cannot assign
   |             ^^^^^^ assignment to borrowed `vec[..]` occurs here
LL |             //~^ assignment to borrowed `vec[..]` occurs here
LL |             _a.use_ref();
   |             -- borrow later used here

error[E0506]: cannot assign to `vec[..]` because it is borrowed
  --> $DIR/borrowck-vec-pattern-nesting.rs:33:13
   |
LL |         &mut [ref _b..] => {
   |               ------ borrow of `vec[..]` occurs here
LL |         //~^ borrow of `vec[..]` occurs here
LL |             vec[0] = box 4; //~ ERROR cannot assign
   |             ^^^^^^ assignment to borrowed `vec[..]` occurs here
LL |             //~^ assignment to borrowed `vec[..]` occurs here
LL |             _b.use_ref();
   |             -- borrow later used here

error[E0508]: cannot move out of type `[std::boxed::Box<isize>]`, a non-copy slice
  --> $DIR/borrowck-vec-pattern-nesting.rs:43:11
   |
LL |     match vec {
   |           ^^^ cannot move out of here
LL |         &mut [_a, //~ ERROR cannot move out
   |               --
   |               |
   |               data moved here
   |               help: to prevent move, use ref or ref mut: `ref _a`

error[E0508]: cannot move out of type `[std::boxed::Box<isize>]`, a non-copy slice
  --> $DIR/borrowck-vec-pattern-nesting.rs:57:13
   |
LL |     let a = vec[0]; //~ ERROR cannot move out
   |             ^^^^^^
   |             |
   |             cannot move out of here
   |             help: consider using a reference instead: `&vec[0]`

error[E0508]: cannot move out of type `[std::boxed::Box<isize>]`, a non-copy slice
  --> $DIR/borrowck-vec-pattern-nesting.rs:64:11
   |
LL |     match vec {
   |           ^^^ cannot move out of here
...
LL |          _b] => {}
   |          --
   |          |
   |          data moved here
   |          help: to prevent move, use ref or ref mut: `ref _b`

error[E0508]: cannot move out of type `[std::boxed::Box<isize>]`, a non-copy slice
  --> $DIR/borrowck-vec-pattern-nesting.rs:70:13
   |
LL |     let a = vec[0]; //~ ERROR cannot move out
   |             ^^^^^^
   |             |
   |             cannot move out of here
   |             help: consider using a reference instead: `&vec[0]`

error[E0508]: cannot move out of type `[std::boxed::Box<isize>]`, a non-copy slice
  --> $DIR/borrowck-vec-pattern-nesting.rs:77:11
   |
LL |     match vec {
   |           ^^^ cannot move out of here
LL |         &mut [_a, _b, _c] => {}  //~ ERROR cannot move out
   |               --  --  -- ... and here
   |               |   |
   |               |   ... and here
   |               data moved here
help: to prevent move, use ref or ref mut
   |
LL |         &mut [ref _a, ref _b, ref _c] => {}  //~ ERROR cannot move out
   |               ^^^^^^  ^^^^^^  ^^^^^^

error[E0508]: cannot move out of type `[std::boxed::Box<isize>]`, a non-copy slice
  --> $DIR/borrowck-vec-pattern-nesting.rs:82:13
   |
LL |     let a = vec[0]; //~ ERROR cannot move out
   |             ^^^^^^
   |             |
   |             cannot move out of here
   |             help: consider using a reference instead: `&vec[0]`

error: aborting due to 8 previous errors

Some errors occurred: E0506, E0508.
For more information about an error, try `rustc --explain E0506`.
