error[E0596]: cannot borrow `s` as mutable, as it is not declared as mutable
  --> $DIR/borrowck-overloaded-call.rs:77:5
   |
LL |     let s = SFnMut {
   |         - help: consider changing this to be mutable: `mut s`
...
LL |     s(3);   //~ ERROR cannot borrow immutable local variable `s` as mutable
   |     ^ cannot borrow as mutable

error[E0382]: use of moved value: `s`
  --> $DIR/borrowck-overloaded-call.rs:85:5
   |
LL |     s(" world".to_string());
   |     - value moved here
LL |     s(" world".to_string());    //~ ERROR use of moved value: `s`
   |     ^ value used here after move
   |
   = note: move occurs because `s` has type `SFnOnce`, which does not implement the `Copy` trait

error: aborting due to 2 previous errors

Some errors occurred: E0382, E0596.
For more information about an error, try `rustc --explain E0382`.
