error[E0502]: cannot borrow `*x` as mutable because it is also borrowed as immutable
  --> $DIR/borrowck-object-lifetime.rs:28:14
   |
LL |     let _y = x.borrowed();
   |              - immutable borrow occurs here
LL |     let _z = x.mut_borrowed(); //~ ERROR cannot borrow
   |              ^ mutable borrow occurs here
LL | }
   | - immutable borrow ends here

error[E0502]: cannot borrow `x` as mutable because `*x` is also borrowed as immutable
  --> $DIR/borrowck-object-lifetime.rs:33:19
   |
LL |     let _y = x.borrowed();
   |              - immutable borrow occurs here
LL |     let _z = &mut x; //~ ERROR cannot borrow
   |                   ^ mutable borrow occurs here
LL | }
   | - immutable borrow ends here

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0502`.
