error[E0502]: cannot borrow `a` as mutable because it is also borrowed as immutable
  --> $DIR/borrowck-bad-nested-calls-free.rs:35:17
   |
LL | /     add(
LL | |         &*a,
   | |         --- immutable borrow occurs here
LL | |         rewrite(&mut a)); //~ ERROR cannot borrow
   | |_________________^^^^^^_- borrow later used here
   |                   |
   |                   mutable borrow occurs here

error[E0502]: cannot borrow `a` as mutable because it is also borrowed as immutable
  --> $DIR/borrowck-bad-nested-calls-free.rs:42:17
   |
LL | /     add(
LL | |         &*a,
   | |         --- immutable borrow occurs here
LL | |         rewrite(&mut a)); //~ ERROR cannot borrow
   | |_________________^^^^^^_- borrow later used here
   |                   |
   |                   mutable borrow occurs here

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0502`.
