error[E0716]: temporary value dropped while borrowed
  --> $DIR/promote-ref-mut-in-let-issue-46557.rs:15:21
   |
LL |     let ref mut x = 1234543; //~ ERROR
   |                     ^^^^^^^ creates a temporary which is freed while still in use
LL |     x
LL | }
   | - temporary value is freed at the end of this statement
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0716]: temporary value dropped while borrowed
  --> $DIR/promote-ref-mut-in-let-issue-46557.rs:20:25
   |
LL |     let (ref mut x, ) = (1234543, ); //~ ERROR
   |                         ^^^^^^^^^^^ creates a temporary which is freed while still in use
LL |     x
LL | }
   | - temporary value is freed at the end of this statement
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0716]: temporary value dropped while borrowed
  --> $DIR/promote-ref-mut-in-let-issue-46557.rs:25:11
   |
LL |     match 1234543 {
   |           ^^^^^^^ creates a temporary which is freed while still in use
...
LL | }
   | - temporary value is freed at the end of this statement
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0716]: temporary value dropped while borrowed
  --> $DIR/promote-ref-mut-in-let-issue-46557.rs:31:11
   |
LL |     match (123443,) {
   |           ^^^^^^^^^ creates a temporary which is freed while still in use
...
LL | }
   | - temporary value is freed at the end of this statement
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0716]: temporary value dropped while borrowed
  --> $DIR/promote-ref-mut-in-let-issue-46557.rs:37:10
   |
LL |     &mut 1234543 //~ ERROR
   |          ^^^^^^^ creates a temporary which is freed while still in use
LL | }
   | - temporary value is freed at the end of this statement
   |
   = note: borrowed value must be valid for the static lifetime...

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0716`.
