error[E0594]: cannot assign to immutable borrowed content `*n`
  --> $DIR/explicit-mut.rs:19:13
   |
18 |         Some(n) => {
   |              - consider changing this to `n`
19 |             *n += 1; //~ ERROR cannot assign to immutable
   |             ^^^^^^^ cannot borrow as mutable

error[E0594]: cannot assign to immutable borrowed content `*n`
  --> $DIR/explicit-mut.rs:27:13
   |
26 |         Some(n) => {
   |              - consider changing this to `n`
27 |             *n += 1; //~ ERROR cannot assign to immutable
   |             ^^^^^^^ cannot borrow as mutable

error[E0594]: cannot assign to immutable borrowed content `*n`
  --> $DIR/explicit-mut.rs:35:13
   |
34 |         Some(n) => {
   |              - consider changing this to `n`
35 |             *n += 1; //~ ERROR cannot assign to immutable
   |             ^^^^^^^ cannot borrow as mutable

error: aborting due to 3 previous errors

If you want more information on this error, try using "rustc --explain E0594"
