error[E0308]: mismatched types
  --> $DIR/lit.rs:19:13
   |
LL |             "abc" => true, //~ ERROR mismatched types
   |             ^^^^^ expected &str, found str
   |
   = note: expected type `&&str`
              found type `&'static str`

error[E0308]: mismatched types
  --> $DIR/lit.rs:28:9
   |
LL |         b"abc" => true, //~ ERROR mismatched types
   |         ^^^^^^ expected &[u8], found array of 3 elements
   |
   = note: expected type `&&[u8]`
              found type `&'static [u8; 3]`

error: aborting due to 2 previous errors

If you want more information on this error, try using "rustc --explain E0308"
