error[E0603]: module `bar` is private
  --> $DIR/use-mod-3.rs:11:10
   |
LL | use foo::bar::{ //~ ERROR module `bar` is private
   |          ^^^

error[E0603]: module `bar` is private
  --> $DIR/use-mod-3.rs:14:10
   |
LL | use foo::bar::{ //~ ERROR module `bar` is private
   |          ^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0603`.
