error: unnecessary `unsafe` block
  --> $DIR/issue-48131.rs:18:9
   |
LL |         unsafe { /* unnecessary */ } //~ ERROR unnecessary `unsafe`
   |         ^^^^^^ unnecessary `unsafe` block
   |
note: lint level defined here
  --> $DIR/issue-48131.rs:13:9
   |
LL | #![deny(unused_unsafe)] //~ NOTE
   |         ^^^^^^^^^^^^^

error: unnecessary `unsafe` block
  --> $DIR/issue-48131.rs:29:13
   |
LL |             unsafe { /* unnecessary */ } //~ ERROR unnecessary `unsafe`
   |             ^^^^^^ unnecessary `unsafe` block

error: aborting due to 2 previous errors

