error[E0412]: cannot find type `Foo` in this scope
  --> $DIR/issue-35075.rs:12:12
   |
LL |     inner: Foo<T> //~ ERROR cannot find type `Foo` in this scope
   |            ^^^---
   |            |
   |            not found in this scope
   |            help: you can try using the variant's enum: `Baz`

error[E0412]: cannot find type `Foo` in this scope
  --> $DIR/issue-35075.rs:16:9
   |
LL |     Foo(Foo<T>) //~ ERROR cannot find type `Foo` in this scope
   |         ^^^---
   |         |
   |         not found in this scope
   |         help: you can try using the variant's enum: `Baz`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0412`.
