warning: not reporting region error due to nll
  --> $DIR/issue-52213.rs:12:11
   |
LL |     match (&t,) { //~ ERROR cannot infer an appropriate lifetime
   |           ^^^^^

error: unsatisfied lifetime constraints
  --> $DIR/issue-52213.rs:13:20
   |
LL | fn transmute_lifetime<'a, 'b, T>(t: &'a (T,)) -> &'b T {
   |                       --  -- lifetime `'b` defined here
   |                       |
   |                       lifetime `'a` defined here
LL |     match (&t,) { //~ ERROR cannot infer an appropriate lifetime
LL |         ((u,),) => u,
   |                    ^ returning this value requires that `'a` must outlive `'b`

error: aborting due to previous error

