error[E0507]: cannot move out of static item
  --> $DIR/issue-17718-static-move.rs:16:14
   |
LL |     let _a = FOO; //~ ERROR: cannot move out of static item
   |              ^^^
   |              |
   |              cannot move out of static item
   |              help: consider borrowing here: `&FOO`

error[E0507]: cannot move out of immutable static item `FOO`
  --> $DIR/issue-17718-static-move.rs:16:14
   |
LL |     let _a = FOO; //~ ERROR: cannot move out of static item
   |              ^^^
   |              |
   |              cannot move out of immutable static item `FOO`
   |              cannot move

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0507`.
