error[E0308]: mismatched types
  --> $DIR/no_revealing_outside_defining_module.rs:26:19
   |
LL |     let _: &str = bomp(); //~ ERROR mismatched types
   |                   ^^^^^^ expected &str, found anonymized type
   |
   = note: expected type `&str`
              found type `Boo`

error[E0308]: mismatched types
  --> $DIR/no_revealing_outside_defining_module.rs:30:5
   |
LL | fn bomp() -> boo::Boo {
   |              -------- expected `Boo` because of return type
LL |     "" //~ ERROR mismatched types
   |     ^^ expected anonymized type, found reference
   |
   = note: expected type `Boo`
              found type `&'static str`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
