error[E0322]: explicit impls for the `Sized` trait are not permitted
  --> $DIR/coherence-impls-sized.rs:14:1
   |
LL | impl Sized for TestE {} //~ ERROR E0322
   | ^^^^^^^^^^^^^^^^^^^^ impl of 'Sized' not allowed

error[E0322]: explicit impls for the `Sized` trait are not permitted
  --> $DIR/coherence-impls-sized.rs:17:1
   |
LL | impl Sized for MyType {} //~ ERROR E0322
   | ^^^^^^^^^^^^^^^^^^^^^ impl of 'Sized' not allowed

error[E0322]: explicit impls for the `Sized` trait are not permitted
  --> $DIR/coherence-impls-sized.rs:20:1
   |
LL | impl Sized for (MyType, MyType) {} //~ ERROR E0322
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ impl of 'Sized' not allowed

error[E0322]: explicit impls for the `Sized` trait are not permitted
  --> $DIR/coherence-impls-sized.rs:24:1
   |
LL | impl Sized for &'static NotSync {} //~ ERROR E0322
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ impl of 'Sized' not allowed

error[E0322]: explicit impls for the `Sized` trait are not permitted
  --> $DIR/coherence-impls-sized.rs:27:1
   |
LL | impl Sized for [MyType] {} //~ ERROR E0322
   | ^^^^^^^^^^^^^^^^^^^^^^^ impl of 'Sized' not allowed

error[E0322]: explicit impls for the `Sized` trait are not permitted
  --> $DIR/coherence-impls-sized.rs:31:1
   |
LL | impl Sized for &'static [NotSync] {} //~ ERROR E0322
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ impl of 'Sized' not allowed

error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
  --> $DIR/coherence-impls-sized.rs:20:1
   |
LL | impl Sized for (MyType, MyType) {} //~ ERROR E0322
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ impl doesn't use types inside crate
   |
   = note: the impl does not reference any types defined in this crate
   = note: define and implement a trait or new type instead

error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
  --> $DIR/coherence-impls-sized.rs:27:1
   |
LL | impl Sized for [MyType] {} //~ ERROR E0322
   | ^^^^^^^^^^^^^^^^^^^^^^^ impl doesn't use types inside crate
   |
   = note: the impl does not reference any types defined in this crate
   = note: define and implement a trait or new type instead

error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
  --> $DIR/coherence-impls-sized.rs:31:1
   |
LL | impl Sized for &'static [NotSync] {} //~ ERROR E0322
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ impl doesn't use types inside crate
   |
   = note: the impl does not reference any types defined in this crate
   = note: define and implement a trait or new type instead

error: aborting due to 9 previous errors

Some errors occurred: E0117, E0322.
For more information about an error, try `rustc --explain E0117`.
