error: no rules expected the token `r#async`
  --> $DIR/edition-keywords-2015-2018-parsing.rs:22:31
   |
LL |     r#async = consumes_async!(r#async); //~ ERROR no rules expected the token `r#async`
   |                               ^^^^^^^

error: no rules expected the token `async`
  --> $DIR/edition-keywords-2015-2018-parsing.rs:23:35
   |
LL |     r#async = consumes_async_raw!(async); //~ ERROR no rules expected the token `async`
   |                                   ^^^^^

error: aborting due to 2 previous errors

