error: expected identifier, found reserved keyword `async`
  --> $DIR/edition-keywords-2018-2018-parsing.rs:18:13
   |
LL |     let mut async = 1; //~ ERROR expected identifier, found reserved keyword `async`
   |             ^^^^^ expected identifier, found reserved keyword

error: expected identifier, found reserved keyword `async`
  --> $DIR/edition-keywords-2018-2018-parsing.rs:28:13
   |
LL |     module::async(); //~ ERROR expected identifier, found reserved keyword `async`
   |             ^^^^^ expected identifier, found reserved keyword

error: no rules expected the token `r#async`
  --> $DIR/edition-keywords-2018-2018-parsing.rs:22:31
   |
LL |     r#async = consumes_async!(r#async); //~ ERROR no rules expected the token `r#async`
   |                               ^^^^^^^

error: no rules expected the token `async`
  --> $DIR/edition-keywords-2018-2018-parsing.rs:23:35
   |
LL |     r#async = consumes_async_raw!(async); //~ ERROR no rules expected the token `async`
   |                                   ^^^^^

error: expected one of `move`, `|`, or `||`, found `<eof>`
  --> <::edition_kw_macro_2018::passes_ident macros>:1:22
   |
LL | ( $ i : ident ) => ( $ i )
   |                      ^^^ expected one of `move`, `|`, or `||` here

error: aborting due to 5 previous errors

