error[E0308]: mismatched types
  --> $DIR/consider-removing-last-semi.rs:11:18
   |
11 |   fn f() -> String {  //~ ERROR mismatched types
   |  __________________^
12 | |     0u8;
13 | |     "bla".to_string();  //~ HELP consider removing this semicolon
   | |                      - help: consider removing this semicolon
14 | | }
   | |_^ expected struct `std::string::String`, found ()
   |
   = note: expected type `std::string::String`
              found type `()`

error[E0308]: mismatched types
  --> $DIR/consider-removing-last-semi.rs:16:18
   |
16 |   fn g() -> String {  //~ ERROR mismatched types
   |  __________________^
17 | |     "this won't work".to_string();
18 | |     "removeme".to_string(); //~ HELP consider removing this semicolon
   | |                           - help: consider removing this semicolon
19 | | }
   | |_^ expected struct `std::string::String`, found ()
   |
   = note: expected type `std::string::String`
              found type `()`

error: aborting due to 2 previous errors

