error[E0308]: mismatched types
  --> $DIR/issue-13428.rs:13:20
   |
13 |   fn foo() -> String {  //~ ERROR mismatched types
   |  ____________________^
14 | |     format!("Hello {}",
15 | |             "world")
16 | |     // Put the trailing semicolon on its own line to test that the
17 | |     // note message gets the offending semicolon exactly
18 | |     ;   //~ HELP consider removing this semicolon
   | |     - help: consider removing this semicolon
19 | | }
   | |_^ expected struct `std::string::String`, found ()
   |
   = note: expected type `std::string::String`
              found type `()`

error[E0308]: mismatched types
  --> $DIR/issue-13428.rs:21:20
   |
21 |   fn bar() -> String {  //~ ERROR mismatched types
   |  ____________________^
22 | |     "foobar".to_string()
23 | |     ;   //~ HELP consider removing this semicolon
   | |     - help: consider removing this semicolon
24 | | }
   | |_^ expected struct `std::string::String`, found ()
   |
   = note: expected type `std::string::String`
              found type `()`

error: aborting due to 2 previous errors

