error[E0530]: match bindings cannot shadow tuple structs
  --> $DIR/pattern-binding-disambiguation.rs:34:9
   |
LL | struct TupleStruct();
   | --------------------- a tuple struct `TupleStruct` is defined here
...
LL |         TupleStruct => {} //~ ERROR match bindings cannot shadow tuple structs
   |         ^^^^^^^^^^^ cannot be named the same as a tuple struct

error[E0530]: match bindings cannot shadow tuple variants
  --> $DIR/pattern-binding-disambiguation.rs:43:9
   |
LL | use E::*;
   |     ---- a tuple variant `TupleVariant` is imported here
...
LL |         TupleVariant => {} //~ ERROR match bindings cannot shadow tuple variants
   |         ^^^^^^^^^^^^ cannot be named the same as a tuple variant

error[E0530]: match bindings cannot shadow struct variants
  --> $DIR/pattern-binding-disambiguation.rs:46:9
   |
LL | use E::*;
   |     ---- a struct variant `BracedVariant` is imported here
...
LL |         BracedVariant => {} //~ ERROR match bindings cannot shadow struct variants
   |         ^^^^^^^^^^^^^ cannot be named the same as a struct variant

error[E0530]: match bindings cannot shadow statics
  --> $DIR/pattern-binding-disambiguation.rs:52:9
   |
LL | static STATIC: () = ();
   | ----------------------- a static `STATIC` is defined here
...
LL |         STATIC => {} //~ ERROR match bindings cannot shadow statics
   |         ^^^^^^ cannot be named the same as a static

error[E0530]: let bindings cannot shadow tuple structs
  --> $DIR/pattern-binding-disambiguation.rs:59:9
   |
LL | struct TupleStruct();
   | --------------------- a tuple struct `TupleStruct` is defined here
...
LL |     let TupleStruct = doesnt_matter; //~ ERROR let bindings cannot shadow tuple structs
   |         ^^^^^^^^^^^ cannot be named the same as a tuple struct

error[E0530]: let bindings cannot shadow tuple variants
  --> $DIR/pattern-binding-disambiguation.rs:62:9
   |
LL | use E::*;
   |     ---- a tuple variant `TupleVariant` is imported here
...
LL |     let TupleVariant = doesnt_matter; //~ ERROR let bindings cannot shadow tuple variants
   |         ^^^^^^^^^^^^ cannot be named the same as a tuple variant

error[E0530]: let bindings cannot shadow struct variants
  --> $DIR/pattern-binding-disambiguation.rs:63:9
   |
LL | use E::*;
   |     ---- a struct variant `BracedVariant` is imported here
...
LL |     let BracedVariant = doesnt_matter; //~ ERROR let bindings cannot shadow struct variants
   |         ^^^^^^^^^^^^^ cannot be named the same as a struct variant

error[E0530]: let bindings cannot shadow statics
  --> $DIR/pattern-binding-disambiguation.rs:65:9
   |
LL | static STATIC: () = ();
   | ----------------------- a static `STATIC` is defined here
...
LL |     let STATIC = doesnt_matter; //~ ERROR let bindings cannot shadow statics
   |         ^^^^^^ cannot be named the same as a static

error: aborting due to 8 previous errors

For more information about this error, try `rustc --explain E0530`.
