error[E0596]: cannot borrow immutable borrowed content `*x` as mutable
  --> $DIR/borrowck-object-mutability.rs:19:5
   |
LL | fn borrowed_receiver(x: &Foo) {
   |                         ---- use `&mut Foo` here to make mutable
LL |     x.borrowed();
LL |     x.borrowed_mut(); //~ ERROR cannot borrow
   |     ^ cannot borrow as mutable

error[E0596]: cannot borrow immutable `Box` content `*x` as mutable
  --> $DIR/borrowck-object-mutability.rs:29:5
   |
LL | fn owned_receiver(x: Box<Foo>) {
   |                   - consider changing this to `mut x`
LL |     x.borrowed();
LL |     x.borrowed_mut(); //~ ERROR cannot borrow
   |     ^ cannot borrow as mutable

error: aborting due to 2 previous errors

If you want more information on this error, try using "rustc --explain E0596"
