error: cannot borrow immutable argument `x` as mutable
  --> $DIR/borrowck-borrow-overloaded-deref-mut.rs:39:25
   |
38 | fn deref_mut1(x: Own<isize>) {
   |               - use `mut x` here to make mutable
39 |     let __isize = &mut *x; //~ ERROR cannot borrow
   |                         ^ cannot borrow mutably

error: cannot borrow immutable borrowed content `*x` as mutable
  --> $DIR/borrowck-borrow-overloaded-deref-mut.rs:51:11
   |
50 | fn deref_extend_mut1<'a>(x: &'a Own<isize>) -> &'a mut isize {
   |                             -------------- use `&'a mut Own<isize>` here to make mutable
51 |     &mut **x //~ ERROR cannot borrow
   |           ^^

error: cannot borrow immutable argument `x` as mutable
  --> $DIR/borrowck-borrow-overloaded-deref-mut.rs:59:6
   |
58 | fn assign1<'a>(x: Own<isize>) {
   |                - use `mut x` here to make mutable
59 |     *x = 3; //~ ERROR cannot borrow
   |      ^ cannot borrow mutably

error: cannot borrow immutable borrowed content `*x` as mutable
  --> $DIR/borrowck-borrow-overloaded-deref-mut.rs:63:6
   |
62 | fn assign2<'a>(x: &'a Own<isize>) {
   |                   -------------- use `&'a mut Own<isize>` here to make mutable
63 |     **x = 3; //~ ERROR cannot borrow
   |      ^^

error: aborting due to 4 previous errors

