error: cannot borrow immutable borrowed content `*x` as mutable
  --> $DIR/borrowck-object-mutability.rs:19:5
   |
17 | fn borrowed_receiver(x: &Foo) {
   |                         ---- use `&mut Foo` here to make mutable
18 |     x.borrowed();
19 |     x.borrowed_mut(); //~ ERROR cannot borrow
   |     ^

error: cannot borrow immutable `Box` content `*x` as mutable
  --> $DIR/borrowck-object-mutability.rs:29:5
   |
29 |     x.borrowed_mut(); //~ ERROR cannot borrow
   |     ^

error: aborting due to 2 previous errors

