error[E0057]: this function takes 1 parameter but 0 parameters were supplied
  --> $DIR/E0057.rs:13:13
   |
13 |     let a = f(); //~ ERROR E0057
   |             ^^^
   |
   = note: the following parameter type was expected: (_,)

error[E0057]: this function takes 1 parameter but 2 parameters were supplied
  --> $DIR/E0057.rs:15:15
   |
15 |     let c = f(2, 3); //~ ERROR E0057
   |               ^^^^
   |
   = note: the following parameter type was expected: (_,)

error: aborting due to 2 previous errors

