error[E0658]: SIMD types are experimental and possibly buggy (see issue #27731)
  --> $DIR/gated-features-attr-spans.rs:20:1
   |
LL | #[repr(simd)] //~ ERROR are experimental
   | ^^^^^^^^^^^^^
   |
   = help: add #![feature(repr_simd)] to the crate attributes to enable

warning: `#[must_use]` on methods is experimental (see issue #43302)
  --> $DIR/gated-features-attr-spans.rs:27:5
   |
LL |     #[must_use] fn summon_weapon(&self) -> Weapon { self.weapon }
   |     ^^^^^^^^^^^
   |
   = help: add #![feature(fn_must_use)] to the crate attributes to enable

warning: `#[must_use]` on functions is experimental (see issue #43302)
  --> $DIR/gated-features-attr-spans.rs:31:1
   |
LL | #[must_use] //~ WARN is experimental
   | ^^^^^^^^^^^
   |
   = help: add #![feature(fn_must_use)] to the crate attributes to enable

error: aborting due to previous error

If you want more information on this error, try using "rustc --explain E0658"
