error: unexpected generic arguments in path
  --> $DIR/import-ty-params.rs:24:15
   |
24 |     import! { a::b::c::S<u8> } //~ ERROR unexpected generic arguments in path
   |               ^^^^^^^^^^^^^^

error: unexpected generic arguments in path
  --> $DIR/import-ty-params.rs:27:15
   |
27 |     import! { a::b::c::S<> } //~ ERROR unexpected generic arguments in path
   |               ^^^^^^^^^^^^

error: aborting due to 2 previous errors

