error[E0597]: `z` does not live long enough
  --> $DIR/regions-escape-loop-via-vec.rs:25:5
   |
22 |         _y.push(&mut z); //~ NOTE borrow occurs here
   |                      - borrow occurs here
...
25 |     } //~ NOTE `z` dropped here while still borrowed
   |     ^ `z` dropped here while still borrowed
26 |     //~^ ERROR `z` does not live long enough
27 | }
   | - borrowed value needs to live until here

error[E0503]: cannot use `x` because it was mutably borrowed
  --> $DIR/regions-escape-loop-via-vec.rs:18:11
   |
14 |     let mut _y = vec![&mut x];
   |                            - borrow of `x` occurs here
...
18 |     while x < 10 { //~ ERROR cannot use `x` because it was mutably borrowed
   |           ^ use of borrowed `x`

error[E0503]: cannot use `x` because it was mutably borrowed
  --> $DIR/regions-escape-loop-via-vec.rs:20:13
   |
14 |     let mut _y = vec![&mut x];
   |                            - borrow of `x` occurs here
...
20 |         let mut z = x; //~ ERROR cannot use `x` because it was mutably borrowed
   |             ^^^^^ use of borrowed `x`

error[E0506]: cannot assign to `x` because it is borrowed
  --> $DIR/regions-escape-loop-via-vec.rs:23:9
   |
14 |     let mut _y = vec![&mut x];
   |                            - borrow of `x` occurs here
...
23 |         x += 1; //~ ERROR cannot assign
   |         ^^^^^^ assignment to borrowed `x` occurs here

error: aborting due to 4 previous errors

