error[E0597]: `z` does not live long enough
  --> $DIR/send-is-not-static-std-sync.rs:26:34
   |
LL |         *lock.lock().unwrap() = &z;
   |                                  ^ borrowed value does not live long enough
LL |     }
   |     - `z` dropped here while still borrowed
LL |     //~^^ ERROR `z` does not live long enough
LL | }
   | - borrowed value needs to live until here

error[E0505]: cannot move out of `y` because it is borrowed
  --> $DIR/send-is-not-static-std-sync.rs:23:10
   |
LL |     *lock.lock().unwrap() = &*y;
   |                              -- borrow of `*y` occurs here
LL |     drop(y); //~ ERROR cannot move out
   |          ^ move out of `y` occurs here

error[E0597]: `z` does not live long enough
  --> $DIR/send-is-not-static-std-sync.rs:39:35
   |
LL |         *lock.write().unwrap() = &z;
   |                                   ^ borrowed value does not live long enough
LL |     }
   |     - `z` dropped here while still borrowed
LL |     //~^^ ERROR `z` does not live long enough
LL | }
   | - borrowed value needs to live until here

error[E0505]: cannot move out of `y` because it is borrowed
  --> $DIR/send-is-not-static-std-sync.rs:36:10
   |
LL |     *lock.write().unwrap() = &*y;
   |                               -- borrow of `*y` occurs here
LL |     drop(y); //~ ERROR cannot move out
   |          ^ move out of `y` occurs here

error[E0597]: `z` does not live long enough
  --> $DIR/send-is-not-static-std-sync.rs:54:18
   |
LL |         tx.send(&z).unwrap();
   |                  ^ borrowed value does not live long enough
LL |     }
   |     - `z` dropped here while still borrowed
LL |     //~^^ ERROR `z` does not live long enough
LL | }
   | - borrowed value needs to live until here

error[E0505]: cannot move out of `y` because it is borrowed
  --> $DIR/send-is-not-static-std-sync.rs:51:10
   |
LL |     tx.send(&*y);
   |              -- borrow of `*y` occurs here
LL |     drop(y); //~ ERROR cannot move out
   |          ^ move out of `y` occurs here

error: aborting due to 6 previous errors

You've got a few errors: E0505, E0597
If you want more information on an error, try using "rustc --explain E0505"
