error[E0597]: `z` does not live long enough
  --> $DIR/send-is-not-static-std-sync.rs:26:34
   |
26 |         *lock.lock().unwrap() = &z;
   |                                  ^ borrowed value does not live long enough
27 |     }
   |     - `z` dropped here while still borrowed
28 |     //~^^ ERROR `z` does not live long enough
29 | }
   | - borrowed value needs to live until here

error[E0505]: cannot move out of `y` because it is borrowed
  --> $DIR/send-is-not-static-std-sync.rs:23:10
   |
22 |     *lock.lock().unwrap() = &*y;
   |                              -- borrow of `*y` occurs here
23 |     drop(y); //~ ERROR cannot move out
   |          ^ move out of `y` occurs here

error[E0597]: `z` does not live long enough
  --> $DIR/send-is-not-static-std-sync.rs:39:35
   |
39 |         *lock.write().unwrap() = &z;
   |                                   ^ borrowed value does not live long enough
40 |     }
   |     - `z` dropped here while still borrowed
41 |     //~^^ ERROR `z` does not live long enough
42 | }
   | - borrowed value needs to live until here

error[E0505]: cannot move out of `y` because it is borrowed
  --> $DIR/send-is-not-static-std-sync.rs:36:10
   |
35 |     *lock.write().unwrap() = &*y;
   |                               -- borrow of `*y` occurs here
36 |     drop(y); //~ ERROR cannot move out
   |          ^ move out of `y` occurs here

error[E0597]: `z` does not live long enough
  --> $DIR/send-is-not-static-std-sync.rs:54:18
   |
54 |         tx.send(&z).unwrap();
   |                  ^ borrowed value does not live long enough
55 |     }
   |     - `z` dropped here while still borrowed
56 |     //~^^ ERROR `z` does not live long enough
57 | }
   | - borrowed value needs to live until here

error[E0505]: cannot move out of `y` because it is borrowed
  --> $DIR/send-is-not-static-std-sync.rs:51:10
   |
50 |     tx.send(&*y);
   |              -- borrow of `*y` occurs here
51 |     drop(y); //~ ERROR cannot move out
   |          ^ move out of `y` occurs here

error: aborting due to 6 previous errors

