error[E0425]: cannot find value `bar` in this scope
  --> $DIR/typo-suggestion.rs:15:26
   |
15 |     println!("Hello {}", bar); //~ ERROR cannot find value
   |                          ^^^ not found in this scope

error[E0425]: cannot find value `fob` in this scope
  --> $DIR/typo-suggestion.rs:18:26
   |
18 |     println!("Hello {}", fob); //~ ERROR cannot find value
   |                          ^^^ did you mean `foo`?

error: aborting due to 2 previous errors

