error[E0692]: transparent struct cannot have other repr hints
  --> $DIR/repr-transparent-other-reprs.rs:15:8
   |
LL | #[repr(transparent, C)] //~ ERROR cannot have other repr
   |        ^^^^^^^^^^^  ^

error[E0692]: transparent struct cannot have other repr hints
  --> $DIR/repr-transparent-other-reprs.rs:20:8
   |
LL | #[repr(transparent, packed)] //~ ERROR cannot have other repr
   |        ^^^^^^^^^^^  ^^^^^^

error[E0692]: transparent struct cannot have other repr hints
  --> $DIR/repr-transparent-other-reprs.rs:23:8
   |
LL | #[repr(transparent, align(2))] //~ ERROR cannot have other repr
   |        ^^^^^^^^^^^  ^^^^^^^^

error[E0692]: transparent struct cannot have other repr hints
  --> $DIR/repr-transparent-other-reprs.rs:26:8
   |
LL | #[repr(transparent)] //~ ERROR cannot have other repr
   |        ^^^^^^^^^^^
LL | #[repr(C)]
   |        ^

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0692`.
