error[E0658]: SIMD types are experimental and possibly buggy (see issue #27731)
  --> $DIR/feature-gate-repr-simd.rs:11:1
   |
LL | #[repr(simd)] //~ error: SIMD types are experimental
   | ^^^^^^^^^^^^^
   |
   = help: add #![feature(repr_simd)] to the crate attributes to enable

error[E0658]: SIMD types are experimental and possibly buggy (see issue #27731)
  --> $DIR/feature-gate-repr-simd.rs:15:1
   |
LL | #[repr(simd)] //~ error: SIMD types are experimental
   | ^^^^^^^^^^^^^
   |
   = help: add #![feature(repr_simd)] to the crate attributes to enable

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0658`.
