error[E0409]: variable `y` is bound in inconsistent ways within the same match arm
  --> $DIR/E0409.rs:15:23
   |
15 |         (0, ref y) | (y, 0) => {} //~ ERROR E0409
   |                 -     ^ bound in different ways
   |                 |
   |                 first binding

error[E0308]: mismatched types
  --> $DIR/E0409.rs:15:23
   |
15 |         (0, ref y) | (y, 0) => {} //~ ERROR E0409
   |                       ^ expected &{integer}, found integral variable
   |
   = note: expected type `&{integer}`
              found type `{integer}`

error: aborting due to 2 previous errors

You've got a few errors: E0308, E0409
If you want more information on an error, try using "rustc --explain E0308"
