error[E0631]: type mismatch in closure arguments
  --> $DIR/E0631.rs:17:5
   |
17 |     foo(|_: isize| {}); //~ ERROR type mismatch
   |     ^^^ ---------- found signature of `fn(isize) -> _`
   |     |
   |     expected signature of `fn(usize) -> _`
   |
note: required by `foo`
  --> $DIR/E0631.rs:13:1
   |
13 | fn foo<F: Fn(usize)>(_: F) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0631]: type mismatch in closure arguments
  --> $DIR/E0631.rs:18:5
   |
18 |     bar(|_: isize| {}); //~ ERROR type mismatch
   |     ^^^ ---------- found signature of `fn(isize) -> _`
   |     |
   |     expected signature of `fn(usize) -> _`
   |
note: required by `bar`
  --> $DIR/E0631.rs:14:1
   |
14 | fn bar<F: Fn<usize>>(_: F) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0631]: type mismatch in function arguments
  --> $DIR/E0631.rs:19:5
   |
16 |     fn f(_: u64) {}
   |     ------------ found signature of `fn(u64) -> _`
...
19 |     foo(f); //~ ERROR type mismatch
   |     ^^^ expected signature of `fn(usize) -> _`
   |
note: required by `foo`
  --> $DIR/E0631.rs:13:1
   |
13 | fn foo<F: Fn(usize)>(_: F) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0631]: type mismatch in function arguments
  --> $DIR/E0631.rs:20:5
   |
16 |     fn f(_: u64) {}
   |     ------------ found signature of `fn(u64) -> _`
...
20 |     bar(f); //~ ERROR type mismatch
   |     ^^^ expected signature of `fn(usize) -> _`
   |
note: required by `bar`
  --> $DIR/E0631.rs:14:1
   |
14 | fn bar<F: Fn<usize>>(_: F) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 4 previous errors

