error[E0308]: mismatched types
  --> $DIR/issue-38371.rs:16:8
   |
16 | fn foo(&foo: Foo) { //~ ERROR mismatched types
   |        ^^^^ expected struct `Foo`, found reference
   |
   = note: expected type `Foo`
              found type `&_`
   = help: did you mean `foo: &Foo`?

error[E0308]: mismatched types
  --> $DIR/issue-38371.rs:30:9
   |
30 | fn agh(&&bar: &u32) { //~ ERROR mismatched types
   |         ^^^^ expected u32, found reference
   |
   = note: expected type `u32`
              found type `&_`
   = help: did you mean `bar: &u32`?

error[E0308]: mismatched types
  --> $DIR/issue-38371.rs:33:8
   |
33 | fn bgh(&&bar: u32) { //~ ERROR mismatched types
   |        ^^^^^ expected u32, found reference
   |
   = note: expected type `u32`
              found type `&_`

error[E0529]: expected an array or slice, found `u32`
  --> $DIR/issue-38371.rs:36:9
   |
36 | fn ugh(&[bar]: &u32) { //~ ERROR expected an array or slice
   |         ^^^^^ pattern cannot match with input type `u32`

error: aborting due to 4 previous errors

You've got a few errors: E0308, E0529
If you want more information on an error, try using "rustc --explain E0308"
