error[E0308]: mismatched types
  --> $DIR/attribute-spans-preserved.rs:19:23
   |
LL | #[ foo ( let y: u32 = "z"; ) ] //~ ERROR: mismatched types
   |                       ^^^ expected u32, found reference
   |
   = note: expected type `u32`
              found type `&'static str`

error[E0308]: mismatched types
  --> $DIR/attribute-spans-preserved.rs:20:21
   |
LL | #[ bar let x: u32 = "y"; ] //~ ERROR: mismatched types
   |                     ^^^ expected u32, found reference
   |
   = note: expected type `u32`
              found type `&'static str`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
