error[E0599]: no method named `closure` found for type `Obj<[closure@$DIR/issue-2392.rs:49:36: 49:41]>` in the current scope
  --> $DIR/issue-2392.rs:50:15
   |
50 |     o_closure.closure(); //~ ERROR no method named `closure` found
   |               ^^^^^^^ field, not a method
   |
   = help: use `(o_closure.closure)(...)` if you meant to call the function stored in the `closure` field

error[E0599]: no method named `not_closure` found for type `Obj<[closure@$DIR/issue-2392.rs:49:36: 49:41]>` in the current scope
  --> $DIR/issue-2392.rs:54:15
   |
54 |     o_closure.not_closure();
   |               ^^^^^^^^^^^ field, not a method
   |
   = help: did you mean to write `o_closure.not_closure` instead of `o_closure.not_closure(...)`?

error[E0599]: no method named `closure` found for type `Obj<fn() -> u32 {func}>` in the current scope
  --> $DIR/issue-2392.rs:60:12
   |
60 |     o_func.closure(); //~ ERROR no method named `closure` found
   |            ^^^^^^^ field, not a method
   |
   = help: use `(o_func.closure)(...)` if you meant to call the function stored in the `closure` field

error[E0599]: no method named `boxed_closure` found for type `BoxedObj` in the current scope
  --> $DIR/issue-2392.rs:65:14
   |
65 |     boxed_fn.boxed_closure();//~ ERROR no method named `boxed_closure` found
   |              ^^^^^^^^^^^^^ field, not a method
   |
   = help: use `(boxed_fn.boxed_closure)(...)` if you meant to call the function stored in the `boxed_closure` field

error[E0599]: no method named `boxed_closure` found for type `BoxedObj` in the current scope
  --> $DIR/issue-2392.rs:70:19
   |
70 |     boxed_closure.boxed_closure();//~ ERROR no method named `boxed_closure` found
   |                   ^^^^^^^^^^^^^ field, not a method
   |
   = help: use `(boxed_closure.boxed_closure)(...)` if you meant to call the function stored in the `boxed_closure` field

error[E0599]: no method named `closure` found for type `Obj<fn() -> u32 {func}>` in the current scope
  --> $DIR/issue-2392.rs:77:12
   |
77 |     w.wrap.closure();//~ ERROR no method named `closure` found
   |            ^^^^^^^ field, not a method
   |
   = help: use `(w.wrap.closure)(...)` if you meant to call the function stored in the `closure` field

error[E0599]: no method named `not_closure` found for type `Obj<fn() -> u32 {func}>` in the current scope
  --> $DIR/issue-2392.rs:81:12
   |
81 |     w.wrap.not_closure();
   |            ^^^^^^^^^^^ field, not a method
   |
   = help: did you mean to write `w.wrap.not_closure` instead of `w.wrap.not_closure(...)`?

error[E0599]: no method named `closure` found for type `Obj<std::boxed::Box<std::boxed::FnBox<(), Output=u32> + 'static>>` in the current scope
  --> $DIR/issue-2392.rs:86:24
   |
86 |     check_expression().closure();//~ ERROR no method named `closure` found
   |                        ^^^^^^^ field, not a method
   |
   = help: use `(check_expression().closure)(...)` if you meant to call the function stored in the `closure` field

error[E0599]: no method named `f1` found for type `FuncContainer` in the current scope
  --> $DIR/issue-2392.rs:94:31
   |
94 |             (*self.container).f1(1); //~ ERROR no method named `f1` found
   |                               ^^ field, not a method
   |
   = help: use `((*self.container).f1)(...)` if you meant to call the function stored in the `f1` field

error[E0599]: no method named `f2` found for type `FuncContainer` in the current scope
  --> $DIR/issue-2392.rs:97:31
   |
97 |             (*self.container).f2(1); //~ ERROR no method named `f2` found
   |                               ^^ field, not a method
   |
   = help: use `((*self.container).f2)(...)` if you meant to call the function stored in the `f2` field

error[E0599]: no method named `f3` found for type `FuncContainer` in the current scope
   --> $DIR/issue-2392.rs:100:31
    |
100 |             (*self.container).f3(1); //~ ERROR no method named `f3` found
    |                               ^^ field, not a method
    |
    = help: use `((*self.container).f3)(...)` if you meant to call the function stored in the `f3` field

error: aborting due to 11 previous errors

