error: attribute can only be applied to a trait
  --> $DIR/marker-attribute-on-non-trait.rs:13:1
   |
LL | #[marker] //~ ERROR attribute can only be applied to a trait
   | ^^^^^^^^^
LL | struct Struct {}
   | ---------------- not a trait

error: attribute can only be applied to a trait
  --> $DIR/marker-attribute-on-non-trait.rs:16:1
   |
LL | #[marker] //~ ERROR attribute can only be applied to a trait
   | ^^^^^^^^^
LL | impl Struct {}
   | -------------- not a trait

error: attribute can only be applied to a trait
  --> $DIR/marker-attribute-on-non-trait.rs:19:1
   |
LL |   #[marker] //~ ERROR attribute can only be applied to a trait
   |   ^^^^^^^^^
LL | / union Union {
LL | |     x: i32,
LL | | }
   | |_- not a trait

error: attribute can only be applied to a trait
  --> $DIR/marker-attribute-on-non-trait.rs:24:1
   |
LL | #[marker] //~ ERROR attribute can only be applied to a trait
   | ^^^^^^^^^
LL | const CONST: usize = 10;
   | ------------------------ not a trait

error: attribute can only be applied to a trait
  --> $DIR/marker-attribute-on-non-trait.rs:27:1
   |
LL | #[marker] //~ ERROR attribute can only be applied to a trait
   | ^^^^^^^^^
LL | fn function() {}
   | ---------------- not a trait

error: attribute can only be applied to a trait
  --> $DIR/marker-attribute-on-non-trait.rs:30:1
   |
LL | #[marker] //~ ERROR attribute can only be applied to a trait
   | ^^^^^^^^^
LL | type Type = ();
   | --------------- not a trait

error: aborting due to 6 previous errors

