error: private type `types::Priv` in public interface (error E0446)
  --> $DIR/private-in-public-warn.rs:25:5
   |
LL |     pub type Alias = Priv; //~ ERROR private type `types::Priv` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/private-in-public-warn.rs:15:9
   |
LL | #![deny(private_in_public)]
   |         ^^^^^^^^^^^^^^^^^
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private type `types::Priv` in public interface (error E0446)
  --> $DIR/private-in-public-warn.rs:28:12
   |
LL |         V1(Priv), //~ ERROR private type `types::Priv` in public interface
   |            ^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private type `types::Priv` in public interface (error E0446)
  --> $DIR/private-in-public-warn.rs:30:14
   |
LL |         V2 { field: Priv }, //~ ERROR private type `types::Priv` in public interface
   |              ^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private type `types::Priv` in public interface (error E0446)
  --> $DIR/private-in-public-warn.rs:34:9
   |
LL |         const C: Priv = Priv; //~ ERROR private type `types::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error[E0446]: private type `types::Priv` in public interface
  --> $DIR/private-in-public-warn.rs:36:9
   |
LL |     struct Priv;
   |     - `types::Priv` declared as private
...
LL |         type Alias = Priv; //~ ERROR private type `types::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^ can't leak private type

error: private type `types::Priv` in public interface (error E0446)
  --> $DIR/private-in-public-warn.rs:37:9
   |
LL |         fn f1(arg: Priv) {} //~ ERROR private type `types::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private type `types::Priv` in public interface (error E0446)
  --> $DIR/private-in-public-warn.rs:39:9
   |
LL |         fn f2() -> Priv { panic!() } //~ ERROR private type `types::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private type `types::Priv` in public interface (error E0446)
  --> $DIR/private-in-public-warn.rs:43:9
   |
LL |         pub static ES: Priv; //~ ERROR private type `types::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private type `types::Priv` in public interface (error E0446)
  --> $DIR/private-in-public-warn.rs:45:9
   |
LL |         pub fn ef1(arg: Priv); //~ ERROR private type `types::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private type `types::Priv` in public interface (error E0446)
  --> $DIR/private-in-public-warn.rs:47:9
   |
LL |         pub fn ef2() -> Priv; //~ ERROR private type `types::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error[E0446]: private type `types::Priv` in public interface
  --> $DIR/private-in-public-warn.rs:51:9
   |
LL |     struct Priv;
   |     - `types::Priv` declared as private
...
LL |         type Alias = Priv; //~ ERROR private type `types::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^ can't leak private type

error: private trait `traits::PrivTr` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:60:5
   |
LL |     pub type Alias<T: PrivTr> = T; //~ ERROR private trait `traits::PrivTr` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private trait `traits::PrivTr` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:62:5
   |
LL |     pub trait Tr1: PrivTr {} //~ ERROR private trait `traits::PrivTr` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private trait `traits::PrivTr` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:64:5
   |
LL |     pub trait Tr2<T: PrivTr> {} //~ ERROR private trait `traits::PrivTr` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private trait `traits::PrivTr` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:66:5
   |
LL | /     pub trait Tr3 {
LL | |         //~^ ERROR private trait `traits::PrivTr` in public interface
LL | |         //~| WARNING hard error
LL | |         type Alias: PrivTr;
LL | |         fn f<T: PrivTr>(arg: T) {} //~ ERROR private trait `traits::PrivTr` in public interface
LL | |         //~^ WARNING hard error
LL | |     }
   | |_____^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private trait `traits::PrivTr` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:70:9
   |
LL |         fn f<T: PrivTr>(arg: T) {} //~ ERROR private trait `traits::PrivTr` in public interface
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private trait `traits::PrivTr` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:73:5
   |
LL |     impl<T: PrivTr> Pub<T> {} //~ ERROR private trait `traits::PrivTr` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private trait `traits::PrivTr` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:75:5
   |
LL |     impl<T: PrivTr> PubTr for Pub<T> {} //~ ERROR private trait `traits::PrivTr` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private trait `traits_where::PrivTr` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:84:5
   |
LL |     pub type Alias<T> where T: PrivTr = T;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private trait `traits_where::PrivTr` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:87:5
   |
LL |     pub trait Tr2<T> where T: PrivTr {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private trait `traits_where::PrivTr` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:91:9
   |
LL |         fn f<T>(arg: T) where T: PrivTr {}
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private trait `traits_where::PrivTr` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:95:5
   |
LL |     impl<T> Pub<T> where T: PrivTr {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private trait `traits_where::PrivTr` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:98:5
   |
LL |     impl<T> PubTr for Pub<T> where T: PrivTr {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private trait `generics::PrivTr<generics::Pub>` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:109:5
   |
LL |     pub trait Tr1: PrivTr<Pub> {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private type `generics::Priv` in public interface (error E0446)
  --> $DIR/private-in-public-warn.rs:112:5
   |
LL |     pub trait Tr2: PubTr<Priv> {} //~ ERROR private type `generics::Priv` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private type `generics::Priv` in public interface (error E0446)
  --> $DIR/private-in-public-warn.rs:114:5
   |
LL |     pub trait Tr3: PubTr<[Priv; 1]> {} //~ ERROR private type `generics::Priv` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private type `generics::Priv` in public interface (error E0446)
  --> $DIR/private-in-public-warn.rs:116:5
   |
LL |     pub trait Tr4: PubTr<Pub<Priv>> {} //~ ERROR private type `generics::Priv` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error[E0446]: private type `impls::Priv` in public interface
  --> $DIR/private-in-public-warn.rs:143:9
   |
LL |     struct Priv;
   |     - `impls::Priv` declared as private
...
LL |         type Alias = Priv; //~ ERROR private type `impls::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^ can't leak private type

error: private type `aliases_pub::Priv` in public interface (error E0446)
  --> $DIR/private-in-public-warn.rs:214:9
   |
LL |         pub fn f(arg: Priv) {} //~ ERROR private type `aliases_pub::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error[E0446]: private type `aliases_pub::Priv` in public interface
  --> $DIR/private-in-public-warn.rs:218:9
   |
LL |     struct Priv;
   |     - `aliases_pub::Priv` declared as private
...
LL |         type Check = Priv; //~ ERROR private type `aliases_pub::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `aliases_pub::Priv` in public interface
  --> $DIR/private-in-public-warn.rs:221:9
   |
LL |     struct Priv;
   |     - `aliases_pub::Priv` declared as private
...
LL |         type Check = Priv; //~ ERROR private type `aliases_pub::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `aliases_pub::Priv` in public interface
  --> $DIR/private-in-public-warn.rs:224:9
   |
LL |     struct Priv;
   |     - `aliases_pub::Priv` declared as private
...
LL |         type Check = Priv; //~ ERROR private type `aliases_pub::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^ can't leak private type

error: private trait `aliases_priv::PrivTr1` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:248:5
   |
LL |     pub trait Tr1: PrivUseAliasTr {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private type `aliases_priv::Priv2` in public interface (error E0446)
  --> $DIR/private-in-public-warn.rs:251:5
   |
LL |     pub trait Tr2: PrivUseAliasTr<PrivAlias> {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: private trait `aliases_priv::PrivTr1<aliases_priv::Priv2>` in public interface (error E0445)
  --> $DIR/private-in-public-warn.rs:251:5
   |
LL |     pub trait Tr2: PrivUseAliasTr<PrivAlias> {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: aborting due to 35 previous errors

For more information about this error, try `rustc --explain E0446`.
