error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:23:1
   |
LL | pub impl Tr for S {  //~ ERROR unnecessary visibility qualifier
   | ^^^ `pub` not permitted here because it's implied

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:24:5
   |
LL |     pub fn f() {} //~ ERROR unnecessary visibility qualifier
   |     ^^^ `pub` not permitted here because it's implied

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:25:5
   |
LL |     pub const C: u8 = 0; //~ ERROR unnecessary visibility qualifier
   |     ^^^ `pub` not permitted here because it's implied

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:26:5
   |
LL |     pub type T = u8; //~ ERROR unnecessary visibility qualifier
   |     ^^^ `pub` not permitted here because it's implied

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:28:1
   |
LL | pub impl S { //~ ERROR unnecessary visibility qualifier
   | ^^^ `pub` not permitted here because it's implied
   |
   = note: place qualifiers on individual impl items instead

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:33:1
   |
LL | pub extern "C" { //~ ERROR unnecessary visibility qualifier
   | ^^^ `pub` not permitted here because it's implied
   |
   = note: place qualifiers on individual foreign items instead

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:49:5
   |
LL |     pub impl Tr for S {  //~ ERROR unnecessary visibility qualifier
   |     ^^^ `pub` not permitted here because it's implied

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:50:9
   |
LL |         pub fn f() {} //~ ERROR unnecessary visibility qualifier
   |         ^^^ `pub` not permitted here because it's implied

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:51:9
   |
LL |         pub const C: u8 = 0; //~ ERROR unnecessary visibility qualifier
   |         ^^^ `pub` not permitted here because it's implied

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:52:9
   |
LL |         pub type T = u8; //~ ERROR unnecessary visibility qualifier
   |         ^^^ `pub` not permitted here because it's implied

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:54:5
   |
LL |     pub impl S { //~ ERROR unnecessary visibility qualifier
   |     ^^^ `pub` not permitted here because it's implied
   |
   = note: place qualifiers on individual impl items instead

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:59:5
   |
LL |     pub extern "C" { //~ ERROR unnecessary visibility qualifier
   |     ^^^ `pub` not permitted here because it's implied
   |
   = note: place qualifiers on individual foreign items instead

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:78:5
   |
LL |     pub impl Tr for S {  //~ ERROR unnecessary visibility qualifier
   |     ^^^ `pub` not permitted here because it's implied

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:79:9
   |
LL |         pub fn f() {} //~ ERROR unnecessary visibility qualifier
   |         ^^^ `pub` not permitted here because it's implied

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:80:9
   |
LL |         pub const C: u8 = 0; //~ ERROR unnecessary visibility qualifier
   |         ^^^ `pub` not permitted here because it's implied

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:81:9
   |
LL |         pub type T = u8; //~ ERROR unnecessary visibility qualifier
   |         ^^^ `pub` not permitted here because it's implied

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:83:5
   |
LL |     pub impl S { //~ ERROR unnecessary visibility qualifier
   |     ^^^ `pub` not permitted here because it's implied
   |
   = note: place qualifiers on individual impl items instead

error[E0449]: unnecessary visibility qualifier
  --> $DIR/privacy-sanity.rs:88:5
   |
LL |     pub extern "C" { //~ ERROR unnecessary visibility qualifier
   |     ^^^ `pub` not permitted here because it's implied
   |
   = note: place qualifiers on individual foreign items instead

error: aborting due to 18 previous errors

For more information about this error, try `rustc --explain E0449`.
