error[E0446]: private type `types::Priv` in public interface
  --> $DIR/private-in-public.rs:23:5
   |
LL |     struct Priv;
   |     - `types::Priv` declared as private
...
LL |     pub const C: Priv = Priv; //~ ERROR private type `types::Priv` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `types::Priv` in public interface
  --> $DIR/private-in-public.rs:24:5
   |
LL |     struct Priv;
   |     - `types::Priv` declared as private
...
LL |     pub static S: Priv = Priv; //~ ERROR private type `types::Priv` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `types::Priv` in public interface
  --> $DIR/private-in-public.rs:25:5
   |
LL |     struct Priv;
   |     - `types::Priv` declared as private
...
LL |     pub fn f1(arg: Priv) {} //~ ERROR private type `types::Priv` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `types::Priv` in public interface
  --> $DIR/private-in-public.rs:26:5
   |
LL |     struct Priv;
   |     - `types::Priv` declared as private
...
LL |     pub fn f2() -> Priv { panic!() } //~ ERROR private type `types::Priv` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `types::Priv` in public interface
  --> $DIR/private-in-public.rs:27:19
   |
LL |     struct Priv;
   |     - `types::Priv` declared as private
...
LL |     pub struct S1(pub Priv); //~ ERROR private type `types::Priv` in public interface
   |                   ^^^^^^^^ can't leak private type

error[E0446]: private type `types::Priv` in public interface
  --> $DIR/private-in-public.rs:28:21
   |
LL |     struct Priv;
   |     - `types::Priv` declared as private
...
LL |     pub struct S2 { pub field: Priv } //~ ERROR private type `types::Priv` in public interface
   |                     ^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `types::Priv` in public interface
  --> $DIR/private-in-public.rs:30:9
   |
LL |     struct Priv;
   |     - `types::Priv` declared as private
...
LL |         pub const C: Priv = Priv; //~ ERROR private type `types::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `types::Priv` in public interface
  --> $DIR/private-in-public.rs:31:9
   |
LL |     struct Priv;
   |     - `types::Priv` declared as private
...
LL |         pub fn f1(arg: Priv) {} //~ ERROR private type `types::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `types::Priv` in public interface
  --> $DIR/private-in-public.rs:32:9
   |
LL |     struct Priv;
   |     - `types::Priv` declared as private
...
LL |         pub fn f2() -> Priv { panic!() } //~ ERROR private type `types::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0445]: private trait `traits::PrivTr` in public interface
  --> $DIR/private-in-public.rs:41:5
   |
LL |     pub enum E<T: PrivTr> { V(T) } //~ ERROR private trait `traits::PrivTr` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private trait

error[E0445]: private trait `traits::PrivTr` in public interface
  --> $DIR/private-in-public.rs:42:5
   |
LL |     pub fn f<T: PrivTr>(arg: T) {} //~ ERROR private trait `traits::PrivTr` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private trait

error[E0445]: private trait `traits::PrivTr` in public interface
  --> $DIR/private-in-public.rs:43:5
   |
LL |     pub struct S1<T: PrivTr>(T); //~ ERROR private trait `traits::PrivTr` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private trait

error[E0445]: private trait `traits::PrivTr` in public interface
  --> $DIR/private-in-public.rs:44:5
   |
LL | /     impl<T: PrivTr> Pub<T> { //~ ERROR private trait `traits::PrivTr` in public interface
LL | |         pub fn f<U: PrivTr>(arg: U) {} //~ ERROR private trait `traits::PrivTr` in public interface
LL | |     }
   | |_____^ can't leak private trait

error[E0445]: private trait `traits::PrivTr` in public interface
  --> $DIR/private-in-public.rs:45:9
   |
LL |         pub fn f<U: PrivTr>(arg: U) {} //~ ERROR private trait `traits::PrivTr` in public interface
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private trait

error[E0445]: private trait `traits_where::PrivTr` in public interface
  --> $DIR/private-in-public.rs:54:5
   |
LL |     pub enum E<T> where T: PrivTr { V(T) }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private trait

error[E0445]: private trait `traits_where::PrivTr` in public interface
  --> $DIR/private-in-public.rs:56:5
   |
LL |     pub fn f<T>(arg: T) where T: PrivTr {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private trait

error[E0445]: private trait `traits_where::PrivTr` in public interface
  --> $DIR/private-in-public.rs:58:5
   |
LL |     pub struct S1<T>(T) where T: PrivTr;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private trait

error[E0445]: private trait `traits_where::PrivTr` in public interface
  --> $DIR/private-in-public.rs:60:5
   |
LL | /     impl<T> Pub<T> where T: PrivTr {
LL | |     //~^ ERROR private trait `traits_where::PrivTr` in public interface
LL | |         pub fn f<U>(arg: U) where U: PrivTr {}
LL | |         //~^ ERROR private trait `traits_where::PrivTr` in public interface
LL | |     }
   | |_____^ can't leak private trait

error[E0445]: private trait `traits_where::PrivTr` in public interface
  --> $DIR/private-in-public.rs:62:9
   |
LL |         pub fn f<U>(arg: U) where U: PrivTr {}
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private trait

error[E0446]: private type `generics::Priv` in public interface
  --> $DIR/private-in-public.rs:73:5
   |
LL |     struct Priv<T = u8>(T);
   |     - `generics::Priv` declared as private
...
LL |     pub fn f1(arg: [Priv; 1]) {} //~ ERROR private type `generics::Priv` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `generics::Priv` in public interface
  --> $DIR/private-in-public.rs:74:5
   |
LL |     struct Priv<T = u8>(T);
   |     - `generics::Priv` declared as private
...
LL |     pub fn f2(arg: Pub<Priv>) {} //~ ERROR private type `generics::Priv` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `generics::Priv<generics::Pub>` in public interface
  --> $DIR/private-in-public.rs:75:5
   |
LL |     struct Priv<T = u8>(T);
   |     - `generics::Priv<generics::Pub>` declared as private
...
LL |     pub fn f3(arg: Priv<Pub>) {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `impls::Priv` in public interface
  --> $DIR/private-in-public.rs:90:9
   |
LL |     struct Priv;
   |     - `impls::Priv` declared as private
...
LL |         pub fn f(arg: Priv) {} //~ ERROR private type `impls::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `<aliases_pub::Priv as aliases_pub::PrivTr>::Assoc` in public interface
  --> $DIR/private-in-public.rs:114:5
   |
LL |     trait PrivTr {
   |     - `<aliases_pub::Priv as aliases_pub::PrivTr>::Assoc` declared as private
...
LL |     pub fn f3(arg: <Priv as PrivTr>::Assoc) {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `aliases_pub::Priv` in public interface
  --> $DIR/private-in-public.rs:114:5
   |
LL |     struct Priv;
   |     - `aliases_pub::Priv` declared as private
...
LL |     pub fn f3(arg: <Priv as PrivTr>::Assoc) {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `aliases_pub::Priv` in public interface
  --> $DIR/private-in-public.rs:119:9
   |
LL |     struct Priv;
   |     - `aliases_pub::Priv` declared as private
...
LL |         pub fn f(arg: Priv) {} //~ ERROR private type `aliases_pub::Priv` in public interface
   |         ^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `aliases_priv::Priv1` in public interface
  --> $DIR/private-in-public.rs:141:5
   |
LL |     struct Priv1;
   |     - `aliases_priv::Priv1` declared as private
...
LL |     pub fn f1(arg: PrivUseAlias) {} //~ ERROR private type `aliases_priv::Priv1` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `aliases_priv::Priv2` in public interface
  --> $DIR/private-in-public.rs:142:5
   |
LL |     struct Priv2;
   |     - `aliases_priv::Priv2` declared as private
...
LL |     pub fn f2(arg: PrivAlias) {} //~ ERROR private type `aliases_priv::Priv2` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `<aliases_priv::Priv as aliases_priv::PrivTr>::Assoc` in public interface
  --> $DIR/private-in-public.rs:143:5
   |
LL |     trait PrivTr {
   |     - `<aliases_priv::Priv as aliases_priv::PrivTr>::Assoc` declared as private
...
LL |     pub fn f3(arg: <Priv as PrivTr>::Assoc) {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `aliases_priv::Priv` in public interface
  --> $DIR/private-in-public.rs:143:5
   |
LL |     struct Priv;
   |     - `aliases_priv::Priv` declared as private
...
LL |     pub fn f3(arg: <Priv as PrivTr>::Assoc) {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `aliases_params::Priv` in public interface
  --> $DIR/private-in-public.rs:153:5
   |
LL |     struct Priv;
   |     - `aliases_params::Priv` declared as private
...
LL |     pub fn f2(arg: PrivAliasGeneric) {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error[E0446]: private type `aliases_params::Priv` in public interface
  --> $DIR/private-in-public.rs:155:5
   |
LL |     struct Priv;
   |     - `aliases_params::Priv` declared as private
...
LL |     pub fn f3(arg: Result<u8>) {} //~ ERROR private type `aliases_params::Priv` in public interface
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type

error: aborting due to 32 previous errors

Some errors occurred: E0445, E0446.
For more information about an error, try `rustc --explain E0445`.
