error[E0603]: tuple struct `A` is private
  --> $DIR/privacy5.rs:61:16
   |
LL |     let a = a::A(()); //~ ERROR tuple struct `A` is private
   |                ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:62:16
   |
LL |     let b = a::B(2); //~ ERROR tuple struct `B` is private
   |                ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:63:16
   |
LL |     let c = a::C(2, 3); //~ ERROR tuple struct `C` is private
   |                ^

error[E0603]: tuple struct `A` is private
  --> $DIR/privacy5.rs:66:12
   |
LL |     let a::A(()) = a; //~ ERROR tuple struct `A` is private
   |            ^

error[E0603]: tuple struct `A` is private
  --> $DIR/privacy5.rs:67:12
   |
LL |     let a::A(_) = a; //~ ERROR tuple struct `A` is private
   |            ^

error[E0603]: tuple struct `A` is private
  --> $DIR/privacy5.rs:68:18
   |
LL |     match a { a::A(()) => {} } //~ ERROR tuple struct `A` is private
   |                  ^

error[E0603]: tuple struct `A` is private
  --> $DIR/privacy5.rs:69:18
   |
LL |     match a { a::A(_) => {} } //~ ERROR tuple struct `A` is private
   |                  ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:71:12
   |
LL |     let a::B(_) = b; //~ ERROR tuple struct `B` is private
   |            ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:72:12
   |
LL |     let a::B(_b) = b; //~ ERROR tuple struct `B` is private
   |            ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:73:18
   |
LL |     match b { a::B(_) => {} } //~ ERROR tuple struct `B` is private
   |                  ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:74:18
   |
LL |     match b { a::B(_b) => {} } //~ ERROR tuple struct `B` is private
   |                  ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:75:18
   |
LL |     match b { a::B(1) => {} a::B(_) => {} } //~ ERROR tuple struct `B` is private
   |                  ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:75:32
   |
LL |     match b { a::B(1) => {} a::B(_) => {} } //~ ERROR tuple struct `B` is private
   |                                ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:78:12
   |
LL |     let a::C(_, _) = c; //~ ERROR tuple struct `C` is private
   |            ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:79:12
   |
LL |     let a::C(_a, _) = c; //~ ERROR tuple struct `C` is private
   |            ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:80:12
   |
LL |     let a::C(_, _b) = c; //~ ERROR tuple struct `C` is private
   |            ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:81:12
   |
LL |     let a::C(_a, _b) = c; //~ ERROR tuple struct `C` is private
   |            ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:82:18
   |
LL |     match c { a::C(_, _) => {} } //~ ERROR tuple struct `C` is private
   |                  ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:83:18
   |
LL |     match c { a::C(_a, _) => {} } //~ ERROR tuple struct `C` is private
   |                  ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:84:18
   |
LL |     match c { a::C(_, _b) => {} } //~ ERROR tuple struct `C` is private
   |                  ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:85:18
   |
LL |     match c { a::C(_a, _b) => {} } //~ ERROR tuple struct `C` is private
   |                  ^

error[E0603]: tuple struct `A` is private
  --> $DIR/privacy5.rs:93:17
   |
LL |     let a2 = a::A; //~ ERROR tuple struct `A` is private
   |                 ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:94:17
   |
LL |     let b2 = a::B; //~ ERROR tuple struct `B` is private
   |                 ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:95:17
   |
LL |     let c2 = a::C; //~ ERROR tuple struct `C` is private
   |                 ^

error[E0603]: tuple struct `A` is private
  --> $DIR/privacy5.rs:100:20
   |
LL |     let a = other::A(()); //~ ERROR tuple struct `A` is private
   |                    ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:101:20
   |
LL |     let b = other::B(2); //~ ERROR tuple struct `B` is private
   |                    ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:102:20
   |
LL |     let c = other::C(2, 3); //~ ERROR tuple struct `C` is private
   |                    ^

error[E0603]: tuple struct `A` is private
  --> $DIR/privacy5.rs:105:16
   |
LL |     let other::A(()) = a; //~ ERROR tuple struct `A` is private
   |                ^

error[E0603]: tuple struct `A` is private
  --> $DIR/privacy5.rs:106:16
   |
LL |     let other::A(_) = a; //~ ERROR tuple struct `A` is private
   |                ^

error[E0603]: tuple struct `A` is private
  --> $DIR/privacy5.rs:107:22
   |
LL |     match a { other::A(()) => {} } //~ ERROR tuple struct `A` is private
   |                      ^

error[E0603]: tuple struct `A` is private
  --> $DIR/privacy5.rs:108:22
   |
LL |     match a { other::A(_) => {} } //~ ERROR tuple struct `A` is private
   |                      ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:110:16
   |
LL |     let other::B(_) = b; //~ ERROR tuple struct `B` is private
   |                ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:111:16
   |
LL |     let other::B(_b) = b; //~ ERROR tuple struct `B` is private
   |                ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:112:22
   |
LL |     match b { other::B(_) => {} } //~ ERROR tuple struct `B` is private
   |                      ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:113:22
   |
LL |     match b { other::B(_b) => {} } //~ ERROR tuple struct `B` is private
   |                      ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:114:22
   |
LL |     match b { other::B(1) => {} other::B(_) => {} } //~ ERROR tuple struct `B` is private
   |                      ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:114:40
   |
LL |     match b { other::B(1) => {} other::B(_) => {} } //~ ERROR tuple struct `B` is private
   |                                        ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:117:16
   |
LL |     let other::C(_, _) = c; //~ ERROR tuple struct `C` is private
   |                ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:118:16
   |
LL |     let other::C(_a, _) = c; //~ ERROR tuple struct `C` is private
   |                ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:119:16
   |
LL |     let other::C(_, _b) = c; //~ ERROR tuple struct `C` is private
   |                ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:120:16
   |
LL |     let other::C(_a, _b) = c; //~ ERROR tuple struct `C` is private
   |                ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:121:22
   |
LL |     match c { other::C(_, _) => {} } //~ ERROR tuple struct `C` is private
   |                      ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:122:22
   |
LL |     match c { other::C(_a, _) => {} } //~ ERROR tuple struct `C` is private
   |                      ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:123:22
   |
LL |     match c { other::C(_, _b) => {} } //~ ERROR tuple struct `C` is private
   |                      ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:124:22
   |
LL |     match c { other::C(_a, _b) => {} } //~ ERROR tuple struct `C` is private
   |                      ^

error[E0603]: tuple struct `A` is private
  --> $DIR/privacy5.rs:132:21
   |
LL |     let a2 = other::A; //~ ERROR tuple struct `A` is private
   |                     ^

error[E0603]: tuple struct `B` is private
  --> $DIR/privacy5.rs:133:21
   |
LL |     let b2 = other::B; //~ ERROR tuple struct `B` is private
   |                     ^

error[E0603]: tuple struct `C` is private
  --> $DIR/privacy5.rs:134:21
   |
LL |     let c2 = other::C; //~ ERROR tuple struct `C` is private
   |                     ^

error: aborting due to 48 previous errors

For more information about this error, try `rustc --explain E0603`.
