error: unknown byte escape: f
  --> $DIR/byte-string-literals.rs:16:32
   |
LL | static FOO: &'static [u8] = b"/f";  //~ ERROR unknown byte escape
   |                                ^

error: unknown byte escape: f
  --> $DIR/byte-string-literals.rs:19:8
   |
LL |     b"/f";  //~ ERROR unknown byte escape
   |        ^

error: invalid character in numeric character escape: Z
  --> $DIR/byte-string-literals.rs:20:10
   |
LL |     b"/x0Z";  //~ ERROR invalid character in numeric character escape: Z
   |          ^

error: byte constant must be ASCII. Use a /xHH escape for a non-ASCII byte
  --> $DIR/byte-string-literals.rs:21:7
   |
LL |     b"Ã©";  //~ ERROR byte constant must be ASCII
   |       ^

error: unterminated double quote byte string
  --> $DIR/byte-string-literals.rs:22:7
   |
LL |       b"a  //~ ERROR unterminated double quote byte string
   |  _______^
LL | | }
   | |__^

error: aborting due to 5 previous errors

