error: negative trait bounds are not supported
  --> $DIR/issue-33418.rs:3:9
   |
LL | trait Tr: !SuperA {} //~ ERROR negative trait bounds are not supported
   |         ^^^^^^^^^ help: remove the trait bound

error: negative trait bounds are not supported
  --> $DIR/issue-33418.rs:4:19
   |
LL | trait Tr2: SuperA + !SuperB {} //~ ERROR negative trait bounds are not supported
   |          ---------^^^^^^^^^
   |          |
   |          help: remove the trait bound

error: negative trait bounds are not supported
  --> $DIR/issue-33418.rs:5:10
   |
LL | trait Tr3: !SuperA + SuperB {} //~ ERROR negative trait bounds are not supported
   |          ^^^^^^^^^---------
   |          |
   |          help: remove the trait bound

error: negative trait bounds are not supported
  --> $DIR/issue-33418.rs:6:10
   |
LL |   trait Tr4: !SuperA + SuperB //~ ERROR negative trait bounds are not supported
   |  __________-^^^^^^^^
LL | |     + !SuperC + SuperD {}
   | |_____^^^^^^^^^________- help: remove the trait bounds

error: negative trait bounds are not supported
  --> $DIR/issue-33418.rs:8:10
   |
LL |   trait Tr5: !SuperA //~ ERROR negative trait bounds are not supported
   |  __________-^^^^^^^^
LL | |     + !SuperB {}
   | |     ^^^^^^^^-
   | |_____________|
   |               help: remove the trait bounds

error: aborting due to 5 previous errors

