error: invalid digit for a base 2 literal
  --> $DIR/lex-bad-binary-literal.rs:14:8
   |
LL |     0b121; //~ ERROR invalid digit for a base 2 literal
   |        ^

error: invalid digit for a base 2 literal
  --> $DIR/lex-bad-binary-literal.rs:15:12
   |
LL |     0b10_10301; //~ ERROR invalid digit for a base 2 literal
   |            ^

error: invalid digit for a base 2 literal
  --> $DIR/lex-bad-binary-literal.rs:16:7
   |
LL |     0b30; //~ ERROR invalid digit for a base 2 literal
   |       ^

error: invalid digit for a base 2 literal
  --> $DIR/lex-bad-binary-literal.rs:17:7
   |
LL |     0b41; //~ ERROR invalid digit for a base 2 literal
   |       ^

error: invalid digit for a base 2 literal
  --> $DIR/lex-bad-binary-literal.rs:18:7
   |
LL |     0b5; //~ ERROR invalid digit for a base 2 literal
   |       ^

error: invalid digit for a base 2 literal
  --> $DIR/lex-bad-binary-literal.rs:19:7
   |
LL |     0b6; //~ ERROR invalid digit for a base 2 literal
   |       ^

error: invalid digit for a base 2 literal
  --> $DIR/lex-bad-binary-literal.rs:20:7
   |
LL |     0b7; //~ ERROR invalid digit for a base 2 literal
   |       ^

error: invalid digit for a base 2 literal
  --> $DIR/lex-bad-binary-literal.rs:21:7
   |
LL |     0b8; //~ ERROR invalid digit for a base 2 literal
   |       ^

error: invalid digit for a base 2 literal
  --> $DIR/lex-bad-binary-literal.rs:22:7
   |
LL |     0b9; //~ ERROR invalid digit for a base 2 literal
   |       ^

error: aborting due to 9 previous errors

