error: bare CR not allowed in doc-comment
  --> $DIR/lex-bare-cr-string-literal-doc-comment.rs:15:32
   |
LL | /// doc comment with bare CR: ''
   |                                ^

error: bare CR not allowed in block doc-comment
  --> $DIR/lex-bare-cr-string-literal-doc-comment.rs:19:38
   |
LL | /** block doc comment with bare CR: '' */
   |                                      ^

error: bare CR not allowed in doc-comment
  --> $DIR/lex-bare-cr-string-literal-doc-comment.rs:24:36
   |
LL |     //! doc comment with bare CR: ''
   |                                    ^

error: bare CR not allowed in block doc-comment
  --> $DIR/lex-bare-cr-string-literal-doc-comment.rs:27:42
   |
LL |     /*! block doc comment with bare CR: '' */
   |                                          ^

error: bare CR not allowed in string, use /r instead
  --> $DIR/lex-bare-cr-string-literal-doc-comment.rs:31:18
   |
LL |     let _s = "foobar"; //~ ERROR: bare CR not allowed in string
   |                  ^

error: bare CR not allowed in raw string, use /r instead
  --> $DIR/lex-bare-cr-string-literal-doc-comment.rs:34:14
   |
LL |     let _s = r"barfoo"; //~ ERROR: bare CR not allowed in raw string
   |              ^^^^^

error: unknown character escape: /r
  --> $DIR/lex-bare-cr-string-literal-doc-comment.rs:37:19
   |
LL |     let _s = "foo/bar"; //~ ERROR: unknown character escape: /r
   |                   ^
   |
help: this is an isolated carriage return; consider checking your editor and version control settings
  --> $DIR/lex-bare-cr-string-literal-doc-comment.rs:37:19
   |
LL |     let _s = "foo/bar"; //~ ERROR: unknown character escape: /r
   |                   ^

error: aborting due to 7 previous errors

