error: character literal may only contain one codepoint
  --> $DIR/lex-bad-char-literals-3.rs:3:5
   |
LL |     '●●' //~ ERROR: character literal may only contain one codepoint
   |     ^^^^
help: if you meant to write a `str` literal, use double quotes
   |
LL |     "●●" //~ ERROR: character literal may only contain one codepoint
   |     ^^^^

error[E0308]: mismatched types
  --> $DIR/lex-bad-char-literals-3.rs:3:5
   |
LL |     '●●' //~ ERROR: character literal may only contain one codepoint
   |     ^^^^ expected char, found reference
   |
   = note: expected type `char`
              found type `&'static str`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
