error: unexpected close delimiter: `}`
  --> $DIR/macro-mismatched-delim-paren-brace.rs:5:1
   |
LL | } //~ ERROR unexpected close delimiter: `}`
   | ^ unexpected close delimiter

error: incorrect close delimiter: `}`
  --> $DIR/macro-mismatched-delim-paren-brace.rs:4:5
   |
LL |     foo! (
   |          - un-closed delimiter
LL |         bar, "baz", 1, 2.0
LL |     } //~ ERROR incorrect close delimiter
   |     ^ incorrect close delimiter

error: aborting due to 2 previous errors

