error: invalid unicode character escape
  --> $DIR/new-unicode-escapes-3.rs:14:14
   |
LL |     let s1 = "/u{d805}"; //~ ERROR invalid unicode character escape
   |              ^^^^^^^^^^
   |
   = help: unicode escape must not be a surrogate

error: invalid unicode character escape
  --> $DIR/new-unicode-escapes-3.rs:15:14
   |
LL |     let s2 = "/u{ffffff}"; //~ ERROR invalid unicode character escape
   |              ^^^^^^^^^^^^
   |
   = help: unicode escape must be at most 10FFFF

error: aborting due to 2 previous errors

