error: expected identifier, found `0`
  --> $DIR/struct-field-numeric-shorthand.rs:4:19
   |
LL |     let _ = Rgb { 0, 1, 2 }; //~ ERROR expected identifier, found `0`
   |             ---   ^ expected identifier
   |             |
   |             while parsing this struct

error[E0063]: missing fields `0`, `1`, `2` in initializer of `Rgb`
  --> $DIR/struct-field-numeric-shorthand.rs:4:13
   |
LL |     let _ = Rgb { 0, 1, 2 }; //~ ERROR expected identifier, found `0`
   |             ^^^ missing `0`, `1`, `2`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0063`.
