error: parenthesized lifetime bounds are not supported
  --> $DIR/trait-object-lifetime-parens.rs:5:24
   |
LL | fn f<'a, T: Trait + ('a)>() {} //~ ERROR parenthesized lifetime bounds are not supported
   |                        ^

error: parenthesized lifetime bounds are not supported
  --> $DIR/trait-object-lifetime-parens.rs:8:27
   |
LL |     let _: Box<Trait + ('a)>; //~ ERROR parenthesized lifetime bounds are not supported
   |                           ^

error: expected type, found `'a`
  --> $DIR/trait-object-lifetime-parens.rs:9:17
   |
LL |     let _: Box<('a) + Trait>; //~ ERROR expected type, found `'a`
   |         -       ^^
   |         |
   |         while parsing the type for `_`

error: aborting due to 3 previous errors

