error: unreachable expression
  --> $DIR/expr_unary.rs:18:28
   |
18 |     let x: ! = ! { return; 22 }; //~ ERROR unreachable
   |                            ^^
   |
note: lint level defined here
  --> $DIR/expr_unary.rs:14:9
   |
14 | #![deny(unreachable_code)]
   |         ^^^^^^^^^^^^^^^^

error[E0600]: cannot apply unary operator `!` to type `!`
  --> $DIR/expr_unary.rs:18:16
   |
18 |     let x: ! = ! { return; 22 }; //~ ERROR unreachable
   |                ^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

