error: unreachable expression
  --> $DIR/expr_array.rs:20:34
   |
LL |     let x: [usize; 2] = [return, 22]; //~ ERROR unreachable
   |                                  ^^
   |
note: lint level defined here
  --> $DIR/expr_array.rs:14:9
   |
LL | #![deny(unreachable_code)]
   |         ^^^^^^^^^^^^^^^^

error: unreachable expression
  --> $DIR/expr_array.rs:25:25
   |
LL |     let x: [usize; 2] = [22, return]; //~ ERROR unreachable
   |                         ^^^^^^^^^^^^

error: aborting due to 2 previous errors

