error: unreachable expression
  --> $DIR/expr_assign.rs:20:5
   |
LL |     x = return; //~ ERROR unreachable
   |     ^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/expr_assign.rs:14:9
   |
LL | #![deny(unreachable_code)]
   |         ^^^^^^^^^^^^^^^^

error: unreachable expression
  --> $DIR/expr_assign.rs:30:14
   |
LL |         *p = return; //~ ERROR unreachable
   |              ^^^^^^

error: unreachable expression
  --> $DIR/expr_assign.rs:36:15
   |
LL |     *{return; &mut i} = 22; //~ ERROR unreachable
   |               ^^^^^^

error: aborting due to 3 previous errors

