error: unreachable expression
  --> $DIR/expr_struct.rs:25:13
   |
LL |     let x = Foo { a: 22, b: 33, ..return }; //~ ERROR unreachable
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/expr_struct.rs:14:9
   |
LL | #![deny(unreachable_code)]
   |         ^^^^^^^^^^^^^^^^

error: unreachable expression
  --> $DIR/expr_struct.rs:30:33
   |
LL |     let x = Foo { a: return, b: 33, ..return }; //~ ERROR unreachable
   |                                 ^^

error: unreachable expression
  --> $DIR/expr_struct.rs:35:39
   |
LL |     let x = Foo { a: 22, b: return, ..return }; //~ ERROR unreachable
   |                                       ^^^^^^

error: unreachable expression
  --> $DIR/expr_struct.rs:40:13
   |
LL |     let x = Foo { a: 22, b: return }; //~ ERROR unreachable
   |             ^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 4 previous errors

