error: unreachable expression
  --> $DIR/expr_tup.rs:20:38
   |
LL |     let x: (usize, usize) = (return, 2); //~ ERROR unreachable
   |                                      ^
   |
note: lint level defined here
  --> $DIR/expr_tup.rs:14:9
   |
LL | #![deny(unreachable_code)]
   |         ^^^^^^^^^^^^^^^^

error: unreachable expression
  --> $DIR/expr_tup.rs:25:29
   |
LL |     let x: (usize, usize) = (2, return); //~ ERROR unreachable
   |                             ^^^^^^^^^^^

error: aborting due to 2 previous errors

