warning: `$crate` may not be imported
  --> $DIR/import-crate-var.rs:19:5
   |
LL |     m!();
   |     ^^^^^
   |
   = note: `use $crate;` was erroneously allowed and will become a hard error in a future release
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: compilation successful
  --> $DIR/import-crate-var.rs:18:1
   |
LL | / fn main() { //~ ERROR compilation successful
LL | |     m!();
LL | |     //~^ WARN `$crate` may not be imported
LL | |     //~| NOTE `use $crate;` was erroneously allowed and will become a hard error
LL | | }
   | |_^

error: aborting due to previous error

