error: `m` is ambiguous
  --> $DIR/macros.rs:50:5
   |
50 |     m!(); //~ ERROR ambiguous
   |     ^
   |
note: `m` could refer to the macro defined here
  --> $DIR/macros.rs:48:5
   |
48 |     macro_rules! m { () => {} }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: `m` could also refer to the macro imported here
  --> $DIR/macros.rs:49:9
   |
49 |     use two_macros::m;
   |         ^^^^^^^^^^^^^

error: `m` is ambiguous
  --> $DIR/macros.rs:28:5
   |
28 |     m! { //~ ERROR ambiguous
   |     ^
   |
note: `m` could refer to the name imported here
  --> $DIR/macros.rs:29:13
   |
29 |         use foo::m;
   |             ^^^^^^
note: `m` could also refer to the name imported here
  --> $DIR/macros.rs:27:9
   |
27 |     use two_macros::*;
   |         ^^^^^^^^^^^^^
   = note: macro-expanded macro imports do not shadow

error: `m` is ambiguous
  --> $DIR/macros.rs:41:9
   |
41 |         m! { //~ ERROR ambiguous
   |         ^
   |
note: `m` could refer to the name imported here
  --> $DIR/macros.rs:42:17
   |
42 |             use two_macros::n as m;
   |                 ^^^^^^^^^^^^^^^^^^
note: `m` could also refer to the name imported here
  --> $DIR/macros.rs:34:9
   |
34 |     use two_macros::m;
   |         ^^^^^^^^^^^^^
   = note: macro-expanded macro imports do not shadow

error[E0601]: main function not found

error: aborting due to 4 previous errors

