error[E0659]: `bar` is ambiguous
  --> $DIR/macro-paths.rs:23:5
   |
LL |     bar::m! { //~ ERROR ambiguous
   |     ^^^ ambiguous name
   |
note: `bar` could refer to the name defined here
  --> $DIR/macro-paths.rs:24:9
   |
LL |         mod bar { pub use two_macros::m; }
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: `bar` could also refer to the name imported here
  --> $DIR/macro-paths.rs:22:9
   |
LL |     use foo::*;
   |         ^^^^^^
   = note: macro-expanded items do not shadow when used in a macro invocation path

error[E0659]: `baz` is ambiguous
  --> $DIR/macro-paths.rs:33:5
   |
LL |     baz::m! { //~ ERROR ambiguous
   |     ^^^ ambiguous name
   |
note: `baz` could refer to the name defined here
  --> $DIR/macro-paths.rs:34:9
   |
LL |         mod baz { pub use two_macros::m; }
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: `baz` could also refer to the name defined here
  --> $DIR/macro-paths.rs:28:1
   |
LL | / pub mod baz {
LL | |     pub use two_macros::m;
LL | | }
   | |_^
   = note: macro-expanded items do not shadow when used in a macro invocation path

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0659`.
