error: `panic` is already in scope
  --> $DIR/shadow_builtin_macros.rs:42:9
   |
42 |         macro_rules! panic { () => {} } //~ ERROR `panic` is already in scope
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
43 |     } }
44 |     m!();
   |     ----- in this macro invocation
   |
   = note: macro-expanded `macro_rules!`s may not shadow existing macros (see RFC 1560)

error: `panic` is ambiguous
  --> $DIR/shadow_builtin_macros.rs:27:14
   |
27 |     fn f() { panic!(); } //~ ERROR ambiguous
   |              ^^^^^
   |
note: `panic` could refer to the name imported here
  --> $DIR/shadow_builtin_macros.rs:26:9
   |
26 |     use foo::*;
   |         ^^^^^^
   = note: `panic` is also a builtin macro
   = note: consider adding an explicit import of `panic` to disambiguate

error: `panic` is ambiguous
  --> $DIR/shadow_builtin_macros.rs:32:14
   |
32 |     fn f() { panic!(); } //~ ERROR ambiguous
   |              ^^^^^
   |
note: `panic` could refer to the name imported here
  --> $DIR/shadow_builtin_macros.rs:31:26
   |
31 |     ::two_macros::m!(use foo::panic;);
   |                          ^^^^^^^^^^
   = note: `panic` is also a builtin macro
   = note: macro-expanded macro imports do not shadow

error: `n` is ambiguous
  --> $DIR/shadow_builtin_macros.rs:61:5
   |
61 |     n!(); //~ ERROR ambiguous
   |     ^
   |
note: `n` could refer to the name imported here
  --> $DIR/shadow_builtin_macros.rs:60:9
   |
60 |     use bar::*;
   |         ^^^^^^
note: `n` could also refer to the name imported here
  --> $DIR/shadow_builtin_macros.rs:48:13
   |
48 | #[macro_use(n)]
   |             ^
   = note: consider adding an explicit import of `n` to disambiguate

error: aborting due to 4 previous errors

