error[E0412]: cannot find type `FromOutside` in this scope
  --> $DIR/generate-mod.rs:45:13
   |
LL |     genmod!(FromOutside, Outer); //~ ERROR cannot find type `FromOutside` in this scope
   |             ^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `Outer` in this scope
  --> $DIR/generate-mod.rs:45:26
   |
LL |     genmod!(FromOutside, Outer); //~ ERROR cannot find type `FromOutside` in this scope
   |                          ^^^^^ not found in this scope

error[E0412]: cannot find type `FromOutside` in this scope
  --> $DIR/generate-mod.rs:29:18
   |
LL |         type A = FromOutside; //~ ERROR cannot find type `FromOutside` in this scope
   |                  ^^^^^^^^^^^ not found in this scope
...
LL |     genmod_transparent!();
   |     ---------------------- in this macro invocation

error[E0412]: cannot find type `Outer` in this scope
  --> $DIR/generate-mod.rs:30:22
   |
LL |         type Inner = Outer; //~ ERROR cannot find type `Outer` in this scope
   |                      ^^^^^ not found in this scope
...
LL |     genmod_transparent!();
   |     ---------------------- in this macro invocation

error[E0412]: cannot find type `FromOutside` in this scope
  --> $DIR/generate-mod.rs:38:18
   |
LL |         type A = FromOutside; //~ ERROR cannot find type `FromOutside` in this scope
   |                  ^^^^^^^^^^^ not found in this scope
...
LL |     genmod_legacy!();
   |     ----------------- in this macro invocation

error[E0412]: cannot find type `Outer` in this scope
  --> $DIR/generate-mod.rs:39:22
   |
LL |         type Inner = Outer; //~ ERROR cannot find type `Outer` in this scope
   |                      ^^^^^ not found in this scope
...
LL |     genmod_legacy!();
   |     ----------------- in this macro invocation

error[E0601]: `main` function not found in crate `generate_mod`
   |
   = note: consider adding a `main` function to `$DIR/generate-mod.rs`

error: aborting due to 7 previous errors

Some errors occurred: E0412, E0601.
For more information about an error, try `rustc --explain E0412`.
