error[E0411]: expected trait, found self type `Self`
  --> $DIR/resolve-self-in-impl-2.rs:14:6
   |
LL | impl Self for S {} //~ ERROR expected trait, found self type `Self`
   |      ^^^^ `Self` is only available in traits and impls

error[E0405]: cannot find trait `N` in `Self`
  --> $DIR/resolve-self-in-impl-2.rs:15:12
   |
LL | impl Self::N for S {} //~ ERROR cannot find trait `N` in `Self`
   |            ^ not found in `Self`

error: aborting due to 2 previous errors

Some errors occurred: E0405, E0411.
For more information about an error, try `rustc --explain E0405`.
