error[E0412]: cannot find type `Path` in this scope
  --> $DIR/use_suggestion_placement.rs:27:16
   |
27 |     type Bar = Path; //~ ERROR cannot find
   |                ^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
   |
23 |     use std::path::Path;
   |

error[E0425]: cannot find value `A` in this scope
  --> $DIR/use_suggestion_placement.rs:32:13
   |
32 |     let _ = A; //~ ERROR cannot find
   |             ^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
   |
13 | use m::A;
   |

error[E0412]: cannot find type `HashMap` in this scope
  --> $DIR/use_suggestion_placement.rs:37:23
   |
37 |     type Dict<K, V> = HashMap<K, V>; //~ ERROR cannot find
   |                       ^^^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
   |
13 | use std::collections::HashMap;
   |
13 | use std::collections::hash_map::HashMap;
   |

error: aborting due to 3 previous errors

