error: expected type, found `0`
  --> $DIR/type-ascription-instead-of-statement-end.rs:15:5
   |
LL |     println!("test"):
   |                     - help: did you mean to use `;` here?
LL |     0; //~ ERROR expected type, found `0`
   |     ^ expecting a type here because of type ascription

error: expected type, found `0`
  --> $DIR/type-ascription-instead-of-statement-end.rs:19:23
   |
LL |     println!("test"): 0; //~ ERROR expected type, found `0`
   |                       ^ expecting a type here because of type ascription

error: aborting due to 2 previous errors

