warning: not reporting region error due to nll
  --> $DIR/ex2b-push-no-existing-names.rs:16:12
   |
LL |     x.push(y); //~ ERROR lifetime mismatch
   |            ^

error: unsatisfied lifetime constraints
  --> $DIR/ex2b-push-no-existing-names.rs:16:5
   |
LL | fn foo(x: &mut Vec<Ref<i32>>, y: Ref<i32>) {
   |                    --------      -------- lifetime `'1` appears in this type
   |                    |
   |                    lifetime `'2` appears in this type
LL |     x.push(y); //~ ERROR lifetime mismatch
   |     ^^^^^^^^^ argument requires that `'1` must outlive `'2`

error: aborting due to previous error

