warning: not reporting region error due to nll
  --> $DIR/ex3-both-anon-regions-self-is-anon.rs:17:19
   |
LL |         if true { x } else { self } //~ ERROR lifetime mismatch
   |                   ^

error: unsatisfied lifetime constraints
  --> $DIR/ex3-both-anon-regions-self-is-anon.rs:17:9
   |
LL |     fn foo<'a>(&self, x: &Foo) -> &Foo {
   |                -         - let's call the lifetime of this reference `'1`
   |                |
   |                let's call the lifetime of this reference `'2`
LL |         if true { x } else { self } //~ ERROR lifetime mismatch
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'1` must outlive `'2`

error: aborting due to previous error

