warning: not reporting region error due to nll
  --> $DIR/ex3-both-anon-regions-one-is-struct.rs:17:11
   |
LL |     x.b = y; //~ ERROR lifetime mismatch
   |           ^

error: unsatisfied lifetime constraints
  --> $DIR/ex3-both-anon-regions-one-is-struct.rs:17:5
   |
LL | fn foo(mut x: Ref, y: &u32) {
   |               ---     - let's call the lifetime of this reference `'1`
   |               |
   |               lifetime `'2` appears in this type
LL |     x.b = y; //~ ERROR lifetime mismatch
   |     ^^^^^^^ requires that `'1` must outlive `'2`

error: aborting due to previous error

