warning: not reporting region error due to -Znll
  --> $DIR/dyn-trait-underscore.rs:20:14
   |
LL |     Box::new(items.iter()) //~ ERROR cannot infer an appropriate lifetime
   |              ^^^^^

warning: not reporting region error due to -Znll
  --> $DIR/dyn-trait-underscore.rs:20:20
   |
LL |     Box::new(items.iter()) //~ ERROR cannot infer an appropriate lifetime
   |                    ^^^^

warning: not reporting region error due to -Znll
  --> $DIR/dyn-trait-underscore.rs:20:5
   |
LL |     Box::new(items.iter()) //~ ERROR cannot infer an appropriate lifetime
   |     ^^^^^^^^

warning: not reporting region error due to -Znll
  --> $DIR/dyn-trait-underscore.rs:20:5
   |
LL |     Box::new(items.iter()) //~ ERROR cannot infer an appropriate lifetime
   |     ^^^^^^^^^^^^^^^^^^^^^^

error: free region `` does not outlive free region `'static`
  --> $DIR/dyn-trait-underscore.rs:18:52
   |
LL |   fn a<T>(items: &[T]) -> Box<dyn Iterator<Item=&T>> {
   |  ____________________________________________________^
LL | |     //                      ^^^^^^^^^^^^^^^^^^^^^ bound *here* defaults to `'static`
LL | |     Box::new(items.iter()) //~ ERROR cannot infer an appropriate lifetime
LL | | }
   | |_^

error: aborting due to previous error

