warning: not reporting region error due to nll
  --> $DIR/dyn-trait-underscore.rs:18:14
   |
LL |     Box::new(items.iter()) //~ ERROR cannot infer an appropriate lifetime
   |              ^^^^^

warning: not reporting region error due to nll
  --> $DIR/dyn-trait-underscore.rs:18:20
   |
LL |     Box::new(items.iter()) //~ ERROR cannot infer an appropriate lifetime
   |                    ^^^^

warning: not reporting region error due to nll
  --> $DIR/dyn-trait-underscore.rs:18:5
   |
LL |     Box::new(items.iter()) //~ ERROR cannot infer an appropriate lifetime
   |     ^^^^^^^^

warning: not reporting region error due to nll
  --> $DIR/dyn-trait-underscore.rs:18:5
   |
LL |     Box::new(items.iter()) //~ ERROR cannot infer an appropriate lifetime
   |     ^^^^^^^^^^^^^^^^^^^^^^

error: cast requires that data must outlive free region `'static`
  --> $DIR/dyn-trait-underscore.rs:18:5
   |
LL |     Box::new(items.iter()) //~ ERROR cannot infer an appropriate lifetime
   |     ^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

