warning: not reporting region error due to nll
  --> $DIR/underscore-lifetime-elison-mismatch.rs:11:49
   |
LL | fn foo(x: &mut Vec<&'_ u8>, y: &'_ u8) { x.push(y); } //~ ERROR lifetime mismatch
   |                                                 ^

error: unsatisfied lifetime constraints
  --> $DIR/underscore-lifetime-elison-mismatch.rs:11:42
   |
LL | fn foo(x: &mut Vec<&'_ u8>, y: &'_ u8) { x.push(y); } //~ ERROR lifetime mismatch
   |                    -           -         ^^^^^^^^^ argument requires that `'1` must outlive `'2`
   |                    |           |
   |                    |           let's call the lifetime of this reference `'1`
   |                    let's call the lifetime of this reference `'2`

error: aborting due to previous error

