error[E0659]: `m` is ambiguous
  --> $DIR/ambiguity-legacy-vs-modern.rs:31:9
   |
LL |         m!() //~ ERROR `m` is ambiguous
   |         ^ ambiguous name
   |
note: `m` could refer to the name defined here
  --> $DIR/ambiguity-legacy-vs-modern.rs:26:5
   |
LL |     macro_rules! m { () => (()) }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: `m` could also refer to the name defined here
  --> $DIR/ambiguity-legacy-vs-modern.rs:29:9
   |
LL |         macro m() { 0 }
   |         ^^^^^^^^^^^^^^^

error[E0659]: `m` is ambiguous
  --> $DIR/ambiguity-legacy-vs-modern.rs:43:5
   |
LL |     m!() //~ ERROR `m` is ambiguous
   |     ^ ambiguous name
   |
note: `m` could refer to the name defined here
  --> $DIR/ambiguity-legacy-vs-modern.rs:40:9
   |
LL |         macro_rules! m { () => (()) }
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: `m` could also refer to the name defined here
  --> $DIR/ambiguity-legacy-vs-modern.rs:36:5
   |
LL |     macro m() { 0 }
   |     ^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0659`.
