error: macro requires a boolean expression as an argument
  --> $DIR/assert.rs:2:5
   |
LL |     assert!();  //~ ERROR requires a boolean expression
   |     ^^^^^^^^^^ boolean expression required

error: macro requires a boolean expression as an argument
  --> $DIR/assert.rs:3:5
   |
LL |     debug_assert!(); //~ ERROR requires a boolean expression
   |     ^^^^^^^^^^^^^^^^ boolean expression required
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to 2 previous errors

