error: expected a literal
  --> $DIR/bad_hello.rs:12:14
   |
LL |     println!(3 + 4); //~ ERROR expected a literal
   |              ^^^^^
help: you might be missing a string literal to format with
   |
LL |     println!("{}", 3 + 4); //~ ERROR expected a literal
   |              ^^^^^^^^^^^

error: aborting due to previous error

