error: multiple unused formatting arguments
  --> $DIR/format-unused-lables.rs:12:22
   |
12 |     println!("Test", 123, 456, 789);
   |     -----------------^^^--^^^--^^^-- multiple unused arguments in this statement
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: multiple unused formatting arguments
  --> $DIR/format-unused-lables.rs:16:9
   |
15 | /     println!("Test2",
16 | |         123,  //~ ERROR multiple unused formatting arguments
   | |         ^^^
17 | |         456,
   | |         ^^^
18 | |         789
   | |         ^^^
19 | |     );
   | |______- multiple unused arguments in this statement
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: named argument never used
  --> $DIR/format-unused-lables.rs:21:35
   |
21 |     println!("Some stuff", UNUSED="args"); //~ ERROR named argument never used
   |                                   ^^^^^^

error: multiple unused formatting arguments
  --> $DIR/format-unused-lables.rs:24:9
   |
23 | /     println!("Some more $STUFF",
24 | |         "woo!",  //~ ERROR multiple unused formatting arguments
   | |         ^^^^^^
25 | |             STUFF=
26 | |        "things"
   | |        ^^^^^^^^
27 | |              , UNUSED="args");
   | |_______________________^^^^^^_- multiple unused arguments in this statement
   |
   = help: `$STUFF` should be written as `{STUFF}`
   = note: shell formatting not supported; see the documentation for `std::fmt`
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to 4 previous errors

