error: cannot find derive macro `inline` in this scope
  --> $DIR/macro-path-prelude-fail-3.rs:11:10
   |
LL | #[derive(inline)] //~ ERROR cannot find derive macro `inline` in this scope
   |          ^^^^^^

error: cannot find macro `inline!` in this scope
  --> $DIR/macro-path-prelude-fail-3.rs:15:5
   |
LL |     inline!(); //~ ERROR cannot find macro `inline!` in this scope
   |     ^^^^^^ help: you could try the macro: `line`

error: aborting due to 2 previous errors

