error: `macro_two` is already in scope
  --> $DIR/macro-shadowing.rs:22:5
   |
LL |     #[macro_use] //~ ERROR `macro_two` is already in scope
   |     ^^^^^^^^^^^^
...
LL | m1!();
   | ------ in this macro invocation
   |
   = note: macro-expanded `#[macro_use]`s may not shadow existing macros (see RFC 1560)

error[E0659]: `foo` is ambiguous
  --> $DIR/macro-shadowing.rs:27:1
   |
LL | foo!(); //~ ERROR `foo` is ambiguous
   | ^^^ ambiguous name
   |
note: `foo` could refer to the name defined here
  --> $DIR/macro-shadowing.rs:20:5
   |
LL |     macro_rules! foo { () => {} }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
LL | m1!();
   | ------ in this macro invocation
note: `foo` could also refer to the name defined here
  --> $DIR/macro-shadowing.rs:15:1
   |
LL | macro_rules! foo { () => {} }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: macro-expanded macros do not shadow

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0659`.
