error: expected `*` or `+`
  --> $DIR/macro-at-most-once-rep-2015-ques-rep-feature-flag.rs:20:10
   |
LL |     ($(a)?) => {} //~ERROR expected `*` or `+`
   |          ^
   |
   = note: `?` is not a macro repetition operator

error: expected `*` or `+`
  --> $DIR/macro-at-most-once-rep-2015-ques-rep-feature-flag.rs:24:11
   |
LL |     ($(a),?) => {} //~ERROR expected `*` or `+`
   |           ^
   |
   = note: `?` is not a macro repetition operator

error: aborting due to 2 previous errors

