error[E0599]: no method named `fake` found for type `{integer}` in the current scope
  --> $DIR/macro-backtrace-invalid-internals.rs:15:13
   |
LL |           1.fake() //~ ERROR no method
   |             ^^^^
...
LL |     fake_method_stmt!();
   |     -------------------- in this macro invocation

error[E0610]: `{integer}` is a primitive type and therefore doesn't have fields
  --> $DIR/macro-backtrace-invalid-internals.rs:21:13
   |
LL |           1.fake //~ ERROR doesn't have fields
   |             ^^^^
...
LL |     fake_field_stmt!();
   |     ------------------- in this macro invocation

error[E0609]: no field `0` on type `{integer}`
  --> $DIR/macro-backtrace-invalid-internals.rs:27:11
   |
LL |           (1).0 //~ ERROR no field
   |           ^^^^^
...
LL |     fake_anon_field_stmt!();
   |     ------------------------ in this macro invocation

error[E0689]: can't call method `powi` on ambiguous numeric type `{float}`
  --> $DIR/macro-backtrace-invalid-internals.rs:51:15
   |
LL |           2.0.powi(2) //~ ERROR can't call method `powi` on ambiguous numeric type `{float}`
   |               ^^^^
...
LL |     real_method_stmt!();
   |     -------------------- in this macro invocation
help: you must specify a concrete type for this numeric value, like `f32`
   |
LL |           2.0_f32.powi(2) //~ ERROR can't call method `powi` on ambiguous numeric type `{float}`
   |           ^^^^^^^

error[E0599]: no method named `fake` found for type `{integer}` in the current scope
  --> $DIR/macro-backtrace-invalid-internals.rs:33:13
   |
LL |           1.fake() //~ ERROR no method
   |             ^^^^
...
LL |     let _ = fake_method_expr!();
   |             ------------------- in this macro invocation

error[E0610]: `{integer}` is a primitive type and therefore doesn't have fields
  --> $DIR/macro-backtrace-invalid-internals.rs:39:13
   |
LL |           1.fake //~ ERROR doesn't have fields
   |             ^^^^
...
LL |     let _ = fake_field_expr!();
   |             ------------------ in this macro invocation

error[E0609]: no field `0` on type `{integer}`
  --> $DIR/macro-backtrace-invalid-internals.rs:45:11
   |
LL |           (1).0 //~ ERROR no field
   |           ^^^^^
...
LL |     let _ = fake_anon_field_expr!();
   |             ----------------------- in this macro invocation

error[E0689]: can't call method `powi` on ambiguous numeric type `{float}`
  --> $DIR/macro-backtrace-invalid-internals.rs:57:15
   |
LL |           2.0.powi(2) //~ ERROR can't call method `powi` on ambiguous numeric type `{float}`
   |               ^^^^
...
LL |     let _ = real_method_expr!();
   |             ------------------- in this macro invocation
help: you must specify a concrete type for this numeric value, like `f32`
   |
LL |           2.0_f32.powi(2) //~ ERROR can't call method `powi` on ambiguous numeric type `{float}`
   |           ^^^^^^^

error: aborting due to 8 previous errors

You've got a few errors: E0599, E0609, E0610, E0689
If you want more information on an error, try using "rustc --explain E0599"
