error: lel
  --> $DIR/macro-comma-support.rs:16:5
   |
LL |     compile_error!("lel"); //~ ERROR lel
   |     ^^^^^^^^^^^^^^^^^^^^^^

error: lel
  --> $DIR/macro-comma-support.rs:17:5
   |
LL |     compile_error!("lel",); //~ ERROR lel
   |     ^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

