error[E0601]: main function not found

error[E0518]: attribute should be applied to function
  --> $DIR/issue-43106-gating-of-inline.rs:21:1
   |
21 |   #[inline = "2100"]
   |   ^^^^^^^^^^^^^^^^^^
22 |   //~^ ERROR attribute should be applied to function
23 | / mod inline {
24 | |     mod inner { #![inline="2100"] }
25 | |     //~^ ERROR attribute should be applied to function
26 | |
...  |
36 | |     //~^ ERROR attribute should be applied to function
37 | | }
   | |_- not a function

error[E0518]: attribute should be applied to function
  --> $DIR/issue-43106-gating-of-inline.rs:24:17
   |
24 |     mod inner { #![inline="2100"] }
   |     ------------^^^^^^^^^^^^^^^^^-- not a function

error[E0518]: attribute should be applied to function
  --> $DIR/issue-43106-gating-of-inline.rs:29:5
   |
29 |     #[inline = "2100"] struct S;
   |     ^^^^^^^^^^^^^^^^^^ --------- not a function

error[E0518]: attribute should be applied to function
  --> $DIR/issue-43106-gating-of-inline.rs:32:5
   |
32 |     #[inline = "2100"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^ ----------- not a function

error[E0518]: attribute should be applied to function
  --> $DIR/issue-43106-gating-of-inline.rs:35:5
   |
35 |     #[inline = "2100"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^ ---------- not a function

error: aborting due to 6 previous errors

