warning: macro_escape is a deprecated synonym for macro_use
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:532:1
    |
532 | #[macro_escape]
    | ^^^^^^^^^^^^^^^

warning: macro_escape is a deprecated synonym for macro_use
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:535:17
    |
535 |     mod inner { #![macro_escape] }
    |                 ^^^^^^^^^^^^^^^^
    |
    = help: consider an outer attribute, #[macro_use] mod ...

warning: `#[must_use]` on functions is experimental (see issue #43302)
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:682:5
    |
682 |     #[must_use = "1400"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^
    |
    = help: add #![feature(fn_must_use)] to the crate attributes to enable

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:49:33
   |
49 | #![warn                        (x5400)] //~ WARN unknown lint: `x5400`
   |                                 ^^^^^
   |
note: lint level defined here
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:44:28
   |
44 | #![warn(unused_attributes, unknown_lints)]
   |                            ^^^^^^^^^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:50:33
   |
50 | #![allow                       (x5300)] //~ WARN unknown lint: `x5300`
   |                                 ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:51:33
   |
51 | #![forbid                      (x5200)] //~ WARN unknown lint: `x5200`
   |                                 ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:52:33
   |
52 | #![deny                        (x5100)] //~ WARN unknown lint: `x5100`
   |                                 ^^^^^

warning: unknown lint: `x5400`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:114:8
    |
114 | #[warn(x5400)]
    |        ^^^^^

warning: unknown lint: `x5400`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:117:25
    |
117 |     mod inner { #![warn(x5400)] }
    |                         ^^^^^

warning: unknown lint: `x5400`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:120:12
    |
120 |     #[warn(x5400)] fn f() { }
    |            ^^^^^

warning: unknown lint: `x5400`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:123:12
    |
123 |     #[warn(x5400)] struct S;
    |            ^^^^^

warning: unknown lint: `x5400`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:126:12
    |
126 |     #[warn(x5400)] type T = S;
    |            ^^^^^

warning: unknown lint: `x5400`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:129:12
    |
129 |     #[warn(x5400)] impl S { }
    |            ^^^^^

warning: unknown lint: `x5300`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:133:9
    |
133 | #[allow(x5300)]
    |         ^^^^^

warning: unknown lint: `x5300`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:136:26
    |
136 |     mod inner { #![allow(x5300)] }
    |                          ^^^^^

warning: unknown lint: `x5300`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:139:13
    |
139 |     #[allow(x5300)] fn f() { }
    |             ^^^^^

warning: unknown lint: `x5300`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:142:13
    |
142 |     #[allow(x5300)] struct S;
    |             ^^^^^

warning: unknown lint: `x5300`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:145:13
    |
145 |     #[allow(x5300)] type T = S;
    |             ^^^^^

warning: unknown lint: `x5300`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:148:13
    |
148 |     #[allow(x5300)] impl S { }
    |             ^^^^^

warning: unknown lint: `x5200`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:152:10
    |
152 | #[forbid(x5200)]
    |          ^^^^^

warning: unknown lint: `x5200`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:155:27
    |
155 |     mod inner { #![forbid(x5200)] }
    |                           ^^^^^

warning: unknown lint: `x5200`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:158:14
    |
158 |     #[forbid(x5200)] fn f() { }
    |              ^^^^^

warning: unknown lint: `x5200`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:161:14
    |
161 |     #[forbid(x5200)] struct S;
    |              ^^^^^

warning: unknown lint: `x5200`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:164:14
    |
164 |     #[forbid(x5200)] type T = S;
    |              ^^^^^

warning: unknown lint: `x5200`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:167:14
    |
167 |     #[forbid(x5200)] impl S { }
    |              ^^^^^

warning: unknown lint: `x5100`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:171:8
    |
171 | #[deny(x5100)]
    |        ^^^^^

warning: unknown lint: `x5100`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:174:25
    |
174 |     mod inner { #![deny(x5100)] }
    |                         ^^^^^

warning: unknown lint: `x5100`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:177:12
    |
177 |     #[deny(x5100)] fn f() { }
    |            ^^^^^

warning: unknown lint: `x5100`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:180:12
    |
180 |     #[deny(x5100)] struct S;
    |            ^^^^^

warning: unknown lint: `x5100`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:183:12
    |
183 |     #[deny(x5100)] type T = S;
    |            ^^^^^

warning: unknown lint: `x5100`
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:186:12
    |
186 |     #[deny(x5100)] impl S { }
    |            ^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:193:17
    |
193 |     mod inner { #![macro_reexport="5000"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^
    |
note: lint level defined here
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:44:9
    |
44  | #![warn(unused_attributes, unknown_lints)]
    |         ^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:196:5
    |
196 |     #[macro_reexport = "5000"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:199:5
    |
199 |     #[macro_reexport = "5000"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:202:5
    |
202 |     #[macro_reexport = "5000"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:205:5
    |
205 |     #[macro_reexport = "5000"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:190:1
    |
190 | #[macro_reexport = "5000"]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:213:5
    |
213 |     #[macro_use] fn f() { }
    |     ^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:216:5
    |
216 |     #[macro_use] struct S;
    |     ^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:219:5
    |
219 |     #[macro_use] type T = S;
    |     ^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:222:5
    |
222 |     #[macro_use] impl S { }
    |     ^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:229:17
    |
229 |     mod inner { #![macro_export="4800"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:232:5
    |
232 |     #[macro_export = "4800"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:235:5
    |
235 |     #[macro_export = "4800"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:238:5
    |
238 |     #[macro_export = "4800"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:241:5
    |
241 |     #[macro_export = "4800"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:226:1
    |
226 | #[macro_export = "4800"]
    | ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:248:17
    |
248 |     mod inner { #![plugin_registrar="4700"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:253:5
    |
253 |     #[plugin_registrar = "4700"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:256:5
    |
256 |     #[plugin_registrar = "4700"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:259:5
    |
259 |     #[plugin_registrar = "4700"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:245:1
    |
245 | #[plugin_registrar = "4700"]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:266:17
    |
266 |     mod inner { #![main="4300"] }
    |                 ^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:271:5
    |
271 |     #[main = "4400"] struct S;
    |     ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:274:5
    |
274 |     #[main = "4400"] type T = S;
    |     ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:277:5
    |
277 |     #[main = "4400"] impl S { }
    |     ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:263:1
    |
263 | #[main = "4400"]
    | ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:284:17
    |
284 |     mod inner { #![start="4300"] }
    |                 ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:289:5
    |
289 |     #[start = "4300"] struct S;
    |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:292:5
    |
292 |     #[start = "4300"] type T = S;
    |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:295:5
    |
295 |     #[start = "4300"] impl S { }
    |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:281:1
    |
281 | #[start = "4300"]
    | ^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:334:17
    |
334 |     mod inner { #![simd="4000"] }
    |                 ^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:337:5
    |
337 |     #[simd = "4000"] fn f() { }
    |     ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:342:5
    |
342 |     #[simd = "4000"] type T = S;
    |     ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:345:5
    |
345 |     #[simd = "4000"] impl S { }
    |     ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:331:1
    |
331 | #[simd = "4000"]
    | ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:352:17
    |
352 |     mod inner { #![repr="3900"] }
    |                 ^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:355:5
    |
355 |     #[repr = "3900"] fn f() { }
    |     ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:360:5
    |
360 |     #[repr = "3900"] type T = S;
    |     ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:363:5
    |
363 |     #[repr = "3900"] impl S { }
    |     ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:349:1
    |
349 | #[repr = "3900"]
    | ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:371:5
    |
371 |     #[path = "3800"] fn f() { }
    |     ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:374:5
    |
374 |     #[path = "3800"]  struct S;
    |     ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:377:5
    |
377 |     #[path = "3800"] type T = S;
    |     ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:380:5
    |
380 |     #[path = "3800"] impl S { }
    |     ^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:387:17
    |
387 |     mod inner { #![abi="3700"] }
    |                 ^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:390:5
    |
390 |     #[abi = "3700"] fn f() { }
    |     ^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:393:5
    |
393 |     #[abi = "3700"] struct S;
    |     ^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:396:5
    |
396 |     #[abi = "3700"] type T = S;
    |     ^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:399:5
    |
399 |     #[abi = "3700"] impl S { }
    |     ^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:384:1
    |
384 | #[abi = "3700"]
    | ^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:406:17
    |
406 |     mod inner { #![automatically_derived="3600"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:409:5
    |
409 |     #[automatically_derived = "3600"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:412:5
    |
412 |     #[automatically_derived = "3600"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:415:5
    |
415 |     #[automatically_derived = "3600"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:418:5
    |
418 |     #[automatically_derived = "3600"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:403:1
    |
403 | #[automatically_derived = "3600"]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: function is marked #[no_mangle], but not exported
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:426:27
    |
426 |     #[no_mangle = "3500"] fn f() { }
    |                           -^^^^^^^^^
    |                           |
    |                           help: try making it public: `pub `
    |
    = note: #[warn(private_no_mangle_fns)] on by default

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:439:17
    |
439 |     mod inner { #![no_link="3400"] }
    |                 ^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:442:5
    |
442 |     #[no_link = "3400"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:445:5
    |
445 |     #[no_link = "3400"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:448:5
    |
448 |     #[no_link = "3400"]type T = S;
    |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:451:5
    |
451 |     #[no_link = "3400"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:436:1
    |
436 | #[no_link = "3400"]
    | ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:458:17
    |
458 |     mod inner { #![should_panic="3200"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:461:5
    |
461 |     #[should_panic = "3200"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:464:5
    |
464 |     #[should_panic = "3200"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:467:5
    |
467 |     #[should_panic = "3200"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:470:5
    |
470 |     #[should_panic = "3200"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:455:1
    |
455 | #[should_panic = "3200"]
    | ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:477:17
    |
477 |     mod inner { #![ignore="3100"] }
    |                 ^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:480:5
    |
480 |     #[ignore = "3100"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:483:5
    |
483 |     #[ignore = "3100"] struct S;
    |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:486:5
    |
486 |     #[ignore = "3100"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:489:5
    |
489 |     #[ignore = "3100"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:474:1
    |
474 | #[ignore = "3100"]
    | ^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:496:17
    |
496 |     mod inner { #![no_implicit_prelude="3000"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:499:5
    |
499 |     #[no_implicit_prelude = "3000"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:502:5
    |
502 |     #[no_implicit_prelude = "3000"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:505:5
    |
505 |     #[no_implicit_prelude = "3000"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:508:5
    |
508 |     #[no_implicit_prelude = "3000"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:493:1
    |
493 | #[no_implicit_prelude = "3000"]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:515:17
    |
515 |     mod inner { #![reexport_test_harness_main="2900"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:518:5
    |
518 |     #[reexport_test_harness_main = "2900"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:521:5
    |
521 |     #[reexport_test_harness_main = "2900"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:524:5
    |
524 |     #[reexport_test_harness_main = "2900"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:527:5
    |
527 |     #[reexport_test_harness_main = "2900"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:512:1
    |
512 | #[reexport_test_harness_main = "2900"]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:538:5
    |
538 |     #[macro_escape] fn f() { }
    |     ^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:541:5
    |
541 |     #[macro_escape] struct S;
    |     ^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:544:5
    |
544 |     #[macro_escape] type T = S;
    |     ^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:547:5
    |
547 |     #[macro_escape] impl S { }
    |     ^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:555:17
    |
555 |     mod inner { #![no_std="2600"] }
    |                 ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:555:17
    |
555 |     mod inner { #![no_std="2600"] }
    |                 ^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:559:5
    |
559 |     #[no_std = "2600"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:559:5
    |
559 |     #[no_std = "2600"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:563:5
    |
563 |     #[no_std = "2600"] struct S;
    |     ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:563:5
    |
563 |     #[no_std = "2600"] struct S;
    |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:567:5
    |
567 |     #[no_std = "2600"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:567:5
    |
567 |     #[no_std = "2600"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:571:5
    |
571 |     #[no_std = "2600"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:571:5
    |
571 |     #[no_std = "2600"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:551:1
    |
551 | #[no_std = "2600"]
    | ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:551:1
    |
551 | #[no_std = "2600"]
    | ^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:711:17
    |
711 |     mod inner { #![crate_name="0900"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:711:17
    |
711 |     mod inner { #![crate_name="0900"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:715:5
    |
715 |     #[crate_name = "0900"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:715:5
    |
715 |     #[crate_name = "0900"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:719:5
    |
719 |     #[crate_name = "0900"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:719:5
    |
719 |     #[crate_name = "0900"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:723:5
    |
723 |     #[crate_name = "0900"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:723:5
    |
723 |     #[crate_name = "0900"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:727:5
    |
727 |     #[crate_name = "0900"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:727:5
    |
727 |     #[crate_name = "0900"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:707:1
    |
707 | #[crate_name = "0900"]
    | ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:707:1
    |
707 | #[crate_name = "0900"]
    | ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:736:17
    |
736 |     mod inner { #![crate_type="0800"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:736:17
    |
736 |     mod inner { #![crate_type="0800"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:740:5
    |
740 |     #[crate_type = "0800"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:740:5
    |
740 |     #[crate_type = "0800"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:744:5
    |
744 |     #[crate_type = "0800"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:744:5
    |
744 |     #[crate_type = "0800"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:748:5
    |
748 |     #[crate_type = "0800"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:748:5
    |
748 |     #[crate_type = "0800"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:752:5
    |
752 |     #[crate_type = "0800"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:752:5
    |
752 |     #[crate_type = "0800"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:732:1
    |
732 | #[crate_type = "0800"]
    | ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:732:1
    |
732 | #[crate_type = "0800"]
    | ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:761:17
    |
761 |     mod inner { #![feature(x0600)] }
    |                 ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:761:17
    |
761 |     mod inner { #![feature(x0600)] }
    |                 ^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:765:5
    |
765 |     #[feature(x0600)] fn f() { }
    |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:765:5
    |
765 |     #[feature(x0600)] fn f() { }
    |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:769:5
    |
769 |     #[feature(x0600)] struct S;
    |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:769:5
    |
769 |     #[feature(x0600)] struct S;
    |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:773:5
    |
773 |     #[feature(x0600)] type T = S;
    |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:773:5
    |
773 |     #[feature(x0600)] type T = S;
    |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:777:5
    |
777 |     #[feature(x0600)] impl S { }
    |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:777:5
    |
777 |     #[feature(x0600)] impl S { }
    |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:757:1
    |
757 | #[feature(x0600)]
    | ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:757:1
    |
757 | #[feature(x0600)]
    | ^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:787:17
    |
787 |     mod inner { #![no_main="0400"] }
    |                 ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:787:17
    |
787 |     mod inner { #![no_main="0400"] }
    |                 ^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:791:5
    |
791 |     #[no_main = "0400"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:791:5
    |
791 |     #[no_main = "0400"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:795:5
    |
795 |     #[no_main = "0400"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:795:5
    |
795 |     #[no_main = "0400"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:799:5
    |
799 |     #[no_main = "0400"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:799:5
    |
799 |     #[no_main = "0400"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:803:5
    |
803 |     #[no_main = "0400"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:803:5
    |
803 |     #[no_main = "0400"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:783:1
    |
783 | #[no_main = "0400"]
    | ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:783:1
    |
783 | #[no_main = "0400"]
    | ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:825:17
    |
825 |     mod inner { #![recursion_limit="0200"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:825:17
    |
825 |     mod inner { #![recursion_limit="0200"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:829:5
    |
829 |     #[recursion_limit="0200"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:829:5
    |
829 |     #[recursion_limit="0200"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:833:5
    |
833 |     #[recursion_limit="0200"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:833:5
    |
833 |     #[recursion_limit="0200"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:837:5
    |
837 |     #[recursion_limit="0200"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:837:5
    |
837 |     #[recursion_limit="0200"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:841:5
    |
841 |     #[recursion_limit="0200"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:841:5
    |
841 |     #[recursion_limit="0200"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:821:1
    |
821 | #[recursion_limit="0200"]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:821:1
    |
821 | #[recursion_limit="0200"]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:850:17
    |
850 |     mod inner { #![type_length_limit="0100"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:850:17
    |
850 |     mod inner { #![type_length_limit="0100"] }
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:854:5
    |
854 |     #[type_length_limit="0100"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:854:5
    |
854 |     #[type_length_limit="0100"] fn f() { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:858:5
    |
858 |     #[type_length_limit="0100"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:858:5
    |
858 |     #[type_length_limit="0100"] struct S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:862:5
    |
862 |     #[type_length_limit="0100"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:862:5
    |
862 |     #[type_length_limit="0100"] type T = S;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:866:5
    |
866 |     #[type_length_limit="0100"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:866:5
    |
866 |     #[type_length_limit="0100"] impl S { }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:846:1
    |
846 | #[type_length_limit="0100"]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:846:1
    |
846 | #[type_length_limit="0100"]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:53:1
   |
53 | #![macro_reexport             = "5000"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:55:1
   |
55 | #![macro_export               = "4800"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:56:1
   |
56 | #![plugin_registrar           = "4700"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:59:1
   |
59 | #![main                      = "x4400"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:60:1
   |
60 | #![start                     = "x4300"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:63:1
   |
63 | #![simd                       = "4000"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:64:1
   |
64 | #![repr                       = "3900"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:65:1
   |
65 | #![path                       = "3800"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:66:1
   |
66 | #![abi                        = "3700"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:67:1
   |
67 | #![automatically_derived      = "3600"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:69:1
   |
69 | #![no_link                    = "3400"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:71:1
   |
71 | #![should_panic               = "3200"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:72:1
   |
72 | #![ignore                     = "3100"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:78:1
   |
78 | #![proc_macro_derive          = "2500"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: compilation successful
   --> $DIR/issue-43106-gating-of-builtin-attrs.rs:877:1
    |
877 | / fn main() { //~ ERROR compilation successful
878 | |     println!("Hello World");
879 | | }
    | |_^

