warning: macro_escape is a deprecated synonym for macro_use
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:493:1
   |
LL | #[macro_escape]
   | ^^^^^^^^^^^^^^^

warning: macro_escape is a deprecated synonym for macro_use
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:496:17
   |
LL |     mod inner { #![macro_escape] }
   |                 ^^^^^^^^^^^^^^^^
   |
   = help: consider an outer attribute, #[macro_use] mod ...

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:49:33
   |
LL | #![warn                        (x5400)] //~ WARN unknown lint: `x5400`
   |                                 ^^^^^
   |
note: lint level defined here
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:44:28
   |
LL | #![warn(unused_attributes, unknown_lints)]
   |                            ^^^^^^^^^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:50:33
   |
LL | #![allow                       (x5300)] //~ WARN unknown lint: `x5300`
   |                                 ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:51:33
   |
LL | #![forbid                      (x5200)] //~ WARN unknown lint: `x5200`
   |                                 ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:52:33
   |
LL | #![deny                        (x5100)] //~ WARN unknown lint: `x5100`
   |                                 ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:112:8
   |
LL | #[warn(x5400)]
   |        ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:115:25
   |
LL |     mod inner { #![warn(x5400)] }
   |                         ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:118:12
   |
LL |     #[warn(x5400)] fn f() { }
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:121:12
   |
LL |     #[warn(x5400)] struct S;
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:124:12
   |
LL |     #[warn(x5400)] type T = S;
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:127:12
   |
LL |     #[warn(x5400)] impl S { }
   |            ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:131:9
   |
LL | #[allow(x5300)]
   |         ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:134:26
   |
LL |     mod inner { #![allow(x5300)] }
   |                          ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:137:13
   |
LL |     #[allow(x5300)] fn f() { }
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:140:13
   |
LL |     #[allow(x5300)] struct S;
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:143:13
   |
LL |     #[allow(x5300)] type T = S;
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:146:13
   |
LL |     #[allow(x5300)] impl S { }
   |             ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:150:10
   |
LL | #[forbid(x5200)]
   |          ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:153:27
   |
LL |     mod inner { #![forbid(x5200)] }
   |                           ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:156:14
   |
LL |     #[forbid(x5200)] fn f() { }
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:159:14
   |
LL |     #[forbid(x5200)] struct S;
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:162:14
   |
LL |     #[forbid(x5200)] type T = S;
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:165:14
   |
LL |     #[forbid(x5200)] impl S { }
   |              ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:169:8
   |
LL | #[deny(x5100)]
   |        ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:172:25
   |
LL |     mod inner { #![deny(x5100)] }
   |                         ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:175:12
   |
LL |     #[deny(x5100)] fn f() { }
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:178:12
   |
LL |     #[deny(x5100)] struct S;
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:181:12
   |
LL |     #[deny(x5100)] type T = S;
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:184:12
   |
LL |     #[deny(x5100)] impl S { }
   |            ^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:192:5
   |
LL |     #[macro_use] fn f() { }
   |     ^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:44:9
   |
LL | #![warn(unused_attributes, unknown_lints)]
   |         ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:195:5
   |
LL |     #[macro_use] struct S;
   |     ^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:198:5
   |
LL |     #[macro_use] type T = S;
   |     ^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:201:5
   |
LL |     #[macro_use] impl S { }
   |     ^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:208:17
   |
LL |     mod inner { #![macro_export="4800"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:211:5
   |
LL |     #[macro_export = "4800"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:214:5
   |
LL |     #[macro_export = "4800"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:217:5
   |
LL |     #[macro_export = "4800"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:220:5
   |
LL |     #[macro_export = "4800"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:205:1
   |
LL | #[macro_export = "4800"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:227:17
   |
LL |     mod inner { #![plugin_registrar="4700"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:232:5
   |
LL |     #[plugin_registrar = "4700"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:235:5
   |
LL |     #[plugin_registrar = "4700"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:238:5
   |
LL |     #[plugin_registrar = "4700"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:224:1
   |
LL | #[plugin_registrar = "4700"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:245:17
   |
LL |     mod inner { #![main="4300"] }
   |                 ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:250:5
   |
LL |     #[main = "4400"] struct S;
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:253:5
   |
LL |     #[main = "4400"] type T = S;
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:256:5
   |
LL |     #[main = "4400"] impl S { }
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:242:1
   |
LL | #[main = "4400"]
   | ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:263:17
   |
LL |     mod inner { #![start="4300"] }
   |                 ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:268:5
   |
LL |     #[start = "4300"] struct S;
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:271:5
   |
LL |     #[start = "4300"] type T = S;
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:274:5
   |
LL |     #[start = "4300"] impl S { }
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:260:1
   |
LL | #[start = "4300"]
   | ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:313:17
   |
LL |     mod inner { #![repr="3900"] }
   |                 ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:316:5
   |
LL |     #[repr = "3900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:321:5
   |
LL |     #[repr = "3900"] type T = S;
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:324:5
   |
LL |     #[repr = "3900"] impl S { }
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:310:1
   |
LL | #[repr = "3900"]
   | ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:332:5
   |
LL |     #[path = "3800"] fn f() { }
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:335:5
   |
LL |     #[path = "3800"]  struct S;
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:338:5
   |
LL |     #[path = "3800"] type T = S;
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:341:5
   |
LL |     #[path = "3800"] impl S { }
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:348:17
   |
LL |     mod inner { #![abi="3700"] }
   |                 ^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:351:5
   |
LL |     #[abi = "3700"] fn f() { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:354:5
   |
LL |     #[abi = "3700"] struct S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:357:5
   |
LL |     #[abi = "3700"] type T = S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:360:5
   |
LL |     #[abi = "3700"] impl S { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:345:1
   |
LL | #[abi = "3700"]
   | ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:367:17
   |
LL |     mod inner { #![automatically_derived="3600"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:370:5
   |
LL |     #[automatically_derived = "3600"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:373:5
   |
LL |     #[automatically_derived = "3600"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:376:5
   |
LL |     #[automatically_derived = "3600"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:379:5
   |
LL |     #[automatically_derived = "3600"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:364:1
   |
LL | #[automatically_derived = "3600"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: function is marked #[no_mangle], but not exported
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:387:27
   |
LL |     #[no_mangle = "3500"] fn f() { }
   |                           -^^^^^^^^^
   |                           |
   |                           help: try making it public: `pub`
   |
   = note: #[warn(private_no_mangle_fns)] on by default

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:400:17
   |
LL |     mod inner { #![no_link="3400"] }
   |                 ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:403:5
   |
LL |     #[no_link = "3400"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:406:5
   |
LL |     #[no_link = "3400"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:409:5
   |
LL |     #[no_link = "3400"]type T = S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:412:5
   |
LL |     #[no_link = "3400"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:397:1
   |
LL | #[no_link = "3400"]
   | ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:419:17
   |
LL |     mod inner { #![should_panic="3200"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:422:5
   |
LL |     #[should_panic = "3200"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:425:5
   |
LL |     #[should_panic = "3200"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:428:5
   |
LL |     #[should_panic = "3200"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:431:5
   |
LL |     #[should_panic = "3200"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:416:1
   |
LL | #[should_panic = "3200"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:438:17
   |
LL |     mod inner { #![ignore="3100"] }
   |                 ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:441:5
   |
LL |     #[ignore = "3100"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:444:5
   |
LL |     #[ignore = "3100"] struct S;
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:447:5
   |
LL |     #[ignore = "3100"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:450:5
   |
LL |     #[ignore = "3100"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:435:1
   |
LL | #[ignore = "3100"]
   | ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:457:17
   |
LL |     mod inner { #![no_implicit_prelude="3000"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:460:5
   |
LL |     #[no_implicit_prelude = "3000"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:463:5
   |
LL |     #[no_implicit_prelude = "3000"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:466:5
   |
LL |     #[no_implicit_prelude = "3000"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:469:5
   |
LL |     #[no_implicit_prelude = "3000"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:454:1
   |
LL | #[no_implicit_prelude = "3000"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:476:17
   |
LL |     mod inner { #![reexport_test_harness_main="2900"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:479:5
   |
LL |     #[reexport_test_harness_main = "2900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:482:5
   |
LL |     #[reexport_test_harness_main = "2900"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:485:5
   |
LL |     #[reexport_test_harness_main = "2900"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:488:5
   |
LL |     #[reexport_test_harness_main = "2900"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:473:1
   |
LL | #[reexport_test_harness_main = "2900"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:499:5
   |
LL |     #[macro_escape] fn f() { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:502:5
   |
LL |     #[macro_escape] struct S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:505:5
   |
LL |     #[macro_escape] type T = S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:508:5
   |
LL |     #[macro_escape] impl S { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:516:17
   |
LL |     mod inner { #![no_std="2600"] }
   |                 ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:516:17
   |
LL |     mod inner { #![no_std="2600"] }
   |                 ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:520:5
   |
LL |     #[no_std = "2600"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:520:5
   |
LL |     #[no_std = "2600"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:524:5
   |
LL |     #[no_std = "2600"] struct S;
   |     ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:524:5
   |
LL |     #[no_std = "2600"] struct S;
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:528:5
   |
LL |     #[no_std = "2600"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:528:5
   |
LL |     #[no_std = "2600"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:532:5
   |
LL |     #[no_std = "2600"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:532:5
   |
LL |     #[no_std = "2600"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:512:1
   |
LL | #[no_std = "2600"]
   | ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:512:1
   |
LL | #[no_std = "2600"]
   | ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:671:17
   |
LL |     mod inner { #![crate_name="0900"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:671:17
   |
LL |     mod inner { #![crate_name="0900"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:675:5
   |
LL |     #[crate_name = "0900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:675:5
   |
LL |     #[crate_name = "0900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:679:5
   |
LL |     #[crate_name = "0900"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:679:5
   |
LL |     #[crate_name = "0900"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:683:5
   |
LL |     #[crate_name = "0900"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:683:5
   |
LL |     #[crate_name = "0900"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:687:5
   |
LL |     #[crate_name = "0900"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:687:5
   |
LL |     #[crate_name = "0900"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:667:1
   |
LL | #[crate_name = "0900"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:667:1
   |
LL | #[crate_name = "0900"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:696:17
   |
LL |     mod inner { #![crate_type="0800"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:696:17
   |
LL |     mod inner { #![crate_type="0800"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:700:5
   |
LL |     #[crate_type = "0800"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:700:5
   |
LL |     #[crate_type = "0800"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:704:5
   |
LL |     #[crate_type = "0800"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:704:5
   |
LL |     #[crate_type = "0800"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:708:5
   |
LL |     #[crate_type = "0800"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:708:5
   |
LL |     #[crate_type = "0800"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:712:5
   |
LL |     #[crate_type = "0800"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:712:5
   |
LL |     #[crate_type = "0800"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:692:1
   |
LL | #[crate_type = "0800"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:692:1
   |
LL | #[crate_type = "0800"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:721:17
   |
LL |     mod inner { #![feature(x0600)] }
   |                 ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:721:17
   |
LL |     mod inner { #![feature(x0600)] }
   |                 ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:725:5
   |
LL |     #[feature(x0600)] fn f() { }
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:725:5
   |
LL |     #[feature(x0600)] fn f() { }
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:729:5
   |
LL |     #[feature(x0600)] struct S;
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:729:5
   |
LL |     #[feature(x0600)] struct S;
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:733:5
   |
LL |     #[feature(x0600)] type T = S;
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:733:5
   |
LL |     #[feature(x0600)] type T = S;
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:737:5
   |
LL |     #[feature(x0600)] impl S { }
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:737:5
   |
LL |     #[feature(x0600)] impl S { }
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:717:1
   |
LL | #[feature(x0600)]
   | ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:717:1
   |
LL | #[feature(x0600)]
   | ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:747:17
   |
LL |     mod inner { #![no_main="0400"] }
   |                 ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:747:17
   |
LL |     mod inner { #![no_main="0400"] }
   |                 ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:751:5
   |
LL |     #[no_main = "0400"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:751:5
   |
LL |     #[no_main = "0400"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:755:5
   |
LL |     #[no_main = "0400"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:755:5
   |
LL |     #[no_main = "0400"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:759:5
   |
LL |     #[no_main = "0400"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:759:5
   |
LL |     #[no_main = "0400"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:763:5
   |
LL |     #[no_main = "0400"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:763:5
   |
LL |     #[no_main = "0400"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:743:1
   |
LL | #[no_main = "0400"]
   | ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:743:1
   |
LL | #[no_main = "0400"]
   | ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:785:17
   |
LL |     mod inner { #![recursion_limit="0200"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:785:17
   |
LL |     mod inner { #![recursion_limit="0200"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:789:5
   |
LL |     #[recursion_limit="0200"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:789:5
   |
LL |     #[recursion_limit="0200"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:793:5
   |
LL |     #[recursion_limit="0200"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:793:5
   |
LL |     #[recursion_limit="0200"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:797:5
   |
LL |     #[recursion_limit="0200"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:797:5
   |
LL |     #[recursion_limit="0200"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:801:5
   |
LL |     #[recursion_limit="0200"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:801:5
   |
LL |     #[recursion_limit="0200"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:781:1
   |
LL | #[recursion_limit="0200"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:781:1
   |
LL | #[recursion_limit="0200"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:810:17
   |
LL |     mod inner { #![type_length_limit="0100"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:810:17
   |
LL |     mod inner { #![type_length_limit="0100"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:814:5
   |
LL |     #[type_length_limit="0100"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:814:5
   |
LL |     #[type_length_limit="0100"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:818:5
   |
LL |     #[type_length_limit="0100"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:818:5
   |
LL |     #[type_length_limit="0100"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:822:5
   |
LL |     #[type_length_limit="0100"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:822:5
   |
LL |     #[type_length_limit="0100"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:826:5
   |
LL |     #[type_length_limit="0100"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:826:5
   |
LL |     #[type_length_limit="0100"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:806:1
   |
LL | #[type_length_limit="0100"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:806:1
   |
LL | #[type_length_limit="0100"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:54:1
   |
LL | #![macro_export               = "4800"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:55:1
   |
LL | #![plugin_registrar           = "4700"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:58:1
   |
LL | #![main                      = "x4400"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:59:1
   |
LL | #![start                     = "x4300"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:62:1
   |
LL | #![repr                       = "3900"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:63:1
   |
LL | #![path                       = "3800"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:64:1
   |
LL | #![abi                        = "3700"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:65:1
   |
LL | #![automatically_derived      = "3600"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:67:1
   |
LL | #![no_link                    = "3400"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:69:1
   |
LL | #![should_panic               = "3200"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:70:1
   |
LL | #![ignore                     = "3100"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:76:1
   |
LL | #![proc_macro_derive          = "2500"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: compilation successful
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:837:1
   |
LL | / fn main() { //~ ERROR compilation successful
LL | |     println!("Hello World");
LL | | }
   | |_^

error: aborting due to previous error

