warning: macro_escape is a deprecated synonym for macro_use
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:513:1
   |
LL | #[macro_escape]
   | ^^^^^^^^^^^^^^^

warning: macro_escape is a deprecated synonym for macro_use
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:516:17
   |
LL |     mod inner { #![macro_escape] }
   |                 ^^^^^^^^^^^^^^^^
   |
   = help: consider an outer attribute, #[macro_use] mod ...

warning: `#[must_use]` on functions is experimental (see issue #43302)
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:663:5
   |
LL |     #[must_use = "1400"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(fn_must_use)] to the crate attributes to enable

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:49:33
   |
LL | #![warn                        (x5400)] //~ WARN unknown lint: `x5400`
   |                                 ^^^^^
   |
note: lint level defined here
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:44:28
   |
LL | #![warn(unused_attributes, unknown_lints)]
   |                            ^^^^^^^^^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:50:33
   |
LL | #![allow                       (x5300)] //~ WARN unknown lint: `x5300`
   |                                 ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:51:33
   |
LL | #![forbid                      (x5200)] //~ WARN unknown lint: `x5200`
   |                                 ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:52:33
   |
LL | #![deny                        (x5100)] //~ WARN unknown lint: `x5100`
   |                                 ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:113:8
   |
LL | #[warn(x5400)]
   |        ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:116:25
   |
LL |     mod inner { #![warn(x5400)] }
   |                         ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:119:12
   |
LL |     #[warn(x5400)] fn f() { }
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:122:12
   |
LL |     #[warn(x5400)] struct S;
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:125:12
   |
LL |     #[warn(x5400)] type T = S;
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:128:12
   |
LL |     #[warn(x5400)] impl S { }
   |            ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:132:9
   |
LL | #[allow(x5300)]
   |         ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:135:26
   |
LL |     mod inner { #![allow(x5300)] }
   |                          ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:138:13
   |
LL |     #[allow(x5300)] fn f() { }
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:141:13
   |
LL |     #[allow(x5300)] struct S;
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:144:13
   |
LL |     #[allow(x5300)] type T = S;
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:147:13
   |
LL |     #[allow(x5300)] impl S { }
   |             ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:151:10
   |
LL | #[forbid(x5200)]
   |          ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:154:27
   |
LL |     mod inner { #![forbid(x5200)] }
   |                           ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:157:14
   |
LL |     #[forbid(x5200)] fn f() { }
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:160:14
   |
LL |     #[forbid(x5200)] struct S;
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:163:14
   |
LL |     #[forbid(x5200)] type T = S;
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:166:14
   |
LL |     #[forbid(x5200)] impl S { }
   |              ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:170:8
   |
LL | #[deny(x5100)]
   |        ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:173:25
   |
LL |     mod inner { #![deny(x5100)] }
   |                         ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:176:12
   |
LL |     #[deny(x5100)] fn f() { }
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:179:12
   |
LL |     #[deny(x5100)] struct S;
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:182:12
   |
LL |     #[deny(x5100)] type T = S;
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:185:12
   |
LL |     #[deny(x5100)] impl S { }
   |            ^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:192:17
   |
LL |     mod inner { #![macro_reexport="5000"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:44:9
   |
LL | #![warn(unused_attributes, unknown_lints)]
   |         ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:195:5
   |
LL |     #[macro_reexport = "5000"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:198:5
   |
LL |     #[macro_reexport = "5000"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:201:5
   |
LL |     #[macro_reexport = "5000"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:204:5
   |
LL |     #[macro_reexport = "5000"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:189:1
   |
LL | #[macro_reexport = "5000"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:212:5
   |
LL |     #[macro_use] fn f() { }
   |     ^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:215:5
   |
LL |     #[macro_use] struct S;
   |     ^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:218:5
   |
LL |     #[macro_use] type T = S;
   |     ^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:221:5
   |
LL |     #[macro_use] impl S { }
   |     ^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:228:17
   |
LL |     mod inner { #![macro_export="4800"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:231:5
   |
LL |     #[macro_export = "4800"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:234:5
   |
LL |     #[macro_export = "4800"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:237:5
   |
LL |     #[macro_export = "4800"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:240:5
   |
LL |     #[macro_export = "4800"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:225:1
   |
LL | #[macro_export = "4800"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:247:17
   |
LL |     mod inner { #![plugin_registrar="4700"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:252:5
   |
LL |     #[plugin_registrar = "4700"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:255:5
   |
LL |     #[plugin_registrar = "4700"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:258:5
   |
LL |     #[plugin_registrar = "4700"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:244:1
   |
LL | #[plugin_registrar = "4700"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:265:17
   |
LL |     mod inner { #![main="4300"] }
   |                 ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:270:5
   |
LL |     #[main = "4400"] struct S;
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:273:5
   |
LL |     #[main = "4400"] type T = S;
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:276:5
   |
LL |     #[main = "4400"] impl S { }
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:262:1
   |
LL | #[main = "4400"]
   | ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:283:17
   |
LL |     mod inner { #![start="4300"] }
   |                 ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:288:5
   |
LL |     #[start = "4300"] struct S;
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:291:5
   |
LL |     #[start = "4300"] type T = S;
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:294:5
   |
LL |     #[start = "4300"] impl S { }
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:280:1
   |
LL | #[start = "4300"]
   | ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:333:17
   |
LL |     mod inner { #![repr="3900"] }
   |                 ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:336:5
   |
LL |     #[repr = "3900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:341:5
   |
LL |     #[repr = "3900"] type T = S;
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:344:5
   |
LL |     #[repr = "3900"] impl S { }
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:330:1
   |
LL | #[repr = "3900"]
   | ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:352:5
   |
LL |     #[path = "3800"] fn f() { }
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:355:5
   |
LL |     #[path = "3800"]  struct S;
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:358:5
   |
LL |     #[path = "3800"] type T = S;
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:361:5
   |
LL |     #[path = "3800"] impl S { }
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:368:17
   |
LL |     mod inner { #![abi="3700"] }
   |                 ^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:371:5
   |
LL |     #[abi = "3700"] fn f() { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:374:5
   |
LL |     #[abi = "3700"] struct S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:377:5
   |
LL |     #[abi = "3700"] type T = S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:380:5
   |
LL |     #[abi = "3700"] impl S { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:365:1
   |
LL | #[abi = "3700"]
   | ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:387:17
   |
LL |     mod inner { #![automatically_derived="3600"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:390:5
   |
LL |     #[automatically_derived = "3600"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:393:5
   |
LL |     #[automatically_derived = "3600"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:396:5
   |
LL |     #[automatically_derived = "3600"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:399:5
   |
LL |     #[automatically_derived = "3600"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:384:1
   |
LL | #[automatically_derived = "3600"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: function is marked #[no_mangle], but not exported
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:407:27
   |
LL |     #[no_mangle = "3500"] fn f() { }
   |                           -^^^^^^^^^
   |                           |
   |                           help: try making it public: `pub`
   |
   = note: #[warn(private_no_mangle_fns)] on by default

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:420:17
   |
LL |     mod inner { #![no_link="3400"] }
   |                 ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:423:5
   |
LL |     #[no_link = "3400"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:426:5
   |
LL |     #[no_link = "3400"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:429:5
   |
LL |     #[no_link = "3400"]type T = S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:432:5
   |
LL |     #[no_link = "3400"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:417:1
   |
LL | #[no_link = "3400"]
   | ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:439:17
   |
LL |     mod inner { #![should_panic="3200"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:442:5
   |
LL |     #[should_panic = "3200"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:445:5
   |
LL |     #[should_panic = "3200"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:448:5
   |
LL |     #[should_panic = "3200"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:451:5
   |
LL |     #[should_panic = "3200"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:436:1
   |
LL | #[should_panic = "3200"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:458:17
   |
LL |     mod inner { #![ignore="3100"] }
   |                 ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:461:5
   |
LL |     #[ignore = "3100"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:464:5
   |
LL |     #[ignore = "3100"] struct S;
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:467:5
   |
LL |     #[ignore = "3100"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:470:5
   |
LL |     #[ignore = "3100"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:455:1
   |
LL | #[ignore = "3100"]
   | ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:477:17
   |
LL |     mod inner { #![no_implicit_prelude="3000"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:480:5
   |
LL |     #[no_implicit_prelude = "3000"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:483:5
   |
LL |     #[no_implicit_prelude = "3000"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:486:5
   |
LL |     #[no_implicit_prelude = "3000"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:489:5
   |
LL |     #[no_implicit_prelude = "3000"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:474:1
   |
LL | #[no_implicit_prelude = "3000"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:496:17
   |
LL |     mod inner { #![reexport_test_harness_main="2900"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:499:5
   |
LL |     #[reexport_test_harness_main = "2900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:502:5
   |
LL |     #[reexport_test_harness_main = "2900"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:505:5
   |
LL |     #[reexport_test_harness_main = "2900"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:508:5
   |
LL |     #[reexport_test_harness_main = "2900"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:493:1
   |
LL | #[reexport_test_harness_main = "2900"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:519:5
   |
LL |     #[macro_escape] fn f() { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:522:5
   |
LL |     #[macro_escape] struct S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:525:5
   |
LL |     #[macro_escape] type T = S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:528:5
   |
LL |     #[macro_escape] impl S { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:536:17
   |
LL |     mod inner { #![no_std="2600"] }
   |                 ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:536:17
   |
LL |     mod inner { #![no_std="2600"] }
   |                 ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:540:5
   |
LL |     #[no_std = "2600"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:540:5
   |
LL |     #[no_std = "2600"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:544:5
   |
LL |     #[no_std = "2600"] struct S;
   |     ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:544:5
   |
LL |     #[no_std = "2600"] struct S;
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:548:5
   |
LL |     #[no_std = "2600"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:548:5
   |
LL |     #[no_std = "2600"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:552:5
   |
LL |     #[no_std = "2600"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:552:5
   |
LL |     #[no_std = "2600"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:532:1
   |
LL | #[no_std = "2600"]
   | ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:532:1
   |
LL | #[no_std = "2600"]
   | ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:692:17
   |
LL |     mod inner { #![crate_name="0900"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:692:17
   |
LL |     mod inner { #![crate_name="0900"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:696:5
   |
LL |     #[crate_name = "0900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:696:5
   |
LL |     #[crate_name = "0900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:700:5
   |
LL |     #[crate_name = "0900"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:700:5
   |
LL |     #[crate_name = "0900"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:704:5
   |
LL |     #[crate_name = "0900"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:704:5
   |
LL |     #[crate_name = "0900"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:708:5
   |
LL |     #[crate_name = "0900"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:708:5
   |
LL |     #[crate_name = "0900"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:688:1
   |
LL | #[crate_name = "0900"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:688:1
   |
LL | #[crate_name = "0900"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:717:17
   |
LL |     mod inner { #![crate_type="0800"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:717:17
   |
LL |     mod inner { #![crate_type="0800"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:721:5
   |
LL |     #[crate_type = "0800"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:721:5
   |
LL |     #[crate_type = "0800"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:725:5
   |
LL |     #[crate_type = "0800"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:725:5
   |
LL |     #[crate_type = "0800"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:729:5
   |
LL |     #[crate_type = "0800"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:729:5
   |
LL |     #[crate_type = "0800"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:733:5
   |
LL |     #[crate_type = "0800"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:733:5
   |
LL |     #[crate_type = "0800"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:713:1
   |
LL | #[crate_type = "0800"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:713:1
   |
LL | #[crate_type = "0800"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:742:17
   |
LL |     mod inner { #![feature(x0600)] }
   |                 ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:742:17
   |
LL |     mod inner { #![feature(x0600)] }
   |                 ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:746:5
   |
LL |     #[feature(x0600)] fn f() { }
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:746:5
   |
LL |     #[feature(x0600)] fn f() { }
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:750:5
   |
LL |     #[feature(x0600)] struct S;
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:750:5
   |
LL |     #[feature(x0600)] struct S;
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:754:5
   |
LL |     #[feature(x0600)] type T = S;
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:754:5
   |
LL |     #[feature(x0600)] type T = S;
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:758:5
   |
LL |     #[feature(x0600)] impl S { }
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:758:5
   |
LL |     #[feature(x0600)] impl S { }
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:738:1
   |
LL | #[feature(x0600)]
   | ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:738:1
   |
LL | #[feature(x0600)]
   | ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:768:17
   |
LL |     mod inner { #![no_main="0400"] }
   |                 ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:768:17
   |
LL |     mod inner { #![no_main="0400"] }
   |                 ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:772:5
   |
LL |     #[no_main = "0400"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:772:5
   |
LL |     #[no_main = "0400"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:776:5
   |
LL |     #[no_main = "0400"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:776:5
   |
LL |     #[no_main = "0400"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:780:5
   |
LL |     #[no_main = "0400"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:780:5
   |
LL |     #[no_main = "0400"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:784:5
   |
LL |     #[no_main = "0400"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:784:5
   |
LL |     #[no_main = "0400"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:764:1
   |
LL | #[no_main = "0400"]
   | ^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:764:1
   |
LL | #[no_main = "0400"]
   | ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:806:17
   |
LL |     mod inner { #![recursion_limit="0200"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:806:17
   |
LL |     mod inner { #![recursion_limit="0200"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:810:5
   |
LL |     #[recursion_limit="0200"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:810:5
   |
LL |     #[recursion_limit="0200"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:814:5
   |
LL |     #[recursion_limit="0200"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:814:5
   |
LL |     #[recursion_limit="0200"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:818:5
   |
LL |     #[recursion_limit="0200"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:818:5
   |
LL |     #[recursion_limit="0200"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:822:5
   |
LL |     #[recursion_limit="0200"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:822:5
   |
LL |     #[recursion_limit="0200"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:802:1
   |
LL | #[recursion_limit="0200"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:802:1
   |
LL | #[recursion_limit="0200"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:831:17
   |
LL |     mod inner { #![type_length_limit="0100"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:831:17
   |
LL |     mod inner { #![type_length_limit="0100"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:835:5
   |
LL |     #[type_length_limit="0100"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:835:5
   |
LL |     #[type_length_limit="0100"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:839:5
   |
LL |     #[type_length_limit="0100"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:839:5
   |
LL |     #[type_length_limit="0100"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:843:5
   |
LL |     #[type_length_limit="0100"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:843:5
   |
LL |     #[type_length_limit="0100"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:847:5
   |
LL |     #[type_length_limit="0100"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:847:5
   |
LL |     #[type_length_limit="0100"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:827:1
   |
LL | #[type_length_limit="0100"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:827:1
   |
LL | #[type_length_limit="0100"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:53:1
   |
LL | #![macro_reexport             = "5000"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:55:1
   |
LL | #![macro_export               = "4800"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:56:1
   |
LL | #![plugin_registrar           = "4700"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:59:1
   |
LL | #![main                      = "x4400"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:60:1
   |
LL | #![start                     = "x4300"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:63:1
   |
LL | #![repr                       = "3900"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:64:1
   |
LL | #![path                       = "3800"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:65:1
   |
LL | #![abi                        = "3700"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:66:1
   |
LL | #![automatically_derived      = "3600"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:68:1
   |
LL | #![no_link                    = "3400"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:70:1
   |
LL | #![should_panic               = "3200"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:71:1
   |
LL | #![ignore                     = "3100"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:77:1
   |
LL | #![proc_macro_derive          = "2500"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: compilation successful
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:858:1
   |
LL | / fn main() { //~ ERROR compilation successful
LL | |     println!("Hello World");
LL | | }
   | |_^

error: aborting due to previous error

