error[E0507]: cannot move out of data in a `&` reference
  --> $DIR/issue-40402-2.rs:15:10
   |
LL |     let (a, b) = x[0]; //~ ERROR cannot move out of indexed content
   |          ^
   |          |
   |          cannot move out of data in a `&` reference
   |          cannot move

error[E0507]: cannot move out of data in a `&` reference
  --> $DIR/issue-40402-2.rs:15:13
   |
LL |     let (a, b) = x[0]; //~ ERROR cannot move out of indexed content
   |             ^
   |             |
   |             cannot move out of data in a `&` reference
   |             cannot move

error[E0507]: cannot move out of borrowed content
  --> $DIR/issue-40402-2.rs:15:18
   |
LL |     let (a, b) = x[0]; //~ ERROR cannot move out of indexed content
   |          -  -    ^^^^
   |          |  |    |
   |          |  |    cannot move out of borrowed content
   |          |  |    help: consider borrowing here: `&x[0]`
   |          |  ...and here
   |          data moved here
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/issue-40402-2.rs:15:10
   |
LL |     let (a, b) = x[0]; //~ ERROR cannot move out of indexed content
   |          ^  ^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0507`.
