error[E0507]: cannot move out of borrowed content
  --> $DIR/issue-40402-1.rs:19:13
   |
LL |     let e = f.v[0]; //~ ERROR cannot move out of indexed content
   |             ^^^^^^
   |             |
   |             cannot move out of borrowed content
   |             help: consider borrowing here: `&f.v[0]`

error[E0507]: cannot move out of data in a `&` reference
  --> $DIR/issue-40402-1.rs:19:13
   |
LL |     let e = f.v[0]; //~ ERROR cannot move out of indexed content
   |             ^^^^^^
   |             |
   |             cannot move out of data in a `&` reference
   |             cannot move

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0507`.
