warning: expected `;`, found `let`
  --> $DIR/missing-semicolon-warning.rs:16:12
   |
LL |         $( let x = $e1 )*; //~ WARN expected `;`
   |            ^^^
...
LL | fn main() { m!(0, 0; 0, 0); } //~ ERROR compilation successful
   |             --------------- in this macro invocation
   |
   = note: This was erroneously allowed and will become a hard error in a future release

warning: expected `;`, found `println`
  --> $DIR/missing-semicolon-warning.rs:17:12
   |
LL |         $( println!("{}", $e2) )*; //~ WARN expected `;`
   |            ^^^^^^^
...
LL | fn main() { m!(0, 0; 0, 0); } //~ ERROR compilation successful
   |             --------------- in this macro invocation
   |
   = note: This was erroneously allowed and will become a hard error in a future release

error: compilation successful
  --> $DIR/missing-semicolon-warning.rs:22:1
   |
LL | fn main() { m!(0, 0; 0, 0); } //~ ERROR compilation successful
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

