error: symbol-name(_ZN5basic4main17h08bcaf310214ed52E)
  --> $DIR/basic.rs:13:1
   |
LL | #[rustc_symbol_name] //~ ERROR _ZN5basic4main
   | ^^^^^^^^^^^^^^^^^^^^

error: item-path(main)
  --> $DIR/basic.rs:14:1
   |
LL | #[rustc_item_path] //~ ERROR item-path(main)
   | ^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

