error: symbol-name(_ZN5impl13foo3Foo3bar17h8da62e6147ff602fE)
  --> $DIR/impl1.rs:18:9
   |
LL |         #[rustc_symbol_name] //~ ERROR _ZN5impl13foo3Foo3bar
   |         ^^^^^^^^^^^^^^^^^^^^

error: item-path(foo::Foo::bar)
  --> $DIR/impl1.rs:19:9
   |
LL |         #[rustc_item_path] //~ ERROR item-path(foo::Foo::bar)
   |         ^^^^^^^^^^^^^^^^^^

error: symbol-name(_ZN5impl13bar33_$LT$impl$u20$impl1..foo..Foo$GT$3baz17h374cb8f6185db9b4E)
  --> $DIR/impl1.rs:28:9
   |
LL |         #[rustc_symbol_name] //~ ERROR _ZN5impl13bar33_$LT$impl$u20$impl1..foo..Foo$GT$3baz
   |         ^^^^^^^^^^^^^^^^^^^^

error: item-path(bar::<impl foo::Foo>::baz)
  --> $DIR/impl1.rs:29:9
   |
LL |         #[rustc_item_path] //~ ERROR item-path(bar::<impl foo::Foo>::baz)
   |         ^^^^^^^^^^^^^^^^^^

error: aborting due to 4 previous errors

