warning: not reporting region error due to nll
  --> $DIR/variance-contravariant-arg-object.rs:24:5
   |
LL |     v //~ ERROR mismatched types
   |     ^

warning: not reporting region error due to nll
  --> $DIR/variance-contravariant-arg-object.rs:32:5
   |
LL |     v //~ ERROR mismatched types
   |     ^

error: unsatisfied lifetime constraints
  --> $DIR/variance-contravariant-arg-object.rs:24:5
   |
LL | fn get_min_from_max<'min, 'max>(v: Box<Get<&'max i32>>)
   |                     ----  ---- lifetime `'max` defined here
   |                     |
   |                     lifetime `'min` defined here
...
LL |     v //~ ERROR mismatched types
   |     ^ cast requires that `'min` must outlive `'max`

error: unsatisfied lifetime constraints
  --> $DIR/variance-contravariant-arg-object.rs:32:5
   |
LL | fn get_max_from_min<'min, 'max, G>(v: Box<Get<&'min i32>>)
   |                     ----  ---- lifetime `'max` defined here
   |                     |
   |                     lifetime `'min` defined here
...
LL |     v //~ ERROR mismatched types
   |     ^ cast requires that `'min` must outlive `'max`

error: aborting due to 2 previous errors

