error: unsatisfied lifetime constraints
  --> $DIR/variance-use-invariant-struct-1.rs:22:5
   |
LL | fn foo<'min,'max>(v: SomeStruct<&'max ()>)
   |        ---- ---- lifetime `'max` defined here
   |        |
   |        lifetime `'min` defined here
...
LL |     v //~ ERROR mismatched types
   |     ^ returning this value requires that `'min` must outlive `'max`

error: unsatisfied lifetime constraints
  --> $DIR/variance-use-invariant-struct-1.rs:29:5
   |
LL | fn bar<'min,'max>(v: SomeStruct<&'min ()>)
   |        ---- ---- lifetime `'max` defined here
   |        |
   |        lifetime `'min` defined here
...
LL |     v //~ ERROR mismatched types
   |     ^ returning this value requires that `'min` must outlive `'max`

error: aborting due to 2 previous errors

