error[E0308]: mismatched types
  --> $DIR/variance-covariant-self-trait-match.rs:24:5
   |
LL |     impls_get::<&'min G>(); //~ ERROR mismatched types
   |     ^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
   |
   = note: expected type `Get`
              found type `Get`
note: the lifetime 'min as defined on the function body at 20:21...
  --> $DIR/variance-covariant-self-trait-match.rs:20:21
   |
LL | fn get_min_from_max<'min, 'max, G>()
   |                     ^^^^
note: ...does not necessarily outlive the lifetime 'max as defined on the function body at 20:27
  --> $DIR/variance-covariant-self-trait-match.rs:20:27
   |
LL | fn get_min_from_max<'min, 'max, G>()
   |                           ^^^^

error[E0308]: mismatched types
  --> $DIR/variance-covariant-self-trait-match.rs:30:5
   |
LL |     impls_get::<&'max G>(); //~ ERROR mismatched types
   |     ^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
   |
   = note: expected type `Get`
              found type `Get`
note: the lifetime 'min as defined on the function body at 27:21...
  --> $DIR/variance-covariant-self-trait-match.rs:27:21
   |
LL | fn get_max_from_min<'min, 'max, G>()
   |                     ^^^^
note: ...does not necessarily outlive the lifetime 'max as defined on the function body at 27:27
  --> $DIR/variance-covariant-self-trait-match.rs:27:27
   |
LL | fn get_max_from_min<'min, 'max, G>()
   |                           ^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
