error[E0308]: mismatched types
  --> $DIR/variance-invariant-self-trait-match.rs:22:5
   |
LL |     impls_get::<&'min G>(); //~ ERROR mismatched types
   |     ^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
   |
   = note: expected type `Get`
              found type `Get`
note: the lifetime 'min as defined on the function body at 19:21...
  --> $DIR/variance-invariant-self-trait-match.rs:19:21
   |
LL | fn get_min_from_max<'min, 'max, G>()
   |                     ^^^^
note: ...does not necessarily outlive the lifetime 'max as defined on the function body at 19:27
  --> $DIR/variance-invariant-self-trait-match.rs:19:27
   |
LL | fn get_min_from_max<'min, 'max, G>()
   |                           ^^^^

error[E0308]: mismatched types
  --> $DIR/variance-invariant-self-trait-match.rs:28:5
   |
LL |     impls_get::<&'max G>(); //~ ERROR mismatched types
   |     ^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
   |
   = note: expected type `Get`
              found type `Get`
note: the lifetime 'min as defined on the function body at 25:21...
  --> $DIR/variance-invariant-self-trait-match.rs:25:21
   |
LL | fn get_max_from_min<'min, 'max, G>()
   |                     ^^^^
note: ...does not necessarily outlive the lifetime 'max as defined on the function body at 25:27
  --> $DIR/variance-invariant-self-trait-match.rs:25:27
   |
LL | fn get_max_from_min<'min, 'max, G>()
   |                           ^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
