error: unsatisfied lifetime constraints
  --> $DIR/variance-invariant-arg-trait-match.rs:20:5
   |
LL | fn get_min_from_max<'min, 'max, G>()
   |                     ----  ---- lifetime `'max` defined here
   |                     |
   |                     lifetime `'min` defined here
...
LL |     impls_get::<G,&'min i32>() //~ ERROR mismatched types
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'min` must outlive `'max`

error: unsatisfied lifetime constraints
  --> $DIR/variance-invariant-arg-trait-match.rs:26:5
   |
LL | fn get_max_from_min<'min, 'max, G>()
   |                     ----  ---- lifetime `'max` defined here
   |                     |
   |                     lifetime `'min` defined here
...
LL |     impls_get::<G,&'max i32>() //~ ERROR mismatched types
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'min` must outlive `'max`

error: aborting due to 2 previous errors

