error: unsatisfied lifetime constraints
  --> $DIR/variance-contravariant-self-trait-match.rs:23:5
   |
LL | fn get_min_from_max<'min, 'max, G>()
   |                     ----  ---- lifetime `'max` defined here
   |                     |
   |                     lifetime `'min` defined here
...
LL |     impls_get::<&'min G>(); //~ ERROR mismatched types
   |     ^^^^^^^^^^^^^^^^^^^^^^ requires that `'min` must outlive `'max`

error: unsatisfied lifetime constraints
  --> $DIR/variance-contravariant-self-trait-match.rs:32:5
   |
LL | fn get_max_from_min<'min, 'max, G>()
   |                     ----  ---- lifetime `'max` defined here
   |                     |
   |                     lifetime `'min` defined here
...
LL |     impls_get::<&'max G>(); //~ ERROR mismatched types
   |     ^^^^^^^^^^^^^^^^^^^^^^ requires that `'min` must outlive `'max`

error: aborting due to 2 previous errors

