warning: not reporting region error due to nll
  --> $DIR/variance-trait-matching.rs:34:10
   |
LL |     pick(get, &22) //~ ERROR 34:5: 34:9: explicit lifetime required in the type of `get` [E0621]
   |          ^^^

warning: not reporting region error due to nll
  --> $DIR/variance-trait-matching.rs:34:5
   |
LL |     pick(get, &22) //~ ERROR 34:5: 34:9: explicit lifetime required in the type of `get` [E0621]
   |     ^^^^^^^^^^^^^^

warning: not reporting region error due to nll
  --> $DIR/variance-trait-matching.rs:34:5
   |
LL |     pick(get, &22) //~ ERROR 34:5: 34:9: explicit lifetime required in the type of `get` [E0621]
   |     ^^^^

error[E0621]: explicit lifetime required in the type of `get`
  --> $DIR/variance-trait-matching.rs:34:5
   |
LL | fn get<'a, G>(get: &G) -> i32
   |                    -- help: add explicit lifetime `'a` to the type of `get`: `&'a G`
...
LL |     pick(get, &22) //~ ERROR 34:5: 34:9: explicit lifetime required in the type of `get` [E0621]
   |     ^^^^^^^^^^^^^^ lifetime `'a` required

error: aborting due to previous error

For more information about this error, try `rustc --explain E0621`.
