error[E0368]: binary assignment operation `+=` cannot be applied to type `std::collections::LinkedList<_>`
  --> $DIR/E0067.rs:14:5
   |
LL |     LinkedList::new() += 1; //~ ERROR E0368
   |     -----------------^^^^^
   |     |
   |     cannot use `+=` on type `std::collections::LinkedList<_>`

error[E0067]: invalid left-hand side expression
  --> $DIR/E0067.rs:14:5
   |
LL |     LinkedList::new() += 1; //~ ERROR E0368
   |     ^^^^^^^^^^^^^^^^^ invalid expression for left-hand side

error: aborting due to 2 previous errors

You've got a few errors: E0067, E0368
If you want more information on an error, try using "rustc --explain E0067"
