error[E0070]: invalid left-hand side expression
  --> $DIR/E0070.rs:16:5
   |
16 |     SOME_CONST = 14; //~ ERROR E0070
   |     ^^^^^^^^^^^^^^^ left-hand of expression not valid

error[E0070]: invalid left-hand side expression
  --> $DIR/E0070.rs:17:5
   |
17 |     1 = 3; //~ ERROR E0070
   |     ^^^^^ left-hand of expression not valid

error[E0308]: mismatched types
  --> $DIR/E0070.rs:18:25
   |
18 |     some_other_func() = 4; //~ ERROR E0070
   |                         ^ expected (), found integral variable
   |
   = note: expected type `()`
              found type `{integer}`

error[E0070]: invalid left-hand side expression
  --> $DIR/E0070.rs:18:5
   |
18 |     some_other_func() = 4; //~ ERROR E0070
   |     ^^^^^^^^^^^^^^^^^^^^^ left-hand of expression not valid

error: aborting due to 4 previous errors

