error[E0088]: wrong number of lifetime arguments: expected 0, found 1
  --> $DIR/E0088.rs:15:9
   |
LL |     f::<'static>(); //~ ERROR E0088
   |         ^^^^^^^ unexpected lifetime argument

error[E0088]: wrong number of lifetime arguments: expected 1, found 2
  --> $DIR/E0088.rs:16:18
   |
LL |     g::<'static, 'static>(); //~ ERROR E0088
   |                  ^^^^^^^ unexpected lifetime argument

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0088`.
