error[E0161]: cannot move a value of type str: the size of str cannot be statically determined
  --> $DIR/E0161.rs:14:28
   |
14 |     let _x: Box<str> = box *"hello"; //~ ERROR E0161
   |                            ^^^^^^^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/E0161.rs:14:28
   |
14 |     let _x: Box<str> = box *"hello"; //~ ERROR E0161
   |                            ^^^^^^^^ cannot move out of borrowed content

error: aborting due to 2 previous errors

You've got a few errors: E0161, E0507
If you want more information on an error, try using "rustc --explain E0161"
