error[E0220]: associated type `F` not found for `Trait`
  --> $DIR/E0220.rs:15:18
   |
15 | type Foo = Trait<F=i32>; //~ ERROR E0220
   |                  ^^^^^ associated type `F` not found

error[E0191]: the value of the associated type `Bar` (from the trait `Trait`) must be specified
  --> $DIR/E0220.rs:15:12
   |
15 | type Foo = Trait<F=i32>; //~ ERROR E0220
   |            ^^^^^^^^^^^^ missing associated type `Bar` value

error: aborting due to 2 previous errors

You've got a few errors: E0191, E0220
If you want more information on an error, try using "rustc --explain E0191"
