error[E0252]: the name `baz` is defined multiple times
  --> $DIR/E0252.rs:12:5
   |
11 | use foo::baz;
   |     -------- previous import of the type `baz` here
12 | use bar::baz; //~ ERROR E0252
   |     ^^^^^^^^ `baz` reimported here
   |
   = note: `baz` must be defined only once in the type namespace of this module
help: You can use `as` to change the binding name of the import
   |
12 | use bar::baz as other_baz; //~ ERROR E0252
   |     ^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

If you want more information on this error, try using "rustc --explain E0252"
