error[E0017]: references in constants may only refer to immutable values
  --> $DIR/E0388.rs:14:30
   |
14 | const CR: &'static mut i32 = &mut C; //~ ERROR E0017
   |                              ^^^^^^ constants require immutable values

error[E0017]: references in statics may only refer to immutable values
  --> $DIR/E0388.rs:15:39
   |
15 | static STATIC_REF: &'static mut i32 = &mut X; //~ ERROR E0017
   |                                       ^^^^^^ statics require immutable values

error[E0596]: cannot borrow immutable static item as mutable
  --> $DIR/E0388.rs:15:44
   |
15 | static STATIC_REF: &'static mut i32 = &mut X; //~ ERROR E0017
   |                                            ^

error[E0017]: references in statics may only refer to immutable values
  --> $DIR/E0388.rs:17:38
   |
17 | static CONST_REF: &'static mut i32 = &mut C; //~ ERROR E0017
   |                                      ^^^^^^ statics require immutable values

error: aborting due to 4 previous errors

