error[E0449]: unnecessary visibility qualifier
  --> $DIR/E0449.rs:17:1
   |
17 | pub impl Bar {} //~ ERROR E0449
   | ^^^^^^^^^^^^^^^ `pub` not needed here
   |
   = note: place qualifiers on individual impl items instead

error[E0449]: unnecessary visibility qualifier
  --> $DIR/E0449.rs:19:1
   |
19 | / pub impl Foo for Bar { //~ ERROR E0449
20 | |     pub fn foo() {} //~ ERROR E0449
21 | | }
   | |_^ `pub` not needed here

error[E0449]: unnecessary visibility qualifier
  --> $DIR/E0449.rs:20:5
   |
20 |     pub fn foo() {} //~ ERROR E0449
   |     ^^^^^^^^^^^^^^^ `pub` not needed here

error: aborting due to 3 previous errors

