error[E0507]: cannot move out of borrowed content
  --> $DIR/E0507.rs:22:5
   |
LL |     x.borrow().nothing_is_true(); //~ ERROR E0507
   |     ^^^^^^^^^^ cannot move out of borrowed content

error[E0507]: cannot move out of data in a `&` reference
  --> $DIR/E0507.rs:22:5
   |
LL |     x.borrow().nothing_is_true(); //~ ERROR E0507
   |     ^^^^^^^^^^
   |     |
   |     cannot move out of data in a `&` reference
   |     cannot move

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0507`.
