error[E0605]: non-primitive cast: `u8` as `std::vec::Vec<u8>`
  --> $DIR/E0605.rs:13:5
   |
13 |     x as Vec<u8>; //~ ERROR E0605
   |     ^^^^^^^^^^^^
   |
   = note: an `as` expression can only be used to convert between primitive types. Consider using the `From` trait

error[E0605]: non-primitive cast: `*const u8` as `&u8`
  --> $DIR/E0605.rs:16:5
   |
16 |     v as &u8; //~ ERROR E0605
   |     ^^^^^^^^
   |
   = note: an `as` expression can only be used to convert between primitive types. Consider using the `From` trait

error: aborting due to 2 previous errors

