warning: not reporting region error due to nll
  --> $DIR/E0621-does-not-trigger-for-closures.rs:25:5
   |
LL |     invoke(&x, |a, b| if a > b { a } else { b }); //~ ERROR E0495
   |     ^^^^^^

error: unsatisfied lifetime constraints
  --> $DIR/E0621-does-not-trigger-for-closures.rs:25:26
   |
LL |     invoke(&x, |a, b| if a > b { a } else { b }); //~ ERROR E0495
   |                ----------^^^^^-----------------
   |                |   |     |
   |                |   |     free region requires that `'1` must outlive `'2`
   |                |   lifetime `'1` appears in this argument
   |                lifetime `'2` appears in return type

error: aborting due to previous error

