error[E0657]: `impl Trait` can only capture lifetimes bound at the fn or impl level
  --> $DIR/E0657.rs:19:31
   |
LL |     -> Box<for<'a> Id<impl Lt<'a>>>
   |                               ^^

error[E0657]: `impl Trait` can only capture lifetimes bound at the fn or impl level
  --> $DIR/E0657.rs:28:35
   |
LL |         -> Box<for<'a> Id<impl Lt<'a>>>
   |                                   ^^

error[E0308]: mismatched types
  --> $DIR/E0657.rs:22:5
   |
LL |     () //~ ERROR mismatched types
   |     ^^ expected struct `std::boxed::Box`, found ()
   |
   = note: expected type `std::boxed::Box<Id<_> + 'static>`
              found type `()`

error[E0308]: mismatched types
  --> $DIR/E0657.rs:31:9
   |
LL |         () //~ ERROR mismatched types
   |         ^^ expected struct `std::boxed::Box`, found ()
   |
   = note: expected type `std::boxed::Box<Id<_> + 'static>`
              found type `()`

error: aborting due to 4 previous errors

Some errors occurred: E0308, E0657.
For more information about an error, try `rustc --explain E0308`.
