error: `c` does not live long enough
  --> $DIR/dropck-eyepatch-extern-crate.rs:39:20
   |
39 |     dt = Dt("dt", &c); //~ ERROR `c` does not live long enough
   |                    ^ does not live long enough
...
55 | }
   | - borrowed value dropped before borrower
   |
   = note: values in a scope are dropped in the opposite order they are created

error: `c` does not live long enough
  --> $DIR/dropck-eyepatch-extern-crate.rs:40:20
   |
40 |     dr = Dr("dr", &c); //~ ERROR `c` does not live long enough
   |                    ^ does not live long enough
...
55 | }
   | - borrowed value dropped before borrower
   |
   = note: values in a scope are dropped in the opposite order they are created

error: `c` does not live long enough
  --> $DIR/dropck-eyepatch-extern-crate.rs:47:29
   |
47 |     pt = Pt("pt", &c_long, &c); //~ ERROR `c` does not live long enough
   |                             ^ does not live long enough
...
55 | }
   | - borrowed value dropped before borrower
   |
   = note: values in a scope are dropped in the opposite order they are created

error: `c` does not live long enough
  --> $DIR/dropck-eyepatch-extern-crate.rs:48:29
   |
48 |     pr = Pr("pr", &c_long, &c); //~ ERROR `c` does not live long enough
   |                             ^ does not live long enough
...
55 | }
   | - borrowed value dropped before borrower
   |
   = note: values in a scope are dropped in the opposite order they are created

error: aborting due to 4 previous errors

