error[E0569]: requires an `unsafe impl` declaration due to `#[may_dangle]` attribute
  --> $DIR/dropck-eyepatch-implies-unsafe-impl.rs:32:1
   |
LL | / impl<#[may_dangle] A, B: fmt::Debug> Drop for Pt<A, B> {
LL | |     //~^ ERROR requires an `unsafe impl` declaration due to `#[may_dangle]` attribute
LL | |
LL | |     // (unsafe to access self.1  due to #[may_dangle] on A)
LL | |     fn drop(&mut self) { println!("drop {} {:?}", self.0, self.2); }
LL | | }
   | |_^

error[E0569]: requires an `unsafe impl` declaration due to `#[may_dangle]` attribute
  --> $DIR/dropck-eyepatch-implies-unsafe-impl.rs:38:1
   |
LL | / impl<#[may_dangle] 'a, 'b, B: fmt::Debug> Drop for Pr<'a, 'b, B> {
LL | |     //~^ ERROR requires an `unsafe impl` declaration due to `#[may_dangle]` attribute
LL | |
LL | |     // (unsafe to access self.1 due to #[may_dangle] on 'a)
LL | |     fn drop(&mut self) { println!("drop {} {:?}", self.0, self.2); }
LL | | }
   | |_^

error: aborting due to 2 previous errors

If you want more information on this error, try using "rustc --explain E0569"
