error: `c` does not live long enough
  --> $DIR/dropck-eyepatch-reorder.rs:57:20
   |
57 |     dt = Dt("dt", &c); //~ ERROR `c` does not live long enough
   |                    ^ does not live long enough
...
73 | }
   | - borrowed value dropped before borrower
   |
   = note: values in a scope are dropped in the opposite order they are created

error: `c` does not live long enough
  --> $DIR/dropck-eyepatch-reorder.rs:58:20
   |
58 |     dr = Dr("dr", &c); //~ ERROR `c` does not live long enough
   |                    ^ does not live long enough
...
73 | }
   | - borrowed value dropped before borrower
   |
   = note: values in a scope are dropped in the opposite order they are created

error: `c` does not live long enough
  --> $DIR/dropck-eyepatch-reorder.rs:65:29
   |
65 |     pt = Pt("pt", &c_long, &c); //~ ERROR `c` does not live long enough
   |                             ^ does not live long enough
...
73 | }
   | - borrowed value dropped before borrower
   |
   = note: values in a scope are dropped in the opposite order they are created

error: `c` does not live long enough
  --> $DIR/dropck-eyepatch-reorder.rs:66:29
   |
66 |     pr = Pr("pr", &c_long, &c); //~ ERROR `c` does not live long enough
   |                             ^ does not live long enough
...
73 | }
   | - borrowed value dropped before borrower
   |
   = note: values in a scope are dropped in the opposite order they are created

error: aborting due to 4 previous errors

