error[E0597]: `o2` does not live long enough
  --> $DIR/dropck_trait_cycle_checked.rs:121:13
   |
LL |     let (o1, o2, o3): (Box<Obj>, Box<Obj>, Box<Obj>) = (O::new(), O::new(), O::new());
   |                                                                   -------- cast requires that `o2` is borrowed for `'static`
LL |     o1.set0(&o2); //~ ERROR `o2` does not live long enough
   |             ^^^ borrowed value does not live long enough
...
LL | }
   | - `o2` dropped here while still borrowed

error[E0597]: `o3` does not live long enough
  --> $DIR/dropck_trait_cycle_checked.rs:122:13
   |
LL |     let (o1, o2, o3): (Box<Obj>, Box<Obj>, Box<Obj>) = (O::new(), O::new(), O::new());
   |                                                                             -------- cast requires that `o3` is borrowed for `'static`
LL |     o1.set0(&o2); //~ ERROR `o2` does not live long enough
LL |     o1.set1(&o3); //~ ERROR `o3` does not live long enough
   |             ^^^ borrowed value does not live long enough
...
LL | }
   | - `o3` dropped here while still borrowed

error[E0597]: `o2` does not live long enough
  --> $DIR/dropck_trait_cycle_checked.rs:123:13
   |
LL |     let (o1, o2, o3): (Box<Obj>, Box<Obj>, Box<Obj>) = (O::new(), O::new(), O::new());
   |                                                                   -------- cast requires that `o2` is borrowed for `'static`
...
LL |     o2.set0(&o2); //~ ERROR `o2` does not live long enough
   |             ^^^ borrowed value does not live long enough
...
LL | }
   | - `o2` dropped here while still borrowed

error[E0597]: `o3` does not live long enough
  --> $DIR/dropck_trait_cycle_checked.rs:124:13
   |
LL |     let (o1, o2, o3): (Box<Obj>, Box<Obj>, Box<Obj>) = (O::new(), O::new(), O::new());
   |                                                                             -------- cast requires that `o3` is borrowed for `'static`
...
LL |     o2.set1(&o3); //~ ERROR `o3` does not live long enough
   |             ^^^ borrowed value does not live long enough
...
LL | }
   | - `o3` dropped here while still borrowed

error[E0597]: `o1` does not live long enough
  --> $DIR/dropck_trait_cycle_checked.rs:125:13
   |
LL |     let (o1, o2, o3): (Box<Obj>, Box<Obj>, Box<Obj>) = (O::new(), O::new(), O::new());
   |                                                         -------- cast requires that `o1` is borrowed for `'static`
...
LL |     o3.set0(&o1); //~ ERROR `o1` does not live long enough
   |             ^^^ borrowed value does not live long enough
LL |     o3.set1(&o2); //~ ERROR `o2` does not live long enough
LL | }
   | - `o1` dropped here while still borrowed

error[E0597]: `o2` does not live long enough
  --> $DIR/dropck_trait_cycle_checked.rs:126:13
   |
LL |     let (o1, o2, o3): (Box<Obj>, Box<Obj>, Box<Obj>) = (O::new(), O::new(), O::new());
   |                                                                   -------- cast requires that `o2` is borrowed for `'static`
...
LL |     o3.set1(&o2); //~ ERROR `o2` does not live long enough
   |             ^^^ borrowed value does not live long enough
LL | }
   | - `o2` dropped here while still borrowed

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0597`.
