error[E0597]: `c` does not live long enough
  --> $DIR/dropck-eyepatch-extern-crate.rs:55:1
   |
39 |     dt = Dt("dt", &c); //~ ERROR `c` does not live long enough
   |                    - borrow occurs here
...
55 | }
   | ^ `c` dropped here while still borrowed
   |
   = note: values in a scope are dropped in the opposite order they are created

error[E0597]: `c` does not live long enough
  --> $DIR/dropck-eyepatch-extern-crate.rs:55:1
   |
40 |     dr = Dr("dr", &c); //~ ERROR `c` does not live long enough
   |                    - borrow occurs here
...
55 | }
   | ^ `c` dropped here while still borrowed
   |
   = note: values in a scope are dropped in the opposite order they are created

error[E0597]: `c` does not live long enough
  --> $DIR/dropck-eyepatch-extern-crate.rs:55:1
   |
47 |     pt = Pt("pt", &c_long, &c); //~ ERROR `c` does not live long enough
   |                             - borrow occurs here
...
55 | }
   | ^ `c` dropped here while still borrowed
   |
   = note: values in a scope are dropped in the opposite order they are created

error[E0597]: `c` does not live long enough
  --> $DIR/dropck-eyepatch-extern-crate.rs:55:1
   |
48 |     pr = Pr("pr", &c_long, &c); //~ ERROR `c` does not live long enough
   |                             - borrow occurs here
...
55 | }
   | ^ `c` dropped here while still borrowed
   |
   = note: values in a scope are dropped in the opposite order they are created

error: aborting due to 4 previous errors

