error: rustc_peek: bit not set
  --> $DIR/def-inits-1.rs:25:14
   |
LL |     unsafe { rustc_peek(&ret); }  //~ ERROR rustc_peek: bit not set
   |              ^^^^^^^^^^^^^^^^

error: rustc_peek: bit not set
  --> $DIR/def-inits-1.rs:41:14
   |
LL |     unsafe { rustc_peek(&z); } //~ ERROR rustc_peek: bit not set
   |              ^^^^^^^^^^^^^^

error: rustc_peek: bit not set
  --> $DIR/def-inits-1.rs:44:14
   |
LL |     unsafe { rustc_peek(&y); } //~ ERROR rustc_peek: bit not set
   |              ^^^^^^^^^^^^^^

error: rustc_peek: bit not set
  --> $DIR/def-inits-1.rs:52:14
   |
LL |     unsafe { rustc_peek(&x); } //~ ERROR rustc_peek: bit not set
   |              ^^^^^^^^^^^^^^

error: stop_after_dataflow ended compilation

error: aborting due to 5 previous errors

