error: rustc_peek: bit not set
  --> $DIR/inits-1.rs:25:14
   |
LL |     unsafe { rustc_peek(&ret); }  //~ ERROR rustc_peek: bit not set
   |              ^^^^^^^^^^^^^^^^

error: rustc_peek: bit not set
  --> $DIR/inits-1.rs:45:14
   |
LL |     unsafe { rustc_peek(&y); }  //~ ERROR rustc_peek: bit not set
   |              ^^^^^^^^^^^^^^

error: rustc_peek: bit not set
  --> $DIR/inits-1.rs:53:14
   |
LL |     unsafe { rustc_peek(&x); } //~ ERROR rustc_peek: bit not set
   |              ^^^^^^^^^^^^^^

error: stop_after_dataflow ended compilation

error: aborting due to 4 previous errors

