error: expected identifier, found `<-`
  --> $DIR/const-expression-parameter.rs:9:19
   |
LL |     i32_identity::<-1>(); //~ ERROR expected identifier, found `<-`
   |                   ^^ expected identifier

error: expected one of `,` or `>`, found `+`
  --> $DIR/const-expression-parameter.rs:13:22
   |
LL |     i32_identity::<1 + 2>(); //~ ERROR expected one of `,` or `>`, found `+`
   |                      ^ expected one of `,` or `>` here

warning: the feature `const_generics` is incomplete and may cause the compiler to crash
  --> $DIR/const-expression-parameter.rs:1:12
   |
LL | #![feature(const_generics)]
   |            ^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

