error: incorrect close delimiter: `}`
  --> $DIR/issue-10636-2.rs:18:1
   |
18 | } //~ ERROR: incorrect close delimiter
   | ^
   |
note: unclosed delimiter
  --> $DIR/issue-10636-2.rs:15:15
   |
15 |     option.map(|some| 42;
   |               ^

error: expected one of `,`, `.`, `?`, or an operator, found `;`
  --> $DIR/issue-10636-2.rs:15:25
   |
15 |     option.map(|some| 42;
   |                         ^ expected one of `,`, `.`, `?`, or an operator here

error: expected expression, found `)`
  --> $DIR/issue-10636-2.rs:18:1
   |
18 | } //~ ERROR: incorrect close delimiter
   | ^

error[E0601]: main function not found

error: aborting due to 4 previous errors

