error: expected one of `!`, `.`, `::`, `;`, `?`, `{`, `}`, or an operator, found `error`
  --> $DIR/main.rs:19:20
   |
19 |     () => { syntax error }; //~ ERROR expected one of
   |                   -^^^^^ unexpected token
   |                   |
   |                   expected one of 8 possible tokens here
$DIR/main.rs:24:5: 24:13 note: in this expansion of pong! (defined in $DIR/main.rs)

error: expected one of `!`, `.`, `::`, `;`, `?`, `{`, `}`, or an operator, found `error`
  --> $DIR/main.rs:19:20
   |
19 |     () => { syntax error }; //~ ERROR expected one of
   |                   -^^^^^ unexpected token
   |                   |
   |                   expected one of 8 possible tokens here
$DIR/main.rs:25:5: 25:13 note: in this expansion of ping! (defined in <ping macros>)
<ping macros>:1:11: 1:24 note: in this expansion of pong! (defined in $DIR/main.rs)

error: aborting due to 2 previous errors

