error[E0088]: wrong number of lifetime arguments: expected 1, found 2
  --> $DIR/generic-arg-mismatch-recover.rs:16:20
   |
LL |     Foo::<'static, 'static, ()>(&0); //~ ERROR wrong number of lifetime arguments
   |                    ^^^^^^^ unexpected lifetime argument

error[E0308]: mismatched types
  --> $DIR/generic-arg-mismatch-recover.rs:16:33
   |
LL |     Foo::<'static, 'static, ()>(&0); //~ ERROR wrong number of lifetime arguments
   |                                 ^^ expected (), found integral variable
   |
   = note: expected type `&'static ()`
              found type `&{integer}`

error[E0088]: wrong number of lifetime arguments: expected 1, found 2
  --> $DIR/generic-arg-mismatch-recover.rs:19:20
   |
LL |     Bar::<'static, 'static, ()>(&()); //~ ERROR wrong number of lifetime arguments
   |                    ^^^^^^^ unexpected lifetime argument

error[E0087]: wrong number of type arguments: expected 0, found 1
  --> $DIR/generic-arg-mismatch-recover.rs:19:29
   |
LL |     Bar::<'static, 'static, ()>(&()); //~ ERROR wrong number of lifetime arguments
   |                             ^^ unexpected type argument

error: aborting due to 4 previous errors

Some errors occurred: E0087, E0088, E0308.
For more information about an error, try `rustc --explain E0087`.
