error: value assigned to `x` is never read
  --> $DIR/liveness-dead.rs:19:13
   |
LL |     let mut x: isize = 3; //~ ERROR: value assigned to `x` is never read
   |             ^
   |
note: lint level defined here
  --> $DIR/liveness-dead.rs:12:9
   |
LL | #![deny(unused_assignments)]
   |         ^^^^^^^^^^^^^^^^^^
   = help: maybe it is overwritten before being read?

error: value assigned to `x` is never read
  --> $DIR/liveness-dead.rs:27:5
   |
LL |     x = 4; //~ ERROR: value assigned to `x` is never read
   |     ^
   |
   = help: maybe it is overwritten before being read?

error: value passed to `x` is never read
  --> $DIR/liveness-dead.rs:30:11
   |
LL | fn f4(mut x: i32) { //~ ERROR: value passed to `x` is never read
   |           ^
   |
   = help: maybe it is overwritten before being read?

error: value assigned to `x` is never read
  --> $DIR/liveness-dead.rs:37:5
   |
LL |     x = 4; //~ ERROR: value assigned to `x` is never read
   |     ^
   |
   = help: maybe it is overwritten before being read?

error: aborting due to 4 previous errors

