warning: deprecated `#[macro_use]` directive used to import macros should be replaced at use sites with a `use` statement to import the macro instead
  --> $DIR/macro-use-warned-against.rs:18:1
   |
LL | #[macro_use] //~ WARN should be replaced at use sites with a `use` statement
   | ^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/macro-use-warned-against.rs:15:9
   |
LL | #![warn(macro_use_extern_crate, unused)]
   |         ^^^^^^^^^^^^^^^^^^^^^^

warning: unused `#[macro_use]` import
  --> $DIR/macro-use-warned-against.rs:20:1
   |
LL | #[macro_use] //~ WARN unused `#[macro_use]`
   | ^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/macro-use-warned-against.rs:15:33
   |
LL | #![warn(macro_use_extern_crate, unused)]
   |                                 ^^^^^^
   = note: #[warn(unused_imports)] implied by #[warn(unused)]

