warning: deprecated `#[macro_use]` directive used to import macros should be replaced at use sites with a `use` statement to import the macro instead
  --> $DIR/macro-use-warned-against.rs:18:1
   |
LL | #[macro_use] //~ WARN should be replaced at use sites with a `use` statement
   | ^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/macro-use-warned-against.rs:15:9
   |
LL | #![warn(rust_2018_idioms, unused)]
   |         ^^^^^^^^^^^^^^^^
   = note: #[warn(macro_use_extern_crate)] implied by #[warn(rust_2018_idioms)]

warning: unused `#[macro_use]` import
  --> $DIR/macro-use-warned-against.rs:20:1
   |
LL | #[macro_use] //~ WARN unused `#[macro_use]`
   | ^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/macro-use-warned-against.rs:15:27
   |
LL | #![warn(rust_2018_idioms, unused)]
   |                           ^^^^^^
   = note: #[warn(unused_imports)] implied by #[warn(unused)]

