warning: not reporting region error due to -Znll
  --> $DIR/propagate-approximated-fail-no-postdom.rs:55:21
   |
55 |             let p = x.get();
   |                     ^^^^^^^

error: free region `'_#5r` does not outlive free region `'_#6r`
  --> $DIR/propagate-approximated-fail-no-postdom.rs:55:17
   |
55 |             let p = x.get();
   |                 ^

note: No external requirements
  --> $DIR/propagate-approximated-fail-no-postdom.rs:53:9
   |
53 | /         |_outlives1, _outlives2, _outlives3, x, y| {
54 | |             // Only works if 'x: 'y:
55 | |             let p = x.get();
56 | |             //~^ WARN not reporting region error due to -Znll
57 | |             //~| ERROR free region `'_#5r` does not outlive free region `'_#6r`
58 | |             demand_y(x, y, p)
59 | |         },
   | |_________^
   |
   = note: defining type: DefId(0/1:20 ~ propagate_approximated_fail_no_postdom[317d]::supply[0]::{{closure}}[0]) with closure substs [
               i16,
               for<'r, 's> extern "rust-call" fn((std::cell::Cell<&'_#1r &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 'r)) u32>, std::cell::Cell<&'_#2r &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 'r)) u32>, std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 's)) &'_#3r u32>, std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 'r)) u32>, std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 's)) u32>))
           ]

note: No external requirements
  --> $DIR/propagate-approximated-fail-no-postdom.rs:48:1
   |
48 | / fn supply<'a, 'b, 'c>(cell_a: Cell<&'a u32>, cell_b: Cell<&'b u32>, cell_c: Cell<&'c u32>) {
49 | |     establish_relationships(
50 | |         cell_a,
51 | |         cell_b,
...  |
60 | |     );
61 | | }
   | |_^
   |
   = note: defining type: DefId(0/0:6 ~ propagate_approximated_fail_no_postdom[317d]::supply[0]) with substs []

error: aborting due to previous error

