warning: not reporting region error due to -Znll
  --> $DIR/propagate-approximated-ref.rs:57:9
   |
57 |         demand_y(x, y, x.get()) //~ WARNING not reporting region error due to -Znll
   |         ^^^^^^^^^^^^^^^^^^^^^^^

note: External requirements
  --> $DIR/propagate-approximated-ref.rs:53:47
   |
53 |       establish_relationships(&cell_a, &cell_b, |_outlives1, _outlives2, x, y| {
   |  _______________________________________________^
54 | |         //~^ ERROR free region `'_#1r` does not outlive free region `'_#2r`
55 | |
56 | |         // Only works if 'x: 'y:
57 | |         demand_y(x, y, x.get()) //~ WARNING not reporting region error due to -Znll
58 | |     });
   | |_____^
   |
   = note: defining type: DefId(0/1:18 ~ propagate_approximated_ref[317d]::supply[0]::{{closure}}[0]) with closure substs [
               i16,
               for<'r, 's, 't0, 't1, 't2, 't3> extern "rust-call" fn((&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 'r)) std::cell::Cell<&'_#1r &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 's)) u32>, &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 't0)) std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 't1)) &'_#2r u32>, &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 't2)) std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 's)) u32>, &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 't3)) std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 't1)) u32>))
           ]
   = note: number of external vids: 3
   = note: where '_#1r: '_#2r

error: free region `'_#1r` does not outlive free region `'_#2r`
  --> $DIR/propagate-approximated-ref.rs:53:38
   |
53 |     establish_relationships(&cell_a, &cell_b, |_outlives1, _outlives2, x, y| {
   |                                      ^^^^^^^

note: No external requirements
  --> $DIR/propagate-approximated-ref.rs:52:1
   |
52 | / fn supply<'a, 'b>(cell_a: Cell<&'a u32>, cell_b: Cell<&'b u32>) {
53 | |     establish_relationships(&cell_a, &cell_b, |_outlives1, _outlives2, x, y| {
54 | |         //~^ ERROR free region `'_#1r` does not outlive free region `'_#2r`
55 | |
...  |
58 | |     });
59 | | }
   | |_^
   |
   = note: defining type: DefId(0/0:6 ~ propagate_approximated_ref[317d]::supply[0]) with substs []

error: aborting due to previous error

