warning: not reporting region error due to -Znll
  --> $DIR/propagate-approximated-shorter-to-static-no-bound.rs:49:9
   |
49 |         demand_y(x, y, x.get()) //~ WARNING not reporting region error due to -Znll
   |         ^^^^^^^^^^^^^^^^^^^^^^^

note: External requirements
  --> $DIR/propagate-approximated-shorter-to-static-no-bound.rs:45:47
   |
45 |       establish_relationships(&cell_a, &cell_b, |_outlives, x, y| {
   |  _______________________________________________^
46 | |         //~^ ERROR does not outlive free region
47 | |
48 | |         // Only works if 'x: 'y:
49 | |         demand_y(x, y, x.get()) //~ WARNING not reporting region error due to -Znll
50 | |     });
   | |_____^
   |
   = note: defining type: DefId(0/1:18 ~ propagate_approximated_shorter_to_static_no_bound[317d]::supply[0]::{{closure}}[0]) with closure substs [
               i16,
               for<'r, 's, 't0, 't1, 't2> extern "rust-call" fn((&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 'r)) std::cell::Cell<&'_#1r &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 's)) u32>, &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 't0)) std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 's)) u32>, &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 't1)) std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 't2)) u32>))
           ]
   = note: number of external vids: 2
   = note: where '_#1r: '_#0r

error: free region `ReFree(DefId(0/0:6 ~ propagate_approximated_shorter_to_static_no_bound[317d]::supply[0]), BrNamed(crate0:DefIndex(1:16), 'a))` does not outlive free region `ReStatic`
  --> $DIR/propagate-approximated-shorter-to-static-no-bound.rs:45:47
   |
45 |       establish_relationships(&cell_a, &cell_b, |_outlives, x, y| {
   |  _______________________________________________^
46 | |         //~^ ERROR does not outlive free region
47 | |
48 | |         // Only works if 'x: 'y:
49 | |         demand_y(x, y, x.get()) //~ WARNING not reporting region error due to -Znll
50 | |     });
   | |_____^

note: No external requirements
  --> $DIR/propagate-approximated-shorter-to-static-no-bound.rs:44:1
   |
44 | / fn supply<'a, 'b>(cell_a: Cell<&'a u32>, cell_b: Cell<&'b u32>) {
45 | |     establish_relationships(&cell_a, &cell_b, |_outlives, x, y| {
46 | |         //~^ ERROR does not outlive free region
47 | |
...  |
50 | |     });
51 | | }
   | |_^
   |
   = note: defining type: DefId(0/0:6 ~ propagate_approximated_shorter_to_static_no_bound[317d]::supply[0]) with substs []

error: aborting due to previous error

