warning: not reporting region error due to -Znll
  --> $DIR/return-wrong-bound-region.rs:21:23
   |
21 |     expect_sig(|a, b| b); // ought to return `a`
   |                       ^

error: free region `'_#3r` does not outlive free region `'_#2r`
  --> $DIR/return-wrong-bound-region.rs:21:23
   |
21 |     expect_sig(|a, b| b); // ought to return `a`
   |                       ^

note: External requirements
  --> $DIR/return-wrong-bound-region.rs:21:16
   |
21 |     expect_sig(|a, b| b); // ought to return `a`
   |                ^^^^^^^^
   |
   = note: defining type: DefId(0/1:9 ~ return_wrong_bound_region[317d]::test[0]::{{closure}}[0]) with closure substs [
               i16,
               for<'r, 's> extern "rust-call" fn((&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 'r)) i32, &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 's)) i32)) -> &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 'r)) i32
           ]
   = note: number of external vids: 1

note: No external requirements
  --> $DIR/return-wrong-bound-region.rs:20:1
   |
20 | / fn test() {
21 | |     expect_sig(|a, b| b); // ought to return `a`
22 | |     //~^ WARN not reporting region error due to -Znll
23 | |     //~| ERROR free region `'_#3r` does not outlive free region `'_#2r`
24 | | }
   | |_^
   |
   = note: defining type: DefId(0/0:3 ~ return_wrong_bound_region[317d]::test[0]) with substs []

error: aborting due to previous error

