error[E0541]: unknown meta item 'reason'
  --> $DIR/deprecation-sanity.rs:14:43
   |
LL |     #[deprecated(since = "a", note = "a", reason)] //~ ERROR unknown meta item 'reason'
   |                                           ^^^^^^ expected one of `since`, `note`

error[E0551]: incorrect meta item
  --> $DIR/deprecation-sanity.rs:17:31
   |
LL |     #[deprecated(since = "a", note)] //~ ERROR incorrect meta item
   |                               ^^^^

error[E0551]: incorrect meta item
  --> $DIR/deprecation-sanity.rs:20:18
   |
LL |     #[deprecated(since, note = "a")] //~ ERROR incorrect meta item
   |                  ^^^^^

error[E0551]: incorrect meta item
  --> $DIR/deprecation-sanity.rs:23:31
   |
LL |     #[deprecated(since = "a", note(b))] //~ ERROR incorrect meta item
   |                               ^^^^^^^

error[E0551]: incorrect meta item
  --> $DIR/deprecation-sanity.rs:26:18
   |
LL |     #[deprecated(since(b), note = "a")] //~ ERROR incorrect meta item
   |                  ^^^^^^^^

error[E0550]: multiple deprecated attributes
  --> $DIR/deprecation-sanity.rs:32:1
   |
LL | fn multiple1() { } //~ ERROR multiple deprecated attributes
   | ^^^^^^^^^^^^^^^^^^

error[E0538]: multiple 'since' items
  --> $DIR/deprecation-sanity.rs:34:27
   |
LL | #[deprecated(since = "a", since = "b", note = "c")] //~ ERROR multiple 'since' items
   |                           ^^^^^^^^^^^

error: aborting due to 7 previous errors

Some errors occurred: E0538, E0541, E0550, E0551.
For more information about an error, try `rustc --explain E0538`.
