error[E0599]: no method named `method` found for type `u32` in the current scope
  --> $DIR/no-method-suggested-traits.rs:38:10
   |
38 |     1u32.method();
   |          ^^^^^^
   |
   = help: items from traits can only be used if the trait is in scope
help: the following traits are implemented but not in scope, perhaps add a `use` for one of them:
   |
14 | use foo::Bar;
   |
14 | use no_method_suggested_traits::foo::PubPub;
   |
14 | use no_method_suggested_traits::qux::PrivPub;
   |
14 | use no_method_suggested_traits::Reexported;
   |

error[E0599]: no method named `method` found for type `std::rc::Rc<&mut std::boxed::Box<&u32>>` in the current scope
  --> $DIR/no-method-suggested-traits.rs:41:44
   |
41 |     std::rc::Rc::new(&mut Box::new(&1u32)).method();
   |                                            ^^^^^^
   |
   = help: items from traits can only be used if the trait is in scope
help: the following traits are implemented but not in scope, perhaps add a `use` for one of them:
   |
14 | use foo::Bar;
   |
14 | use no_method_suggested_traits::foo::PubPub;
   |
14 | use no_method_suggested_traits::qux::PrivPub;
   |
14 | use no_method_suggested_traits::Reexported;
   |

error[E0599]: no method named `method` found for type `char` in the current scope
  --> $DIR/no-method-suggested-traits.rs:45:9
   |
45 |     'a'.method();
   |         ^^^^^^
   |
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
   |
14 | use foo::Bar;
   |

error[E0599]: no method named `method` found for type `std::rc::Rc<&mut std::boxed::Box<&char>>` in the current scope
  --> $DIR/no-method-suggested-traits.rs:48:43
   |
48 |     std::rc::Rc::new(&mut Box::new(&'a')).method();
   |                                           ^^^^^^
   |
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
   |
14 | use foo::Bar;
   |

error[E0599]: no method named `method` found for type `i32` in the current scope
  --> $DIR/no-method-suggested-traits.rs:52:10
   |
52 |     1i32.method();
   |          ^^^^^^
   |
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
   |
14 | use no_method_suggested_traits::foo::PubPub;
   |

error[E0599]: no method named `method` found for type `std::rc::Rc<&mut std::boxed::Box<&i32>>` in the current scope
  --> $DIR/no-method-suggested-traits.rs:55:44
   |
55 |     std::rc::Rc::new(&mut Box::new(&1i32)).method();
   |                                            ^^^^^^
   |
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
   |
14 | use no_method_suggested_traits::foo::PubPub;
   |

error[E0599]: no method named `method` found for type `Foo` in the current scope
  --> $DIR/no-method-suggested-traits.rs:59:9
   |
14 | struct Foo; //~ HELP perhaps add a `use` for it
   | ----------- method `method` not found for this
...
59 |     Foo.method();
   |         ^^^^^^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following traits define an item `method`, perhaps you need to implement one of them:
           candidate #1: `foo::Bar`
           candidate #2: `no_method_suggested_traits::foo::PubPub`
           candidate #3: `no_method_suggested_traits::bar::PubPriv`
           candidate #4: `no_method_suggested_traits::qux::PrivPub`
           candidate #5: `no_method_suggested_traits::quz::PrivPriv`
           candidate #6: `no_method_suggested_traits::Reexported`

error[E0599]: no method named `method` found for type `std::rc::Rc<&mut std::boxed::Box<&Foo>>` in the current scope
  --> $DIR/no-method-suggested-traits.rs:62:43
   |
62 |     std::rc::Rc::new(&mut Box::new(&Foo)).method();
   |                                           ^^^^^^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following traits define an item `method`, perhaps you need to implement one of them:
           candidate #1: `foo::Bar`
           candidate #2: `no_method_suggested_traits::foo::PubPub`
           candidate #3: `no_method_suggested_traits::bar::PubPriv`
           candidate #4: `no_method_suggested_traits::qux::PrivPub`
           candidate #5: `no_method_suggested_traits::quz::PrivPriv`
           candidate #6: `no_method_suggested_traits::Reexported`

error[E0599]: no method named `method2` found for type `u64` in the current scope
  --> $DIR/no-method-suggested-traits.rs:66:10
   |
66 |     1u64.method2();
   |          ^^^^^^^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `method2`, perhaps you need to implement it:
           candidate #1: `foo::Bar`

error[E0599]: no method named `method2` found for type `std::rc::Rc<&mut std::boxed::Box<&u64>>` in the current scope
  --> $DIR/no-method-suggested-traits.rs:69:44
   |
69 |     std::rc::Rc::new(&mut Box::new(&1u64)).method2();
   |                                            ^^^^^^^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `method2`, perhaps you need to implement it:
           candidate #1: `foo::Bar`

error[E0599]: no method named `method2` found for type `no_method_suggested_traits::Foo` in the current scope
  --> $DIR/no-method-suggested-traits.rs:73:37
   |
73 |     no_method_suggested_traits::Foo.method2();
   |                                     ^^^^^^^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `method2`, perhaps you need to implement it:
           candidate #1: `foo::Bar`

error[E0599]: no method named `method2` found for type `std::rc::Rc<&mut std::boxed::Box<&no_method_suggested_traits::Foo>>` in the current scope
  --> $DIR/no-method-suggested-traits.rs:76:71
   |
76 |     std::rc::Rc::new(&mut Box::new(&no_method_suggested_traits::Foo)).method2();
   |                                                                       ^^^^^^^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `method2`, perhaps you need to implement it:
           candidate #1: `foo::Bar`

error[E0599]: no method named `method2` found for type `no_method_suggested_traits::Bar` in the current scope
  --> $DIR/no-method-suggested-traits.rs:79:40
   |
79 |     no_method_suggested_traits::Bar::X.method2();
   |                                        ^^^^^^^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `method2`, perhaps you need to implement it:
           candidate #1: `foo::Bar`

error[E0599]: no method named `method2` found for type `std::rc::Rc<&mut std::boxed::Box<&no_method_suggested_traits::Bar>>` in the current scope
  --> $DIR/no-method-suggested-traits.rs:82:74
   |
82 |     std::rc::Rc::new(&mut Box::new(&no_method_suggested_traits::Bar::X)).method2();
   |                                                                          ^^^^^^^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `method2`, perhaps you need to implement it:
           candidate #1: `foo::Bar`

error[E0599]: no method named `method3` found for type `Foo` in the current scope
  --> $DIR/no-method-suggested-traits.rs:86:9
   |
14 | struct Foo; //~ HELP perhaps add a `use` for it
   | ----------- method `method3` not found for this
...
86 |     Foo.method3();
   |         ^^^^^^^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `method3`, perhaps you need to implement it:
           candidate #1: `no_method_suggested_traits::foo::PubPub`

error[E0599]: no method named `method3` found for type `std::rc::Rc<&mut std::boxed::Box<&Foo>>` in the current scope
  --> $DIR/no-method-suggested-traits.rs:89:43
   |
89 |     std::rc::Rc::new(&mut Box::new(&Foo)).method3();
   |                                           ^^^^^^^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `method3`, perhaps you need to implement it:
           candidate #1: `no_method_suggested_traits::foo::PubPub`

error[E0599]: no method named `method3` found for type `Bar` in the current scope
  --> $DIR/no-method-suggested-traits.rs:92:12
   |
20 | enum Bar { X }
   | -------- method `method3` not found for this
...
92 |     Bar::X.method3();
   |            ^^^^^^^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `method3`, perhaps you need to implement it:
           candidate #1: `no_method_suggested_traits::foo::PubPub`

error[E0599]: no method named `method3` found for type `std::rc::Rc<&mut std::boxed::Box<&Bar>>` in the current scope
  --> $DIR/no-method-suggested-traits.rs:95:46
   |
95 |     std::rc::Rc::new(&mut Box::new(&Bar::X)).method3();
   |                                              ^^^^^^^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `method3`, perhaps you need to implement it:
           candidate #1: `no_method_suggested_traits::foo::PubPub`

error[E0599]: no method named `method3` found for type `usize` in the current scope
   --> $DIR/no-method-suggested-traits.rs:100:13
    |
100 |     1_usize.method3(); //~ ERROR no method named
    |             ^^^^^^^

error[E0599]: no method named `method3` found for type `std::rc::Rc<&mut std::boxed::Box<&usize>>` in the current scope
   --> $DIR/no-method-suggested-traits.rs:101:47
    |
101 |     std::rc::Rc::new(&mut Box::new(&1_usize)).method3(); //~ ERROR no method named
    |                                               ^^^^^^^

error[E0599]: no method named `method3` found for type `no_method_suggested_traits::Foo` in the current scope
   --> $DIR/no-method-suggested-traits.rs:102:37
    |
102 |     no_method_suggested_traits::Foo.method3();  //~ ERROR no method named
    |                                     ^^^^^^^

error[E0599]: no method named `method3` found for type `std::rc::Rc<&mut std::boxed::Box<&no_method_suggested_traits::Foo>>` in the current scope
   --> $DIR/no-method-suggested-traits.rs:103:71
    |
103 |     std::rc::Rc::new(&mut Box::new(&no_method_suggested_traits::Foo)).method3();
    |                                                                       ^^^^^^^

error[E0599]: no method named `method3` found for type `no_method_suggested_traits::Bar` in the current scope
   --> $DIR/no-method-suggested-traits.rs:105:40
    |
105 |     no_method_suggested_traits::Bar::X.method3();  //~ ERROR no method named
    |                                        ^^^^^^^

error[E0599]: no method named `method3` found for type `std::rc::Rc<&mut std::boxed::Box<&no_method_suggested_traits::Bar>>` in the current scope
   --> $DIR/no-method-suggested-traits.rs:106:74
    |
106 |     std::rc::Rc::new(&mut Box::new(&no_method_suggested_traits::Bar::X)).method3();
    |                                                                          ^^^^^^^

error: aborting due to 24 previous errors

