error: BaseDefault
  --> $DIR/object-lifetime-default.rs:16:1
   |
LL | struct A<T>(T); //~ ERROR BaseDefault
   | ^^^^^^^^^^^^^^^

error: BaseDefault
  --> $DIR/object-lifetime-default.rs:19:1
   |
LL | struct B<'a,T>(&'a (), T); //~ ERROR BaseDefault
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: 'a
  --> $DIR/object-lifetime-default.rs:22:1
   |
LL | struct C<'a,T:'a>(&'a T); //~ ERROR 'a
   | ^^^^^^^^^^^^^^^^^^^^^^^^^

error: Ambiguous
  --> $DIR/object-lifetime-default.rs:25:1
   |
LL | struct D<'a,'b,T:'a+'b>(&'a T, &'b T); //~ ERROR Ambiguous
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: 'b
  --> $DIR/object-lifetime-default.rs:28:1
   |
LL | struct E<'a,'b:'a,T:'b>(&'a T, &'b T); //~ ERROR 'b
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: 'a,'b
  --> $DIR/object-lifetime-default.rs:31:1
   |
LL | struct F<'a,'b,T:'a,U:'b>(&'a T, &'b U); //~ ERROR 'a,'b
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: 'a,Ambiguous
  --> $DIR/object-lifetime-default.rs:34:1
   |
LL | struct G<'a,'b,T:'a,U:'a+'b>(&'a T, &'b U); //~ ERROR 'a,Ambiguous
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 7 previous errors

