error: `proc-macro` crate types cannot export any items other than functions tagged with `#[proc_macro_derive]` currently
  --> $DIR/exports.rs:17:1
   |
LL | pub fn a() {} //~ ERROR: cannot export any items
   | ^^^^^^^^^^^^^

error: `proc-macro` crate types cannot export any items other than functions tagged with `#[proc_macro_derive]` currently
  --> $DIR/exports.rs:18:1
   |
LL | pub struct B; //~ ERROR: cannot export any items
   | ^^^^^^^^^^^^^

error: `proc-macro` crate types cannot export any items other than functions tagged with `#[proc_macro_derive]` currently
  --> $DIR/exports.rs:19:1
   |
LL | pub enum C {} //~ ERROR: cannot export any items
   | ^^^^^^^^^^^^^

error: `proc-macro` crate types cannot export any items other than functions tagged with `#[proc_macro_derive]` currently
  --> $DIR/exports.rs:20:1
   |
LL | pub mod d {} //~ ERROR: cannot export any items
   | ^^^^^^^^^^^^

error: aborting due to 4 previous errors

