error: can't use a procedural macro from the same crate that defines it
  --> $DIR/macro-namespace-reserved-2.rs:35:5
   |
LL |     my_macro!(); //~ ERROR can't use a procedural macro from the same crate that defines it
   |     ^^^^^^^^

error: can't use a procedural macro from the same crate that defines it
  --> $DIR/macro-namespace-reserved-2.rs:38:5
   |
LL |     my_macro_attr!(); //~ ERROR can't use a procedural macro from the same crate that defines it
   |     ^^^^^^^^^^^^^

error: can't use a procedural macro from the same crate that defines it
  --> $DIR/macro-namespace-reserved-2.rs:41:5
   |
LL |     MyTrait!(); //~ ERROR can't use a procedural macro from the same crate that defines it
   |     ^^^^^^^

error: can't use a procedural macro from the same crate that defines it
  --> $DIR/macro-namespace-reserved-2.rs:44:3
   |
LL | #[my_macro] //~ ERROR can't use a procedural macro from the same crate that defines it
   |   ^^^^^^^^

error: can't use a procedural macro from the same crate that defines it
  --> $DIR/macro-namespace-reserved-2.rs:46:3
   |
LL | #[my_macro_attr] //~ ERROR can't use a procedural macro from the same crate that defines it
   |   ^^^^^^^^^^^^^

error: can't use a procedural macro from the same crate that defines it
  --> $DIR/macro-namespace-reserved-2.rs:48:3
   |
LL | #[MyTrait] //~ ERROR can't use a procedural macro from the same crate that defines it
   |   ^^^^^^^

error: can't use a procedural macro from the same crate that defines it
  --> $DIR/macro-namespace-reserved-2.rs:51:10
   |
LL | #[derive(my_macro)] //~ ERROR can't use a procedural macro from the same crate that defines it
   |          ^^^^^^^^

error: can't use a procedural macro from the same crate that defines it
  --> $DIR/macro-namespace-reserved-2.rs:53:10
   |
LL | #[derive(my_macro_attr)] //~ ERROR can't use a procedural macro from the same crate that defines it
   |          ^^^^^^^^^^^^^

error: can't use a procedural macro from the same crate that defines it
  --> $DIR/macro-namespace-reserved-2.rs:55:10
   |
LL | #[derive(MyTrait)] //~ ERROR can't use a procedural macro from the same crate that defines it
   |          ^^^^^^^

error: aborting due to 9 previous errors

