error[E0658]: non-builtin inner attributes are unstable (see issue #54726)
  --> $DIR/proc-macro-gates.rs:21:5
   |
LL |     #![a] //~ ERROR: non-builtin inner attributes are unstable
   |     ^^^^^
   |
   = help: add #![feature(custom_inner_attributes)] to the crate attributes to enable

error[E0658]: non-builtin inner attributes are unstable (see issue #54726)
  --> $DIR/proc-macro-gates.rs:28:5
   |
LL |     #![a] //~ ERROR: custom attributes cannot be applied to modules
   |     ^^^^^
   |
   = help: add #![feature(custom_inner_attributes)] to the crate attributes to enable

error[E0658]: custom attributes cannot be applied to modules (see issue #54727)
  --> $DIR/proc-macro-gates.rs:24:1
   |
LL | #[a] //~ ERROR: custom attributes cannot be applied to modules
   | ^^^^
   |
   = help: add #![feature(proc_macro_hygiene)] to the crate attributes to enable

error[E0658]: custom attributes cannot be applied to modules (see issue #54727)
  --> $DIR/proc-macro-gates.rs:28:5
   |
LL |     #![a] //~ ERROR: custom attributes cannot be applied to modules
   |     ^^^^^
   |
   = help: add #![feature(proc_macro_hygiene)] to the crate attributes to enable

error: custom attribute invocations must be of the form #[foo] or #[foo(..)], the macro name must only be followed by a delimiter token
  --> $DIR/proc-macro-gates.rs:32:1
   |
LL | #[a = y] //~ ERROR: must only be followed by a delimiter token
   | ^^^^^^^^

error[E0658]: custom attributes cannot be applied to statements (see issue #54727)
  --> $DIR/proc-macro-gates.rs:41:5
   |
LL |     #[a] //~ ERROR: custom attributes cannot be applied to statements
   |     ^^^^
   |
   = help: add #![feature(proc_macro_hygiene)] to the crate attributes to enable

error[E0658]: custom attributes cannot be applied to statements (see issue #54727)
  --> $DIR/proc-macro-gates.rs:45:5
   |
LL |     #[a] //~ ERROR: custom attributes cannot be applied to statements
   |     ^^^^
   |
   = help: add #![feature(proc_macro_hygiene)] to the crate attributes to enable

error[E0658]: custom attributes cannot be applied to statements (see issue #54727)
  --> $DIR/proc-macro-gates.rs:49:5
   |
LL |     #[a] //~ ERROR: custom attributes cannot be applied to statements
   |     ^^^^
   |
   = help: add #![feature(proc_macro_hygiene)] to the crate attributes to enable

error[E0658]: custom attributes cannot be applied to expressions (see issue #54727)
  --> $DIR/proc-macro-gates.rs:53:14
   |
LL |     let _x = #[a] 2; //~ ERROR: custom attributes cannot be applied to expressions
   |              ^^^^
   |
   = help: add #![feature(proc_macro_hygiene)] to the crate attributes to enable

error[E0658]: custom attributes cannot be applied to expressions (see issue #54727)
  --> $DIR/proc-macro-gates.rs:56:15
   |
LL |     let _x = [#[a] 2]; //~ ERROR: custom attributes cannot be applied to expressions
   |               ^^^^
   |
   = help: add #![feature(proc_macro_hygiene)] to the crate attributes to enable

error[E0658]: custom attributes cannot be applied to expressions (see issue #54727)
  --> $DIR/proc-macro-gates.rs:59:14
   |
LL |     let _x = #[a] println!(); //~ ERROR: custom attributes cannot be applied to expressions
   |              ^^^^
   |
   = help: add #![feature(proc_macro_hygiene)] to the crate attributes to enable

error[E0658]: procedural macros cannot be expanded to types (see issue #54727)
  --> $DIR/proc-macro-gates.rs:63:13
   |
LL |     let _x: m!(u32) = 3; //~ ERROR: procedural macros cannot be expanded to types
   |             ^^^^^^^
   |
   = help: add #![feature(proc_macro_hygiene)] to the crate attributes to enable

error[E0658]: procedural macros cannot be expanded to patterns (see issue #54727)
  --> $DIR/proc-macro-gates.rs:64:12
   |
LL |     if let m!(Some(_x)) = Some(3) {} //~ ERROR: procedural macros cannot be expanded to patterns
   |            ^^^^^^^^^^^^
   |
   = help: add #![feature(proc_macro_hygiene)] to the crate attributes to enable

error[E0658]: procedural macros cannot be expanded to statements (see issue #54727)
  --> $DIR/proc-macro-gates.rs:66:5
   |
LL |     m!(struct S;); //~ ERROR: procedural macros cannot be expanded to statements
   |     ^^^^^^^^^^^^^^
   |
   = help: add #![feature(proc_macro_hygiene)] to the crate attributes to enable

error[E0658]: procedural macros cannot be expanded to statements (see issue #54727)
  --> $DIR/proc-macro-gates.rs:67:5
   |
LL |     m!(let _x = 3;); //~ ERROR: procedural macros cannot be expanded to statements
   |     ^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(proc_macro_hygiene)] to the crate attributes to enable

error[E0658]: procedural macros cannot be expanded to expressions (see issue #54727)
  --> $DIR/proc-macro-gates.rs:69:14
   |
LL |     let _x = m!(3); //~ ERROR: procedural macros cannot be expanded to expressions
   |              ^^^^^
   |
   = help: add #![feature(proc_macro_hygiene)] to the crate attributes to enable

error[E0658]: procedural macros cannot be expanded to expressions (see issue #54727)
  --> $DIR/proc-macro-gates.rs:70:15
   |
LL |     let _x = [m!(3)]; //~ ERROR: procedural macros cannot be expanded to expressions
   |               ^^^^^
   |
   = help: add #![feature(proc_macro_hygiene)] to the crate attributes to enable

error: aborting due to 17 previous errors

For more information about this error, try `rustc --explain E0658`.
