error: expected identifier, found `+`
  --> $DIR/issue-33569.rs:12:8
   |
LL |     { $+ } => { //~ ERROR expected identifier, found `+`
   |        ^

error: expected `*` or `+`
  --> $DIR/issue-33569.rs:14:13
   |
LL |         $(x)(y) //~ ERROR expected `*` or `+`
   |             ^^^

error: missing fragment specifier
  --> $DIR/issue-33569.rs:12:8
   |
LL |     { $+ } => { //~ ERROR expected identifier, found `+`
   |        ^

error: aborting due to 3 previous errors

