error[E0119]: conflicting implementations of trait `std::ops::Deref` for type `&_`:
  --> $DIR/issue-28981.rs:15:1
   |
15 | impl<Foo> Deref for Foo { } //~ ERROR must be used
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `core`:
           - impl<'a, T> std::ops::Deref for &'a T
             where T: ?Sized;

error[E0210]: type parameter `Foo` must be used as the type parameter for some local type (e.g. `MyStruct<T>`); only traits defined in the current crate can be implemented for a type parameter
  --> $DIR/issue-28981.rs:15:1
   |
15 | impl<Foo> Deref for Foo { } //~ ERROR must be used
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

