error[E0597]: `c` does not live long enough
  --> $DIR/adt-tuple-struct.rs:35:32
   |
LL |     SomeStruct::<&'static u32>(&c); //~ ERROR
   |                                ^^ borrowed value does not live long enough
LL | }
   | - `c` dropped here while still borrowed
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0597]: `c` does not live long enough
  --> $DIR/adt-tuple-struct.rs:40:27
   |
LL |     SomeStruct::<&'a u32>(&c); //~ ERROR
   |                           ^^ borrowed value does not live long enough
LL | }
   | - `c` dropped here while still borrowed
   |
note: borrowed value must be valid for the lifetime 'a as defined on the function body at 38:35...
  --> $DIR/adt-tuple-struct.rs:38:35
   |
LL | fn annot_reference_named_lifetime<'a>(_d: &'a u32) {
   |                                   ^^

error[E0597]: `c` does not live long enough
  --> $DIR/adt-tuple-struct.rs:50:31
   |
LL |         SomeStruct::<&'a u32>(&c); //~ ERROR
   |                               ^^ borrowed value does not live long enough
LL |     };
   |     - `c` dropped here while still borrowed
   |
note: borrowed value must be valid for the lifetime 'a as defined on the function body at 47:46...
  --> $DIR/adt-tuple-struct.rs:47:46
   |
LL | fn annot_reference_named_lifetime_in_closure<'a>(_: &'a u32) {
   |                                              ^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0597`.
