error[E0597]: `c` does not live long enough
  --> $DIR/adt-brace-enums.rs:37:48
   |
LL |     SomeEnum::SomeVariant::<&'static u32> { t: &c }; //~ ERROR
   |                                                ^^ borrowed value does not live long enough
LL | }
   | - `c` dropped here while still borrowed
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0597]: `c` does not live long enough
  --> $DIR/adt-brace-enums.rs:42:43
   |
LL |     SomeEnum::SomeVariant::<&'a u32> { t: &c }; //~ ERROR
   |                                           ^^ borrowed value does not live long enough
LL | }
   | - `c` dropped here while still borrowed
   |
note: borrowed value must be valid for the lifetime 'a as defined on the function body at 40:35...
  --> $DIR/adt-brace-enums.rs:40:35
   |
LL | fn annot_reference_named_lifetime<'a>(_d: &'a u32) {
   |                                   ^^

error[E0597]: `c` does not live long enough
  --> $DIR/adt-brace-enums.rs:52:47
   |
LL |         SomeEnum::SomeVariant::<&'a u32> { t: &c }; //~ ERROR
   |                                               ^^ borrowed value does not live long enough
LL |     };
   |     - `c` dropped here while still borrowed
   |
note: borrowed value must be valid for the lifetime 'a as defined on the function body at 49:46...
  --> $DIR/adt-brace-enums.rs:49:46
   |
LL | fn annot_reference_named_lifetime_in_closure<'a>(_: &'a u32) {
   |                                              ^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0597`.
