error[E0597]: `c` does not live long enough
  --> $DIR/adt-brace-structs.rs:35:37
   |
LL |     SomeStruct::<&'static u32> { t: &c }; //~ ERROR
   |                                     ^^ borrowed value does not live long enough
LL | }
   | - `c` dropped here while still borrowed
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0597]: `c` does not live long enough
  --> $DIR/adt-brace-structs.rs:40:32
   |
LL |     SomeStruct::<&'a u32> { t: &c }; //~ ERROR
   |                                ^^ borrowed value does not live long enough
LL | }
   | - `c` dropped here while still borrowed
   |
note: borrowed value must be valid for the lifetime 'a as defined on the function body at 38:35...
  --> $DIR/adt-brace-structs.rs:38:35
   |
LL | fn annot_reference_named_lifetime<'a>(_d: &'a u32) {
   |                                   ^^

error[E0597]: `c` does not live long enough
  --> $DIR/adt-brace-structs.rs:50:36
   |
LL |         SomeStruct::<&'a u32> { t: &c }; //~ ERROR
   |                                    ^^ borrowed value does not live long enough
LL |     };
   |     - `c` dropped here while still borrowed
   |
note: borrowed value must be valid for the lifetime 'a as defined on the function body at 47:46...
  --> $DIR/adt-brace-structs.rs:47:46
   |
LL | fn annot_reference_named_lifetime_in_closure<'a>(_: &'a u32) {
   |                                              ^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0597`.
