error[E0597]: `c` does not live long enough
  --> $DIR/adt-nullary-enums.rs:44:41
   |
LL |         SomeEnum::SomeVariant(Cell::new(&c)), //~ ERROR
   |                                         ^^ borrowed value does not live long enough
...
LL | }
   | - `c` dropped here while still borrowed
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0597]: `c` does not live long enough
  --> $DIR/adt-nullary-enums.rs:52:41
   |
LL |         SomeEnum::SomeVariant(Cell::new(&c)), //~ ERROR
   |                                         ^^ borrowed value does not live long enough
...
LL | }
   | - `c` dropped here while still borrowed
   |
note: borrowed value must be valid for the lifetime 'a as defined on the function body at 49:35...
  --> $DIR/adt-nullary-enums.rs:49:35
   |
LL | fn annot_reference_named_lifetime<'a>(_d: &'a u32) {
   |                                   ^^

error[E0597]: `c` does not live long enough
  --> $DIR/adt-nullary-enums.rs:65:45
   |
LL |             SomeEnum::SomeVariant(Cell::new(&c)), //~ ERROR
   |                                             ^^ borrowed value does not live long enough
...
LL |     };
   |     - `c` dropped here while still borrowed
   |
note: borrowed value must be valid for the lifetime 'a as defined on the function body at 61:46...
  --> $DIR/adt-nullary-enums.rs:61:46
   |
LL | fn annot_reference_named_lifetime_in_closure<'a>(_: &'a u32) {
   |                                              ^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0597`.
