error[E0597]: `c` does not live long enough
  --> $DIR/adt-tuple-enums.rs:39:43
   |
LL |     SomeEnum::SomeVariant::<&'static u32>(&c); //~ ERROR
   |                                           ^^ borrowed value does not live long enough
LL | }
   | - `c` dropped here while still borrowed
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0597]: `c` does not live long enough
  --> $DIR/adt-tuple-enums.rs:44:38
   |
LL |     SomeEnum::SomeVariant::<&'a u32>(&c); //~ ERROR
   |                                      ^^ borrowed value does not live long enough
LL | }
   | - `c` dropped here while still borrowed
   |
note: borrowed value must be valid for the lifetime 'a as defined on the function body at 42:35...
  --> $DIR/adt-tuple-enums.rs:42:35
   |
LL | fn annot_reference_named_lifetime<'a>(_d: &'a u32) {
   |                                   ^^

error[E0597]: `c` does not live long enough
  --> $DIR/adt-tuple-enums.rs:54:42
   |
LL |         SomeEnum::SomeVariant::<&'a u32>(&c); //~ ERROR
   |                                          ^^ borrowed value does not live long enough
LL |     };
   |     - `c` dropped here while still borrowed
   |
note: borrowed value must be valid for the lifetime 'a as defined on the function body at 51:46...
  --> $DIR/adt-tuple-enums.rs:51:46
   |
LL | fn annot_reference_named_lifetime_in_closure<'a>(_: &'a u32) {
   |                                              ^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0597`.
