error[E0626]: borrow may still be in use when generator yields (Ast)
  --> $DIR/generator-with-nll.rs:19:23
   |
19 |         let _a = &mut true; //~ ERROR borrow may still be in use when generator yields (Ast)
   |                       ^^^^
...
22 |         yield ();
   |         -------- possible yield occurs here

error[E0626]: borrow may still be in use when generator yields (Ast)
  --> $DIR/generator-with-nll.rs:20:22
   |
20 |         let b = &mut true; //~ ERROR borrow may still be in use when generator yields (Ast)
   |                      ^^^^
21 |         //~^ borrow may still be in use when generator yields (Mir)
22 |         yield ();
   |         -------- possible yield occurs here

error[E0626]: borrow may still be in use when generator yields (Mir)
  --> $DIR/generator-with-nll.rs:20:17
   |
20 |         let b = &mut true; //~ ERROR borrow may still be in use when generator yields (Ast)
   |                 ^^^^^^^^^
21 |         //~^ borrow may still be in use when generator yields (Mir)
22 |         yield ();
   |         -------- possible yield occurs here

error: aborting due to 3 previous errors

