error[E0277]: the trait bound `str: std::marker::Sized` is not satisfied
  --> $DIR/sized-yield.rs:17:26
   |
17 |      let mut gen = move || { //~ ERROR the trait bound `str: std::marker::Sized` is not satisfied
   |  __________________________^
18 | |        yield s[..];
19 | |    };
   | |____^ `str` does not have a constant size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `str`
   = note: the yield type of a generator must have a statically known size

error[E0277]: the trait bound `str: std::marker::Sized` is not satisfied
  --> $DIR/sized-yield.rs:20:8
   |
20 |    gen.resume(); //~ ERROR the trait bound `str: std::marker::Sized` is not satisfied
   |        ^^^^^^ `str` does not have a constant size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `str`

error: aborting due to 2 previous errors

