error[E0507]: cannot move out of static item
  --> $DIR/static-items-cant-move.rs:28:10
   |
LL |     test(BAR); //~ ERROR cannot move out of static item
   |          ^^^ cannot move out of static item

error[E0507]: cannot move out of immutable static item `BAR`
  --> $DIR/static-items-cant-move.rs:28:10
   |
LL |     test(BAR); //~ ERROR cannot move out of static item
   |          ^^^
   |          |
   |          cannot move out of immutable static item `BAR`
   |          cannot move

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0507`.
