error[E0531]: cannot find unit struct/variant or constant `Self` in this scope
  --> $DIR/keyword-self-as-identifier.rs:12:9
   |
LL |     let Self = 22; //~ ERROR cannot find unit struct/variant or constant `Self` in this scope
   |         ^^^^ not found in this scope

error[E0658]: `Self` struct constructors are unstable (see issue #51994)
  --> $DIR/keyword-self-as-identifier.rs:12:9
   |
LL |     let Self = 22; //~ ERROR cannot find unit struct/variant or constant `Self` in this scope
   |         ^^^^
   |
   = help: add #![feature(self_struct_ctor)] to the crate attributes to enable

error: aborting due to 2 previous errors

Some errors occurred: E0531, E0658.
For more information about an error, try `rustc --explain E0531`.
