error[E0596]: cannot borrow field `f.v` of immutable binding as mutable
  --> $DIR/issue-35937.rs:17:5
   |
LL |     let f = Foo { v: Vec::new() };
   |         - consider changing this to `mut f`
LL |     f.v.push("cat".to_string()); //~ ERROR cannot borrow
   |     ^^^ cannot mutably borrow field of immutable binding

error[E0594]: cannot assign to field `s.x` of immutable binding
  --> $DIR/issue-35937.rs:26:5
   |
LL |     let s = S { x: 42 };
   |         - consider changing this to `mut s`
LL |     s.x += 1; //~ ERROR cannot assign
   |     ^^^^^^^^ cannot mutably borrow field of immutable binding

error[E0594]: cannot assign to field `s.x` of immutable binding
  --> $DIR/issue-35937.rs:30:5
   |
LL | fn bar(s: S) {
   |        - consider changing this to `mut s`
LL |     s.x += 1; //~ ERROR cannot assign
   |     ^^^^^^^^ cannot mutably borrow field of immutable binding

error: aborting due to 3 previous errors

You've got a few errors: E0594, E0596
If you want more information on an error, try using "rustc --explain E0594"
