error[E0596]: cannot borrow immutable argument `self` as mutable
  --> $DIR/issue-31424.rs:17:15
   |
LL |         (&mut self).bar(); //~ ERROR cannot borrow
   |               ^^^^
   |               |
   |               cannot reborrow mutably
   |               try removing `&mut` here

error[E0596]: cannot borrow immutable argument `self` as mutable
  --> $DIR/issue-31424.rs:23:15
   |
LL |     fn bar(self: &mut Self) {
   |            --------------- consider changing this to `mut self: &mut Self`
LL |         (&mut self).bar(); //~ ERROR cannot borrow
   |               ^^^^ cannot borrow mutably

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0596`.
