error[E0055]: reached the recursion limit while auto-dereferencing I
  --> $DIR/recursion_limit_deref.rs:62:22
   |
LL |     let x: &Bottom = &t; //~ ERROR mismatched types
   |                      ^^ deref recursion limit reached
   |
   = help: consider adding a `#![recursion_limit="20"]` attribute to your crate

error[E0055]: reached the recursion limit while auto-dereferencing I
   |
   = help: consider adding a `#![recursion_limit="20"]` attribute to your crate

error[E0308]: mismatched types
  --> $DIR/recursion_limit_deref.rs:62:22
   |
LL |     let x: &Bottom = &t; //~ ERROR mismatched types
   |                      ^^ expected struct `Bottom`, found struct `Top`
   |
   = note: expected type `&Bottom`
              found type `&Top`

error: aborting due to 3 previous errors

You've got a few errors: E0055, E0308
If you want more information on an error, try using "rustc --explain E0055"
