error[E0178]: expected a path on the left-hand side of `+`, not `&Copy`
  --> $DIR/trait-object-reference-without-parens-suggestion.rs:12:12
   |
12 |     let _: &Copy + 'static; //~ ERROR expected a path
   |            ^^^^^^^^^^^^^^^ help: try adding parentheses: `&(Copy + 'static)`

error[E0178]: expected a path on the left-hand side of `+`, not `&'static Copy`
  --> $DIR/trait-object-reference-without-parens-suggestion.rs:14:12
   |
14 |     let _: &'static Copy + 'static; //~ ERROR expected a path
   |            ^^^^^^^^^^^^^^^^^^^^^^^ help: try adding parentheses: `&'static (Copy + 'static)`

error[E0038]: the trait `std::marker::Copy` cannot be made into an object
  --> $DIR/trait-object-reference-without-parens-suggestion.rs:12:12
   |
12 |     let _: &Copy + 'static; //~ ERROR expected a path
   |            ^^^^^ the trait `std::marker::Copy` cannot be made into an object
   |
   = note: the trait cannot require that `Self : Sized`

error: aborting due to 3 previous errors

You've got a few errors: E0038, E0178
If you want more information on an error, try using "rustc --explain E0038"
