error[E0631]: type mismatch in closure arguments
  --> $DIR/anonymous-higher-ranked-lifetime.rs:12:5
   |
12 |     f1(|_: (), _: ()| {}); //~ ERROR type mismatch
   |     ^^ -------------- found signature of `fn((), ()) -> _`
   |     |
   |     expected signature of `for<'r, 's> fn(&'r (), &'s ()) -> _`
   |
note: required by `f1`
  --> $DIR/anonymous-higher-ranked-lifetime.rs:26:1
   |
26 | fn f1<F>(_: F) where F: Fn(&(), &()) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0631]: type mismatch in closure arguments
  --> $DIR/anonymous-higher-ranked-lifetime.rs:13:5
   |
13 |     f2(|_: (), _: ()| {}); //~ ERROR type mismatch
   |     ^^ -------------- found signature of `fn((), ()) -> _`
   |     |
   |     expected signature of `for<'a, 'r> fn(&'a (), &'r ()) -> _`
   |
note: required by `f2`
  --> $DIR/anonymous-higher-ranked-lifetime.rs:27:1
   |
27 | fn f2<F>(_: F) where F: for<'a> Fn(&'a (), &()) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0631]: type mismatch in closure arguments
  --> $DIR/anonymous-higher-ranked-lifetime.rs:14:5
   |
14 |     f3(|_: (), _: ()| {}); //~ ERROR type mismatch
   |     ^^ -------------- found signature of `fn((), ()) -> _`
   |     |
   |     expected signature of `for<'r> fn(&(), &'r ()) -> _`
   |
note: required by `f3`
  --> $DIR/anonymous-higher-ranked-lifetime.rs:28:1
   |
28 | fn f3<'a, F>(_: F) where F: Fn(&'a (), &()) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0631]: type mismatch in closure arguments
  --> $DIR/anonymous-higher-ranked-lifetime.rs:15:5
   |
15 |     f4(|_: (), _: ()| {}); //~ ERROR type mismatch
   |     ^^ -------------- found signature of `fn((), ()) -> _`
   |     |
   |     expected signature of `for<'s, 'r> fn(&'s (), &'r ()) -> _`
   |
note: required by `f4`
  --> $DIR/anonymous-higher-ranked-lifetime.rs:29:1
   |
29 | fn f4<F>(_: F) where F: for<'r> Fn(&(), &'r ()) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0631]: type mismatch in closure arguments
  --> $DIR/anonymous-higher-ranked-lifetime.rs:16:5
   |
16 |     f5(|_: (), _: ()| {}); //~ ERROR type mismatch
   |     ^^ -------------- found signature of `fn((), ()) -> _`
   |     |
   |     expected signature of `for<'r> fn(&'r (), &'r ()) -> _`
   |
note: required by `f5`
  --> $DIR/anonymous-higher-ranked-lifetime.rs:30:1
   |
30 | fn f5<F>(_: F) where F: for<'r> Fn(&'r (), &'r ()) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0631]: type mismatch in closure arguments
  --> $DIR/anonymous-higher-ranked-lifetime.rs:17:5
   |
17 |     g1(|_: (), _: ()| {}); //~ ERROR type mismatch
   |     ^^ -------------- found signature of `fn((), ()) -> _`
   |     |
   |     expected signature of `for<'r> fn(&'r (), std::boxed::Box<for<'s> std::ops::Fn(&'s ()) + 'static>) -> _`
   |
note: required by `g1`
  --> $DIR/anonymous-higher-ranked-lifetime.rs:33:1
   |
33 | fn g1<F>(_: F) where F: Fn(&(), Box<Fn(&())>) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0631]: type mismatch in closure arguments
  --> $DIR/anonymous-higher-ranked-lifetime.rs:18:5
   |
18 |     g2(|_: (), _: ()| {}); //~ ERROR type mismatch
   |     ^^ -------------- found signature of `fn((), ()) -> _`
   |     |
   |     expected signature of `for<'r> fn(&'r (), for<'s> fn(&'s ())) -> _`
   |
note: required by `g2`
  --> $DIR/anonymous-higher-ranked-lifetime.rs:34:1
   |
34 | fn g2<F>(_: F) where F: Fn(&(), fn(&())) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0631]: type mismatch in closure arguments
  --> $DIR/anonymous-higher-ranked-lifetime.rs:19:5
   |
19 |     g3(|_: (), _: ()| {}); //~ ERROR type mismatch
   |     ^^ -------------- found signature of `fn((), ()) -> _`
   |     |
   |     expected signature of `for<'s> fn(&'s (), std::boxed::Box<for<'r> std::ops::Fn(&'r ()) + 'static>) -> _`
   |
note: required by `g3`
  --> $DIR/anonymous-higher-ranked-lifetime.rs:35:1
   |
35 | fn g3<F>(_: F) where F: for<'s> Fn(&'s (), Box<Fn(&())>) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0631]: type mismatch in closure arguments
  --> $DIR/anonymous-higher-ranked-lifetime.rs:20:5
   |
20 |     g4(|_: (), _: ()| {}); //~ ERROR type mismatch
   |     ^^ -------------- found signature of `fn((), ()) -> _`
   |     |
   |     expected signature of `for<'s> fn(&'s (), for<'r> fn(&'r ())) -> _`
   |
note: required by `g4`
  --> $DIR/anonymous-higher-ranked-lifetime.rs:36:1
   |
36 | fn g4<F>(_: F) where F: Fn(&(), for<'r> fn(&'r ())) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0631]: type mismatch in closure arguments
  --> $DIR/anonymous-higher-ranked-lifetime.rs:21:5
   |
21 |     h1(|_: (), _: (), _: (), _: ()| {}); //~ ERROR type mismatch
   |     ^^ ---------------------------- found signature of `fn((), (), (), ()) -> _`
   |     |
   |     expected signature of `for<'r, 's> fn(&'r (), std::boxed::Box<for<'t0> std::ops::Fn(&'t0 ()) + 'static>, &'s (), for<'t0, 't1> fn(&'t0 (), &'t1 ())) -> _`
   |
note: required by `h1`
  --> $DIR/anonymous-higher-ranked-lifetime.rs:39:1
   |
39 | fn h1<F>(_: F) where F: Fn(&(), Box<Fn(&())>, &(), fn(&(), &())) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0631]: type mismatch in closure arguments
  --> $DIR/anonymous-higher-ranked-lifetime.rs:22:5
   |
22 |     h2(|_: (), _: (), _: (), _: ()| {}); //~ ERROR type mismatch
   |     ^^ ---------------------------- found signature of `fn((), (), (), ()) -> _`
   |     |
   |     expected signature of `for<'r, 't0> fn(&'r (), std::boxed::Box<for<'s> std::ops::Fn(&'s ()) + 'static>, &'t0 (), for<'s, 't1> fn(&'s (), &'t1 ())) -> _`
   |
note: required by `h2`
  --> $DIR/anonymous-higher-ranked-lifetime.rs:40:1
   |
40 | fn h2<F>(_: F) where F: for<'t0> Fn(&(), Box<Fn(&())>, &'t0 (), fn(&(), &())) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 11 previous errors

