error: expected `)`, found `,`
  --> $DIR/cfg-attr-trailing-comma.rs:9:25
   |
LL | #[cfg_attr(TRUE, inline,,)] //~ ERROR expected `)`, found `,`
   |                         ^ expected `)`

error: expected `)`, found `,`
  --> $DIR/cfg-attr-trailing-comma.rs:12:26
   |
LL | #[cfg_attr(FALSE, inline,,)] //~ ERROR expected `)`, found `,`
   |                          ^ expected `)`

error: aborting due to 2 previous errors

