error[E0592]: duplicate definitions with name `dummy`
  --> $DIR/coherence-overlap-downstream-inherent.rs:17:26
   |
17 | impl<T:Sugar> Sweet<T> { fn dummy(&self) { } }
   |                          ^^^^^^^^^^^^^^^^^^^ duplicate definitions for `dummy`
18 | //~^ ERROR E0592
19 | impl<T:Fruit> Sweet<T> { fn dummy(&self) { } }
   |                          ------------------- other definition for `dummy`

error[E0592]: duplicate definitions with name `f`
  --> $DIR/coherence-overlap-downstream-inherent.rs:23:38
   |
23 | impl<X, T> A<T, X> where T: Bar<X> { fn f(&self) {} }
   |                                      ^^^^^^^^^^^^^^ duplicate definitions for `f`
24 | //~^ ERROR E0592
25 | impl<X> A<i32, X> { fn f(&self) {} }
   |                     -------------- other definition for `f`
   |
   = note: downstream crates may implement trait `Bar<_>` for type `i32`

error: aborting due to 2 previous errors

If you want more information on this error, try using "rustc --explain E0592"
