error[E0370]: enum discriminant overflowed
  --> $DIR/discrim-overflow-2.rs:27:9
   |
LL |         OhNo, //~ ERROR enum discriminant overflowed [E0370]
   |         ^^^^ overflowed on value after 127i8
   |
   = note: explicitly set `OhNo = -128i8` if that is desired outcome

error[E0370]: enum discriminant overflowed
  --> $DIR/discrim-overflow-2.rs:36:9
   |
LL |         OhNo, //~ ERROR enum discriminant overflowed [E0370]
   |         ^^^^ overflowed on value after 255u8
   |
   = note: explicitly set `OhNo = 0u8` if that is desired outcome

error[E0370]: enum discriminant overflowed
  --> $DIR/discrim-overflow-2.rs:45:9
   |
LL |         OhNo, //~ ERROR enum discriminant overflowed [E0370]
   |         ^^^^ overflowed on value after 32767i16
   |
   = note: explicitly set `OhNo = -32768i16` if that is desired outcome

error[E0370]: enum discriminant overflowed
  --> $DIR/discrim-overflow-2.rs:54:9
   |
LL |         OhNo, //~ ERROR enum discriminant overflowed [E0370]
   |         ^^^^ overflowed on value after 65535u16
   |
   = note: explicitly set `OhNo = 0u16` if that is desired outcome

error[E0370]: enum discriminant overflowed
  --> $DIR/discrim-overflow-2.rs:63:9
   |
LL |         OhNo, //~ ERROR enum discriminant overflowed [E0370]
   |         ^^^^ overflowed on value after 2147483647i32
   |
   = note: explicitly set `OhNo = -2147483648i32` if that is desired outcome

error[E0370]: enum discriminant overflowed
  --> $DIR/discrim-overflow-2.rs:72:9
   |
LL |         OhNo, //~ ERROR enum discriminant overflowed [E0370]
   |         ^^^^ overflowed on value after 4294967295u32
   |
   = note: explicitly set `OhNo = 0u32` if that is desired outcome

error[E0370]: enum discriminant overflowed
  --> $DIR/discrim-overflow-2.rs:81:9
   |
LL |         OhNo, //~ ERROR enum discriminant overflowed [E0370]
   |         ^^^^ overflowed on value after 9223372036854775807i64
   |
   = note: explicitly set `OhNo = -9223372036854775808i64` if that is desired outcome

error[E0370]: enum discriminant overflowed
  --> $DIR/discrim-overflow-2.rs:90:9
   |
LL |         OhNo, //~ ERROR enum discriminant overflowed [E0370]
   |         ^^^^ overflowed on value after 18446744073709551615u64
   |
   = note: explicitly set `OhNo = 0u64` if that is desired outcome

error: aborting due to 8 previous errors

If you want more information on this error, try using "rustc --explain E0370"
