error[E0252]: the name `foo` is defined multiple times
  --> $DIR/double-import.rs:23:5
   |
22 | use sub1::foo;
   |     --------- previous import of the value `foo` here
23 | use sub2::foo; //~ ERROR the name `foo` is defined multiple times
   |     ^^^^^^^^^ `foo` reimported here
   |
   = note: `foo` must be defined only once in the value namespace of this module
help: You can use `as` to change the binding name of the import
   |
23 | use sub2::foo as Otherfoo; //~ ERROR the name `foo` is defined multiple times
   |     ^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

