warning: `#[must_use]` on methods is experimental (see issue #43302)
  --> $DIR/feature-gate-fn_must_use.rs:16:5
   |
LL |     #[must_use] //~ WARN `#[must_use]` on methods is experimental
   |     ^^^^^^^^^^^
   |
   = help: add #![feature(fn_must_use)] to the crate attributes to enable

warning: `#[must_use]` on functions is experimental (see issue #43302)
  --> $DIR/feature-gate-fn_must_use.rs:20:1
   |
LL | #[must_use] //~ WARN `#[must_use]` on functions is experimental
   | ^^^^^^^^^^^
   |
   = help: add #![feature(fn_must_use)] to the crate attributes to enable

error: compilation successful
  --> $DIR/feature-gate-fn_must_use.rs:31:1
   |
LL | fn main() {} //~ ERROR compilation successful
   | ^^^^^^^^^^^^

error: aborting due to previous error

