error[E0658]: 128-bit type is unstable (see issue #35118)
  --> $DIR/feature-gate-i128_type2.rs:13:15
   |
LL | fn test1() -> i128 { //~ ERROR 128-bit type is unstable
   |               ^^^^
   |
   = help: add #![feature(i128_type)] to the crate attributes to enable

error[E0658]: 128-bit type is unstable (see issue #35118)
  --> $DIR/feature-gate-i128_type2.rs:17:17
   |
LL | fn test1_2() -> u128 { //~ ERROR 128-bit type is unstable
   |                 ^^^^
   |
   = help: add #![feature(i128_type)] to the crate attributes to enable

error[E0658]: 128-bit type is unstable (see issue #35118)
  --> $DIR/feature-gate-i128_type2.rs:22:12
   |
LL |     let x: i128 = 0; //~ ERROR 128-bit type is unstable
   |            ^^^^
   |
   = help: add #![feature(i128_type)] to the crate attributes to enable

error[E0658]: 128-bit type is unstable (see issue #35118)
  --> $DIR/feature-gate-i128_type2.rs:26:12
   |
LL |     let x: u128 = 0; //~ ERROR 128-bit type is unstable
   |            ^^^^
   |
   = help: add #![feature(i128_type)] to the crate attributes to enable

error[E0658]: repr with 128-bit type is unstable (see issue #35118)
  --> $DIR/feature-gate-i128_type2.rs:30:1
   |
LL | / enum A { //~ ERROR 128-bit type is unstable
LL | |     A(u64)
LL | | }
   | |_^
   |
   = help: add #![feature(repr128)] to the crate attributes to enable

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0658`.
