error: `concat_idents` is not stable enough for use and is subject to change (see issue #29599)
  --> $DIR/feature-gate-concat_idents3.rs:17:20
   |
17 |     assert_eq!(10, concat_idents!(X, Y_1)); //~ ERROR `concat_idents` is not stable
   |                    ^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(concat_idents)] to the crate attributes to enable

error: `concat_idents` is not stable enough for use and is subject to change (see issue #29599)
  --> $DIR/feature-gate-concat_idents3.rs:18:20
   |
18 |     assert_eq!(20, concat_idents!(X, Y_2)); //~ ERROR `concat_idents` is not stable
   |                    ^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(concat_idents)] to the crate attributes to enable

error: aborting due to 2 previous errors

