error[E0658]: multiple patterns in `if let` and `while let` are unstable (see issue #48215)
  --> $DIR/feature-gate-if_while_or_patterns.rs:12:5
   |
12 | /     if let 0 | 1 = 0 { //~ ERROR multiple patterns in `if let` and `while let` are unstable
13 | |         ;
14 | |     }
   | |_____^
   |
   = help: add #![feature(if_while_or_patterns)] to the crate attributes to enable

error[E0658]: multiple patterns in `if let` and `while let` are unstable (see issue #48215)
  --> $DIR/feature-gate-if_while_or_patterns.rs:15:5
   |
15 | /     while let 0 | 1 = 1 { //~ ERROR multiple patterns in `if let` and `while let` are unstable
16 | |         break;
17 | |     }
   | |_____^
   |
   = help: add #![feature(if_while_or_patterns)] to the crate attributes to enable

error: aborting due to 2 previous errors

