error: use of SIMD type `LocalSimd` in FFI is highly experimental and may result in invalid code
  --> $DIR/feature-gate-simd-ffi.rs:20:17
   |
20 |     fn baz() -> LocalSimd; //~ ERROR use of SIMD type
   |                 ^^^^^^^^^
   |
   = help: add #![feature(simd_ffi)] to the crate attributes to enable

error: use of SIMD type `LocalSimd` in FFI is highly experimental and may result in invalid code
  --> $DIR/feature-gate-simd-ffi.rs:21:15
   |
21 |     fn qux(x: LocalSimd); //~ ERROR use of SIMD type
   |               ^^^^^^^^^
   |
   = help: add #![feature(simd_ffi)] to the crate attributes to enable

error: aborting due to 2 previous errors

