error[E0379]: trait fns cannot be declared const
  --> $DIR/feature-gate-const_fn.rs:16:5
   |
16 |     const fn foo() -> u32; //~ ERROR const fn is unstable
   |     ^^^^^ trait fns cannot be const

error[E0379]: trait fns cannot be declared const
  --> $DIR/feature-gate-const_fn.rs:18:5
   |
18 |     const fn bar() -> u32 { 0 } //~ ERROR const fn is unstable
   |     ^^^^^ trait fns cannot be const

error[E0379]: trait fns cannot be declared const
  --> $DIR/feature-gate-const_fn.rs:27:5
   |
27 |     const fn foo() -> u32 { 0 } //~ ERROR const fn is unstable
   |     ^^^^^ trait fns cannot be const

error: const fn is unstable (see issue #24111)
  --> $DIR/feature-gate-const_fn.rs:13:1
   |
13 | const fn foo() -> usize { 0 } //~ ERROR const fn is unstable
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(const_fn)] to the crate attributes to enable

error: const fn is unstable (see issue #24111)
  --> $DIR/feature-gate-const_fn.rs:16:5
   |
16 |     const fn foo() -> u32; //~ ERROR const fn is unstable
   |     ^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(const_fn)] to the crate attributes to enable

error: const fn is unstable (see issue #24111)
  --> $DIR/feature-gate-const_fn.rs:18:5
   |
18 |     const fn bar() -> u32 { 0 } //~ ERROR const fn is unstable
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(const_fn)] to the crate attributes to enable

error: const fn is unstable (see issue #24111)
  --> $DIR/feature-gate-const_fn.rs:23:5
   |
23 |     const fn baz() -> u32 { 0 } //~ ERROR const fn is unstable
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(const_fn)] to the crate attributes to enable

error: const fn is unstable (see issue #24111)
  --> $DIR/feature-gate-const_fn.rs:27:5
   |
27 |     const fn foo() -> u32 { 0 } //~ ERROR const fn is unstable
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(const_fn)] to the crate attributes to enable

error: aborting due to 8 previous errors

