error[E0658]: 128-bit type is unstable (see issue #35118)
  --> $DIR/feature-gate-i128_type2.rs:13:15
   |
13 | fn test1() -> i128 { //~ ERROR 128-bit type is unstable
   |               ^^^^
   |
   = help: add #![feature(i128_type)] to the crate attributes to enable

error[E0658]: 128-bit type is unstable (see issue #35118)
  --> $DIR/feature-gate-i128_type2.rs:17:17
   |
17 | fn test1_2() -> u128 { //~ ERROR 128-bit type is unstable
   |                 ^^^^
   |
   = help: add #![feature(i128_type)] to the crate attributes to enable

error[E0658]: 128-bit type is unstable (see issue #35118)
  --> $DIR/feature-gate-i128_type2.rs:22:12
   |
22 |     let x: i128 = 0; //~ ERROR 128-bit type is unstable
   |            ^^^^
   |
   = help: add #![feature(i128_type)] to the crate attributes to enable

error[E0658]: 128-bit type is unstable (see issue #35118)
  --> $DIR/feature-gate-i128_type2.rs:26:12
   |
26 |     let x: u128 = 0; //~ ERROR 128-bit type is unstable
   |            ^^^^
   |
   = help: add #![feature(i128_type)] to the crate attributes to enable

error[E0601]: main function not found

error[E0658]: repr with 128-bit type is unstable (see issue #35118)
  --> $DIR/feature-gate-i128_type2.rs:30:1
   |
30 | / enum A { //~ ERROR 128-bit type is unstable
31 | |     A(u64)
32 | | }
   | |_^
   |
   = help: add #![feature(repr128)] to the crate attributes to enable

error: aborting due to 6 previous errors

