error[E0658]: The `!` type is experimental (see issue #35121)
  --> $DIR/feature-gate-never_type.rs:17:17
   |
17 | type Ma = (u32, !, i32); //~ ERROR type is experimental
   |                 ^
   |
   = help: add #![feature(never_type)] to the crate attributes to enable

error[E0658]: The `!` type is experimental (see issue #35121)
  --> $DIR/feature-gate-never_type.rs:18:20
   |
18 | type Meeshka = Vec<!>; //~ ERROR type is experimental
   |                    ^
   |
   = help: add #![feature(never_type)] to the crate attributes to enable

error[E0658]: The `!` type is experimental (see issue #35121)
  --> $DIR/feature-gate-never_type.rs:19:16
   |
19 | type Mow = &fn(!) -> !; //~ ERROR type is experimental
   |                ^
   |
   = help: add #![feature(never_type)] to the crate attributes to enable

error[E0658]: The `!` type is experimental (see issue #35121)
  --> $DIR/feature-gate-never_type.rs:20:19
   |
20 | type Skwoz = &mut !; //~ ERROR type is experimental
   |                   ^
   |
   = help: add #![feature(never_type)] to the crate attributes to enable

error[E0658]: The `!` type is experimental (see issue #35121)
  --> $DIR/feature-gate-never_type.rs:23:16
   |
23 |     type Wub = !; //~ ERROR type is experimental
   |                ^
   |
   = help: add #![feature(never_type)] to the crate attributes to enable

error: aborting due to 5 previous errors

