error[E0658]: intrinsics are subject to change
  --> $DIR/feature-gate-intrinsics.rs:11:1
   |
11 | / extern "rust-intrinsic" {   //~ ERROR intrinsics are subject to change
12 | |     fn bar();
13 | | }
   | |_^
   |
   = help: add #![feature(intrinsics)] to the crate attributes to enable

error[E0658]: intrinsics are subject to change
  --> $DIR/feature-gate-intrinsics.rs:15:1
   |
15 | / extern "rust-intrinsic" fn baz() {  //~ ERROR intrinsics are subject to change
16 | | }
   | |_^
   |
   = help: add #![feature(intrinsics)] to the crate attributes to enable

error: aborting due to 2 previous errors

If you want more information on this error, try using "rustc --explain E0658"
