error[E0586]: inclusive range with no end
  --> $DIR/impossible_range.rs:21:8
   |
21 |     ..=; //~ERROR inclusive range with no end
   |        ^
   |
   = help: inclusive ranges must be bounded at the end (`..=b` or `a..=b`)

error[E0586]: inclusive range with no end
  --> $DIR/impossible_range.rs:22:9
   |
22 |     0..=; //~ERROR inclusive range with no end
   |         ^
   |
   = help: inclusive ranges must be bounded at the end (`..=b` or `a..=b`)

error: aborting due to 2 previous errors

