error[E0668]: malformed inline assembly
  --> $DIR/inline-asm-bad-constraint.rs:31:9
   |
LL |         asm!("" :"={rax"(rax)) //~ ERROR E0668
   |         ^^^^^^^^^^^^^^^^^^^^^^

error[E0668]: malformed inline assembly
  --> $DIR/inline-asm-bad-constraint.rs:39:9
   |
LL |         asm!("callq $0" : : "0"(foo)) //~ ERROR E0668
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0668]: malformed inline assembly
  --> $DIR/inline-asm-bad-constraint.rs:46:9
   |
LL |         asm!("addb $1, $0" : "={rax}"((0i32, rax))); //~ ERROR E0668
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0668`.
