error[E0308]: method not compatible with trait
  --> $DIR/issue-37884.rs:16:5
   |
16 | /     fn next(&'a mut self) -> Option<Self::Item>
17 | |     //~^ ERROR method not compatible with trait
18 | |     //~| lifetime mismatch
19 | |     {
20 | |         Some(&mut self.0)
21 | |     }
   | |_____^ lifetime mismatch
   |
   = note: expected type `fn(&mut RepeatMut<'a, T>) -> std::option::Option<&mut T>`
              found type `fn(&'a mut RepeatMut<'a, T>) -> std::option::Option<&mut T>`
note: the anonymous lifetime #1 defined on the method body at 16:5...
  --> $DIR/issue-37884.rs:16:5
   |
16 | /     fn next(&'a mut self) -> Option<Self::Item>
17 | |     //~^ ERROR method not compatible with trait
18 | |     //~| lifetime mismatch
19 | |     {
20 | |         Some(&mut self.0)
21 | |     }
   | |_____^
note: ...does not necessarily outlive the lifetime 'a as defined on the impl at 13:1
  --> $DIR/issue-37884.rs:13:1
   |
13 | / impl<'a, T: 'a> Iterator for RepeatMut<'a, T> {
14 | |
15 | |     type Item = &'a mut T;
16 | |     fn next(&'a mut self) -> Option<Self::Item>
...  |
21 | |     }
22 | | }
   | |_^

error: aborting due to previous error

