error[E0618]: expected function, found `i32`
  --> $DIR/issue-10969.rs:12:5
   |
12 |     i(); //~ERROR expected function, found `i32`
   |     ^^^
   |
note: defined here
  --> $DIR/issue-10969.rs:11:9
   |
11 | fn func(i: i32) {
   |         ^

error[E0618]: expected function, found `i32`
  --> $DIR/issue-10969.rs:16:5
   |
16 |     i(); //~ERROR expected function, found `i32`
   |     ^^^
   |
note: defined here
  --> $DIR/issue-10969.rs:15:9
   |
15 |     let i = 0i32;
   |         ^

error: aborting due to 2 previous errors

