warning: not reporting region error due to nll
  --> $DIR/issue-16683.rs:14:9
   |
LL |         self.a(); //~ ERROR cannot infer
   |         ^^^^

warning: not reporting region error due to nll
  --> $DIR/issue-16683.rs:14:14
   |
LL |         self.a(); //~ ERROR cannot infer
   |              ^

error: borrowed data escapes outside of closure
  --> $DIR/issue-16683.rs:14:9
   |
LL |     fn b(&self) {
   |          ----- `self` is a reference that is only valid in the closure body
LL |         self.a(); //~ ERROR cannot infer
   |         ^^^^^^^^ `self` escapes the closure body here

error: aborting due to previous error

