warning: not reporting region error due to nll
  --> $DIR/issue-14285.rs:22:7
   |
LL |     B(a)    //~ ERROR 22:5: 22:9: explicit lifetime required in the type of `a` [E0621]
   |       ^

error[E0621]: explicit lifetime required in the type of `a`
  --> $DIR/issue-14285.rs:22:7
   |
LL | fn foo<'a>(a: &Foo) -> B<'a> {
   |            - consider changing the type of `a` to `&'a (dyn Foo + 'a)`
LL |     B(a)    //~ ERROR 22:5: 22:9: explicit lifetime required in the type of `a` [E0621]
   |       ^ lifetime `'a` required

error: aborting due to previous error

For more information about this error, try `rustc --explain E0621`.
