error: `derive` may only be applied to structs, enums and unions
  --> $DIR/issue-43023.rs:14:5
   |
LL |     #[derive(Debug)] //~ ERROR `derive` may only be applied to structs, enums and unions
   |     ^^^^^^^^^^^^^^^^

error: `derive` may only be applied to structs, enums and unions
  --> $DIR/issue-43023.rs:21:5
   |
LL |     #[derive(Debug)] //~ ERROR `derive` may only be applied to structs, enums and unions
   |     ^^^^^^^^^^^^^^^^

error: `derive` may only be applied to structs, enums and unions
  --> $DIR/issue-43023.rs:26:5
   |
LL |     #[derive(Debug)] //~ ERROR `derive` may only be applied to structs, enums and unions
   |     ^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

