error[E0271]: type mismatch resolving `<std::collections::hash_map::Iter<'_, _, _> as std::iter::Iterator>::Item == &_`
  --> $DIR/issue-33941.rs:14:36
   |
14 |     for _ in HashMap::new().iter().cloned() {} //~ ERROR type mismatch
   |                                    ^^^^^^ expected tuple, found reference
   |
   = note: expected type `(&_, &_)`
              found type `&_`

error[E0271]: type mismatch resolving `<std::collections::hash_map::Iter<'_, _, _> as std::iter::Iterator>::Item == &_`
  --> $DIR/issue-33941.rs:14:5
   |
14 |     for _ in HashMap::new().iter().cloned() {} //~ ERROR type mismatch
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected tuple, found reference
   |
   = note: expected type `(&_, &_)`
              found type `&_`
   = note: required because of the requirements on the impl of `std::iter::Iterator` for `std::iter::Cloned<std::collections::hash_map::Iter<'_, _, _>>`

error: aborting due to 2 previous errors

