error[E0391]: cyclic dependency detected
  --> $DIR/issue-23302-1.rs:14:9
   |
14 |     A = X::A as isize, //~ ERROR E0391
   |         ^^^^^^^^^^^^^ cyclic reference
   |
note: the cycle begins when const-evaluating `X::A::{{initializer}}`...
  --> $DIR/issue-23302-1.rs:14:5
   |
14 |     A = X::A as isize, //~ ERROR E0391
   |     ^^^^^^^^^^^^^^^^^
   = note: ...which then again requires const-evaluating `X::A::{{initializer}}`, completing the cycle.

error: aborting due to previous error

If you want more information on this error, try using "rustc --explain E0391"
