error[E0423]: expected function, found struct `B`
  --> $DIR/issue-42944.rs:9:9
   |
LL |         B(()); //~ ERROR expected function, found struct `B` [E0423]
   |         ^ constructor is not visible here due to private fields

error[E0425]: cannot find function `B` in this scope
  --> $DIR/issue-42944.rs:15:9
   |
LL |         B(()); //~ ERROR cannot find function `B` in this scope [E0425]
   |         ^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
   |
LL |     use foo::B;
   |

error: aborting due to 2 previous errors

Some errors occurred: E0423, E0425.
For more information about an error, try `rustc --explain E0423`.
