error[E0433]: failed to resolve. Not a module `Vec`
  --> $DIR/macro-path-prelude-fail-1.rs:15:9
   |
LL |         Vec::clone!(); //~ ERROR failed to resolve. Not a module `Vec`
   |         ^^^ Not a module `Vec`

error[E0433]: failed to resolve. Not a module `u8`
  --> $DIR/macro-path-prelude-fail-1.rs:16:9
   |
LL |         u8::clone!(); //~ ERROR failed to resolve. Not a module `u8`
   |         ^^ Not a module `u8`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0433`.
