error: expected a macro, found built-in attribute
  --> $DIR/macro-path-prelude-fail-3.rs:13:10
   |
LL | #[derive(inline)] //~ ERROR expected a macro, found built-in attribute
   |          ^^^^^^

error: expected a macro, found built-in attribute
  --> $DIR/macro-path-prelude-fail-3.rs:17:5
   |
LL |     inline!(); //~ ERROR expected a macro, found built-in attribute
   |     ^^^^^^

error: aborting due to 2 previous errors

