error[E0615]: attempted to take value of method `get_x` on type `Point`
  --> $DIR/method-missing-call.rs:32:26
   |
32 |                         .get_x;//~ ERROR attempted to take value of method `get_x` on type `Point`
   |                          ^^^^^
   |
   = help: maybe a `()` to call it is missing?

error[E0615]: attempted to take value of method `filter_map` on type `std::iter::Filter<std::iter::Map<std::slice::Iter<'_, {integer}>, [closure@$DIR/method-missing-call.rs:37:20: 37:25]>, [closure@$DIR/method-missing-call.rs:38:23: 38:35]>`
  --> $DIR/method-missing-call.rs:39:16
   |
39 |               .filter_map; //~ ERROR attempted to take value of method `filter_map` on type
   |                ^^^^^^^^^^
   |
   = help: maybe a `()` to call it is missing?

error: aborting due to 2 previous errors

If you want more information on this error, try using "rustc --explain E0615"
