error[E0405]: cannot find trait `Add` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:21:14
   |
LL |         impl Add for Test {} //~ ERROR cannot find trait `Add` in this scope
   |              ^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
   |
LL |         use std::ops::Add;
   |

error[E0405]: cannot find trait `Clone` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:22:14
   |
LL |         impl Clone for Test {} //~ ERROR cannot find trait `Clone` in this scope
   |              ^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
   |
LL |         use std::clone::Clone;
   |
LL |         use std::prelude::v1::Clone;
   |

error[E0405]: cannot find trait `Iterator` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:23:14
   |
LL |         impl Iterator for Test {} //~ ERROR cannot find trait `Iterator` in this scope
   |              ^^^^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
   |
LL |         use std::iter::Iterator;
   |
LL |         use std::prelude::v1::Iterator;
   |

error[E0405]: cannot find trait `ToString` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:24:14
   |
LL |         impl ToString for Test {} //~ ERROR cannot find trait `ToString` in this scope
   |              ^^^^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
   |
LL |         use std::prelude::v1::ToString;
   |
LL |         use std::string::ToString;
   |

error[E0405]: cannot find trait `Writer` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:25:14
   |
LL |         impl Writer for Test {} //~ ERROR cannot find trait `Writer` in this scope
   |              ^^^^^^ not found in this scope

error[E0425]: cannot find function `drop` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:28:13
   |
LL |             drop(2) //~ ERROR cannot find function `drop` in this scope
   |             ^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
   |
LL |         use std::mem::drop;
   |
LL |         use std::prelude::v1::drop;
   |

error[E0405]: cannot find trait `Add` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:33:10
   |
LL |     impl Add for Test {} //~ ERROR cannot find trait `Add` in this scope
   |          ^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
   |
LL |     use std::ops::Add;
   |

error[E0405]: cannot find trait `Clone` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:34:10
   |
LL |     impl Clone for Test {} //~ ERROR cannot find trait `Clone` in this scope
   |          ^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
   |
LL |     use std::clone::Clone;
   |
LL |     use std::prelude::v1::Clone;
   |

error[E0405]: cannot find trait `Iterator` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:35:10
   |
LL |     impl Iterator for Test {} //~ ERROR cannot find trait `Iterator` in this scope
   |          ^^^^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
   |
LL |     use std::iter::Iterator;
   |
LL |     use std::prelude::v1::Iterator;
   |

error[E0405]: cannot find trait `ToString` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:36:10
   |
LL |     impl ToString for Test {} //~ ERROR cannot find trait `ToString` in this scope
   |          ^^^^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
   |
LL |     use std::prelude::v1::ToString;
   |
LL |     use std::string::ToString;
   |

error[E0405]: cannot find trait `Writer` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:37:10
   |
LL |     impl Writer for Test {} //~ ERROR cannot find trait `Writer` in this scope
   |          ^^^^^^ not found in this scope

error[E0425]: cannot find function `drop` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:40:9
   |
LL |         drop(2) //~ ERROR cannot find function `drop` in this scope
   |         ^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
   |
LL |     use std::mem::drop;
   |
LL |     use std::prelude::v1::drop;
   |

error[E0405]: cannot find trait `Add` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:48:14
   |
LL |         impl Add for Test {} //~ ERROR cannot find trait `Add` in this scope
   |              ^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
   |
LL |         use std::ops::Add;
   |

error[E0405]: cannot find trait `Clone` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:49:14
   |
LL |         impl Clone for Test {} //~ ERROR cannot find trait `Clone` in this scope
   |              ^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
   |
LL |         use std::clone::Clone;
   |
LL |         use std::prelude::v1::Clone;
   |

error[E0405]: cannot find trait `Iterator` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:50:14
   |
LL |         impl Iterator for Test {} //~ ERROR cannot find trait `Iterator` in this scope
   |              ^^^^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
   |
LL |         use std::iter::Iterator;
   |
LL |         use std::prelude::v1::Iterator;
   |

error[E0405]: cannot find trait `ToString` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:51:14
   |
LL |         impl ToString for Test {} //~ ERROR cannot find trait `ToString` in this scope
   |              ^^^^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
   |
LL |         use std::prelude::v1::ToString;
   |
LL |         use std::string::ToString;
   |

error[E0405]: cannot find trait `Writer` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:52:14
   |
LL |         impl Writer for Test {} //~ ERROR cannot find trait `Writer` in this scope
   |              ^^^^^^ not found in this scope

error[E0425]: cannot find function `drop` in this scope
  --> $DIR/no-implicit-prelude-nested.rs:55:13
   |
LL |             drop(2) //~ ERROR cannot find function `drop` in this scope
   |             ^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
   |
LL |         use std::mem::drop;
   |
LL |         use std::prelude::v1::drop;
   |

error: aborting due to 18 previous errors

Some errors occurred: E0405, E0425.
For more information about an error, try `rustc --explain E0405`.
