warning: unnecessary parentheses around `return` value
  --> $DIR/path-lookahead.rs:18:10
   |
18 |   return (<T as ToString>::to_string(&arg)); //~WARN unnecessary parentheses around `return` value
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: #[warn(unused_parens)] on by default

warning: function is never used: `with_parens`
  --> $DIR/path-lookahead.rs:17:1
   |
17 | / fn with_parens<T: ToString>(arg: T) -> String { //~WARN function is never used: `with_parens`
18 | |   return (<T as ToString>::to_string(&arg)); //~WARN unnecessary parentheses around `return` value
19 | | }
   | |_^
   |
note: lint level defined here
  --> $DIR/path-lookahead.rs:13:9
   |
13 | #![warn(unused)]
   |         ^^^^^^
   = note: #[warn(dead_code)] implied by #[warn(unused)]

warning: function is never used: `no_parens`
  --> $DIR/path-lookahead.rs:21:1
   |
21 | / fn no_parens<T: ToString>(arg: T) -> String { //~WARN function is never used: `no_parens`
22 | |   return <T as ToString>::to_string(&arg);
23 | | }
   | |_^

