error[E0107]: wrong number of type arguments: expected 0, found 1
  --> $DIR/seq-args.rs:14:13
   |
LL | impl<T> seq<T> for Vec<T> { //~ ERROR wrong number of type arguments
   |             ^ unexpected type argument

error[E0107]: wrong number of type arguments: expected 0, found 1
  --> $DIR/seq-args.rs:17:10
   |
LL | impl seq<bool> for u32 { //~ ERROR wrong number of type arguments
   |          ^^^^ unexpected type argument

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0107`.
