error[E0507]: cannot move out of borrowed content
  --> $DIR/std-uncopyable-atomics.rs:19:13
   |
LL |     let x = *&x; //~ ERROR: cannot move out of borrowed content
   |             ^^^
   |             |
   |             cannot move out of borrowed content
   |             help: consider removing the `*`: `&x`

error[E0507]: cannot move out of borrowed content
  --> $DIR/std-uncopyable-atomics.rs:21:13
   |
LL |     let x = *&x; //~ ERROR: cannot move out of borrowed content
   |             ^^^
   |             |
   |             cannot move out of borrowed content
   |             help: consider removing the `*`: `&x`

error[E0507]: cannot move out of borrowed content
  --> $DIR/std-uncopyable-atomics.rs:23:13
   |
LL |     let x = *&x; //~ ERROR: cannot move out of borrowed content
   |             ^^^
   |             |
   |             cannot move out of borrowed content
   |             help: consider removing the `*`: `&x`

error[E0507]: cannot move out of borrowed content
  --> $DIR/std-uncopyable-atomics.rs:25:13
   |
LL |     let x = *&x; //~ ERROR: cannot move out of borrowed content
   |             ^^^
   |             |
   |             cannot move out of borrowed content
   |             help: consider removing the `*`: `&x`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0507`.
