error[E0658]: attributes on expressions are experimental. (see issue #15701)
  --> $DIR/stmt_expr_attrs_no_feature.rs:23:5
   |
LL |     #[attr] //~ ERROR attributes on expressions are experimental
   |     ^^^^^^^
   |
   = help: add #![feature(stmt_expr_attributes)] to the crate attributes to enable

error[E0658]: attributes on expressions are experimental. (see issue #15701)
  --> $DIR/stmt_expr_attrs_no_feature.rs:104:18
   |
LL |     fn y(a: [u8; #[attr] 5]); //~ ERROR 15701
   |                  ^^^^^^^
   |
   = help: add #![feature(stmt_expr_attributes)] to the crate attributes to enable

error[E0658]: attributes on expressions are experimental. (see issue #15701)
  --> $DIR/stmt_expr_attrs_no_feature.rs:111:19
   |
LL |     const Y: u8 = #[attr] 5; //~ ERROR 15701
   |                   ^^^^^^^
   |
   = help: add #![feature(stmt_expr_attributes)] to the crate attributes to enable

error[E0658]: attributes on expressions are experimental. (see issue #15701)
  --> $DIR/stmt_expr_attrs_no_feature.rs:117:19
   |
LL |     const Y: [u8; #[attr] 5]; //~ ERROR 15701
   |                   ^^^^^^^
   |
   = help: add #![feature(stmt_expr_attributes)] to the crate attributes to enable

error[E0658]: attributes on expressions are experimental. (see issue #15701)
  --> $DIR/stmt_expr_attrs_no_feature.rs:123:18
   |
LL |     field2: [u8; #[attr] 5] //~ ERROR 15701
   |                  ^^^^^^^
   |
   = help: add #![feature(stmt_expr_attributes)] to the crate attributes to enable

error[E0658]: attributes on expressions are experimental. (see issue #15701)
  --> $DIR/stmt_expr_attrs_no_feature.rs:128:10
   |
LL |     [u8; #[attr] 5] //~ ERROR 15701
   |          ^^^^^^^
   |
   = help: add #![feature(stmt_expr_attributes)] to the crate attributes to enable

error[E0658]: attributes on expressions are experimental. (see issue #15701)
  --> $DIR/stmt_expr_attrs_no_feature.rs:134:14
   |
LL |         [u8; #[attr] 5] //~ ERROR 15701
   |              ^^^^^^^
   |
   = help: add #![feature(stmt_expr_attributes)] to the crate attributes to enable

error[E0658]: attributes on expressions are experimental. (see issue #15701)
  --> $DIR/stmt_expr_attrs_no_feature.rs:139:22
   |
LL |         field2: [u8; #[attr] 5] //~ ERROR 15701
   |                      ^^^^^^^
   |
   = help: add #![feature(stmt_expr_attributes)] to the crate attributes to enable

error[E0658]: attributes on expressions are experimental. (see issue #15701)
  --> $DIR/stmt_expr_attrs_no_feature.rs:147:14
   |
LL |         6 => #[attr] (), //~ ERROR 15701
   |              ^^^^^^^
   |
   = help: add #![feature(stmt_expr_attributes)] to the crate attributes to enable

error: aborting due to 9 previous errors

For more information about this error, try `rustc --explain E0658`.
