error[E0308]: mismatched types
  --> $DIR/str-lit-type-mismatch.rs:13:20
   |
13 |     let x: &[u8] = "foo"; //~ ERROR mismatched types
   |                    ^^^^^
   |                    |
   |                    expected slice, found str
   |                    help: consider adding a leading `b`: `b"foo"`
   |
   = note: expected type `&[u8]`
              found type `&'static str`

error[E0308]: mismatched types
  --> $DIR/str-lit-type-mismatch.rs:14:23
   |
14 |     let y: &[u8; 4] = "baaa"; //~ ERROR mismatched types
   |                       ^^^^^^
   |                       |
   |                       expected array of 4 elements, found str
   |                       help: consider adding a leading `b`: `b"baaa"`
   |
   = note: expected type `&[u8; 4]`
              found type `&'static str`

error[E0308]: mismatched types
  --> $DIR/str-lit-type-mismatch.rs:15:19
   |
15 |     let z: &str = b"foo"; //~ ERROR mismatched types
   |                   ^^^^^^
   |                   |
   |                   expected str, found array of 3 elements
   |                   help: consider removing the leading `b`: `"foo"`
   |
   = note: expected type `&str`
              found type `&'static [u8; 3]`

error: aborting due to 3 previous errors

If you want more information on this error, try using "rustc --explain E0308"
