error[E0412]: cannot find type `B` in this scope
 --> $DIR/ui-testing-optout.rs:4:10
  |
4 | type A = B; //~ ERROR
  |          ^ help: a type alias with a similar name exists: `A`

error[E0412]: cannot find type `D` in this scope
  --> $DIR/ui-testing-optout.rs:10:10
   |
10 | type C = D; //~ ERROR
   |          ^ help: a type alias with a similar name exists: `A`

error[E0412]: cannot find type `F` in this scope
  --> $DIR/ui-testing-optout.rs:95:10
   |
95 | type E = F; //~ ERROR
   |          ^ help: a type alias with a similar name exists: `A`

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0412`.
