error: attribute must be applied to a `static` variable
  --> $DIR/used.rs:16:1
   |
LL | #[used] //~ ERROR attribute must be applied to a `static` variable
   | ^^^^^^^

error: attribute must be applied to a `static` variable
  --> $DIR/used.rs:19:1
   |
LL | #[used] //~ ERROR attribute must be applied to a `static` variable
   | ^^^^^^^

error: attribute must be applied to a `static` variable
  --> $DIR/used.rs:22:1
   |
LL | #[used] //~ ERROR attribute must be applied to a `static` variable
   | ^^^^^^^

error: attribute must be applied to a `static` variable
  --> $DIR/used.rs:25:1
   |
LL | #[used] //~ ERROR attribute must be applied to a `static` variable
   | ^^^^^^^

error: aborting due to 4 previous errors

