error: attribute must be applied to a `static` variable
  --> $DIR/used.rs:14:1
   |
LL | #[used] //~ ERROR attribute must be applied to a `static` variable
   | ^^^^^^^

error: attribute must be applied to a `static` variable
  --> $DIR/used.rs:17:1
   |
LL | #[used] //~ ERROR attribute must be applied to a `static` variable
   | ^^^^^^^

error: attribute must be applied to a `static` variable
  --> $DIR/used.rs:20:1
   |
LL | #[used] //~ ERROR attribute must be applied to a `static` variable
   | ^^^^^^^

error: attribute must be applied to a `static` variable
  --> $DIR/used.rs:23:1
   |
LL | #[used] //~ ERROR attribute must be applied to a `static` variable
   | ^^^^^^^

error: aborting due to 4 previous errors

