error[E0601]: `main` function not found in crate `wasm_custom_section_relocations`
   |
   = note: consider adding a `main` function to `$DIR/wasm-custom-section-relocations.rs`

error: statics with a custom `#[link_section]` must be a simple list of bytes on the wasm target with no extra levels of indirection such as references
  --> $DIR/wasm-custom-section-relocations.rs:14:1
   |
LL | pub static A: &[u8] = &[1]; //~ ERROR: no extra levels of indirection
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: statics with a custom `#[link_section]` must be a simple list of bytes on the wasm target with no extra levels of indirection such as references
  --> $DIR/wasm-custom-section-relocations.rs:23:1
   |
LL | pub static D: &usize = &C; //~ ERROR: no extra levels of indirection
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0601`.
