warning: constant evaluation error
  --> $DIR/issue-44578.rs:35:20
   |
LL |     println!("{}", <Bar<u16, u8> as Foo>::AMT); //~ WARN const_err
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^ referenced constant has errors
   |
   = note: #[warn(const_err)] on by default

warning: constant evaluation error
  --> $DIR/issue-44578.rs:35:20
   |
LL |     println!("{}", <Bar<u16, u8> as Foo>::AMT); //~ WARN const_err
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^ referenced constant has errors

