error[E0018]: raw pointers cannot be cast to integers in constants
  --> $DIR/match-test-ptr-null.rs:16:15
   |
LL |         match &1 as *const i32 as usize { //~ ERROR raw pointers cannot be cast to integers
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0019]: constant contains unimplemented expression type
  --> $DIR/match-test-ptr-null.rs:17:13
   |
LL |             0 => 42, //~ ERROR constant contains unimplemented expression type
   |             ^

error[E0080]: could not evaluate repeat length
  --> $DIR/match-test-ptr-null.rs:15:26
   |
LL |       let _: [u8; 0] = [4; { //~ ERROR could not evaluate repeat length
   |  __________________________^
LL | |         match &1 as *const i32 as usize { //~ ERROR raw pointers cannot be cast to integers
LL | |             0 => 42, //~ ERROR constant contains unimplemented expression type
   | |             - "pointer arithmetic or comparison" needs an rfc before being allowed inside constants
LL | |             //~^ NOTE "pointer arithmetic or comparison" needs an rfc before being allowed
LL | |             n => n,
LL | |         }
LL | |     }];
   | |_____^

error: aborting due to 3 previous errors

Some errors occurred: E0018, E0019, E0080.
For more information about an error, try `rustc --explain E0018`.
