error[E0658]: casting pointers to integers in constants is unstable (see issue #51910)
  --> $DIR/match-test-ptr-null.rs:16:15
   |
LL |         match &1 as *const i32 as usize { //~ ERROR casting pointers to integers in constants
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(const_raw_ptr_to_usize_cast)] to the crate attributes to enable

error[E0019]: constant contains unimplemented expression type
  --> $DIR/match-test-ptr-null.rs:17:13
   |
LL |             0 => 42, //~ ERROR constant contains unimplemented expression type
   |             ^

error[E0080]: could not evaluate repeat length
  --> $DIR/match-test-ptr-null.rs:15:26
   |
LL |       let _: [u8; 0] = [4; { //~ ERROR could not evaluate repeat length
   |  __________________________^
LL | |         match &1 as *const i32 as usize { //~ ERROR casting pointers to integers in constants
LL | |             0 => 42, //~ ERROR constant contains unimplemented expression type
   | |             - "pointer arithmetic or comparison" needs an rfc before being allowed inside constants
LL | |             //~^ NOTE "pointer arithmetic or comparison" needs an rfc before being allowed
LL | |             n => n,
LL | |         }
LL | |     }];
   | |_____^

error: aborting due to 3 previous errors

Some errors occurred: E0019, E0080, E0658.
For more information about an error, try `rustc --explain E0019`.
