error[E0687]: lifetimes used in `fn` or `Fn` syntax must be explicitly declared using `<...>` binders
  --> $DIR/E0687_where.rs:14:31
   |
14 | fn bar<F>(x: &F) where F: Fn(&'a u32) {} //~ ERROR must be explicitly
   |                               ^^ in-band lifetime definition

error[E0687]: lifetimes used in `fn` or `Fn` syntax must be explicitly declared using `<...>` binders
  --> $DIR/E0687_where.rs:16:21
   |
16 | fn baz(x: &impl Fn(&'a u32)) {} //~ ERROR must be explicitly
   |                     ^^ in-band lifetime definition

error: aborting due to 2 previous errors

