error[E0496]: lifetime name `'s` shadows a lifetime name that is already in scope
  --> $DIR/shadow.rs:17:12
   |
16 | impl Foo<&'s u8> {
   |           -- first declared here
17 |     fn bar<'s>(&self, x: &'s u8) {} //~ ERROR shadows a lifetime name
   |            ^^ lifetime 's already in scope

error[E0496]: lifetime name `'s` shadows a lifetime name that is already in scope
  --> $DIR/shadow.rs:18:19
   |
16 | impl Foo<&'s u8> {
   |           -- first declared here
17 |     fn bar<'s>(&self, x: &'s u8) {} //~ ERROR shadows a lifetime name
18 |     fn baz(x: for<'s> fn(&'s u32)) {} //~ ERROR shadows a lifetime name
   |                   ^^ lifetime 's already in scope

error: aborting due to 2 previous errors

If you want more information on this error, try using "rustc --explain E0496"
