error: lifetime name `'x` only used once
  --> $DIR/single_use_lifetimes-4.rs:12:12
   |
12 | struct Foo<'x> { //~ ERROR lifetime name `'x` only used once
   |            ^^
   |
note: lint level defined here
  --> $DIR/single_use_lifetimes-4.rs:10:9
   |
10 | #![deny(single_use_lifetime)]
   |         ^^^^^^^^^^^^^^^^^^^

error: lifetime name `'x` only used once
  --> $DIR/single_use_lifetimes-4.rs:16:10
   |
16 | enum Bar<'x> { //~ ERROR lifetime name `'x` only used once
   |          ^^

error: aborting due to 2 previous errors

