error: unreachable pattern
  --> $DIR/uninhabited-patterns.rs:37:9
   |
LL |         &[..] => (),    //~ ERROR unreachable pattern
   |         ^^^^^
   |
note: lint level defined here
  --> $DIR/uninhabited-patterns.rs:16:9
   |
LL | #![deny(unreachable_patterns)]
   |         ^^^^^^^^^^^^^^^^^^^^

error: unreachable pattern
  --> $DIR/uninhabited-patterns.rs:42:9
   |
LL |         Ok(box _) => (),    //~ ERROR unreachable pattern
   |         ^^^^^^^^^

error: unreachable pattern
  --> $DIR/uninhabited-patterns.rs:44:9
   |
LL |         Err(&[..]) => (),   //~ ERROR unreachable pattern
   |         ^^^^^^^^^^

error: unreachable pattern
  --> $DIR/uninhabited-patterns.rs:51:9
   |
LL |         Err(Ok(_y)) => (),  //~ ERROR unreachable pattern
   |         ^^^^^^^^^^^

error: unreachable pattern
  --> $DIR/uninhabited-patterns.rs:54:15
   |
LL |     while let Some(_y) = foo() {
   |               ^^^^^^^^

error: aborting due to 5 previous errors

