warning: not reporting region error due to nll
  --> $DIR/dyn-trait.rs:32:16
   |
LL |     static_val(x); //~ ERROR cannot infer
   |                ^

error: unsatisfied lifetime constraints
  --> $DIR/dyn-trait.rs:32:5
   |
LL |     static_val(x); //~ ERROR cannot infer
   |     ^^^^^^^^^^^^^ argument requires that `'a` must outlive `'static`

error: aborting due to previous error

