error[E0515]: cannot return value referencing function parameter
  --> $DIR/regions-ref-in-fn-arg.rs:15:5
   |
LL | fn arg_item(box ref x: Box<isize>) -> &'static isize {
   |             --------- function parameter borrowed here
LL |     x //~^ ERROR borrowed value does not live long enough
   |     ^ returns a value referencing data owned by the current function

error[E0515]: cannot return value referencing function parameter
  --> $DIR/regions-ref-in-fn-arg.rs:21:22
   |
LL |     with(|box ref x| x) //~ ERROR borrowed value does not live long enough
   |           ---------  ^ returns a value referencing data owned by the current function
   |           |
   |           function parameter borrowed here

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0515`.
