error: unsatisfied lifetime constraints
  --> $DIR/regions-bounded-by-trait-requiring-static.rs:32:5
   |
LL | fn param_not_ok<'a>(x: &'a isize) {
   |                 -- lifetime `'a` defined here
LL |     assert_send::<&'a isize>(); //~ ERROR does not fulfill the required lifetime
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'a` must outlive `'static`

error: unsatisfied lifetime constraints
  --> $DIR/regions-bounded-by-trait-requiring-static.rs:36:5
   |
LL | fn param_not_ok1<'a>(_: &'a isize) {
   |                  -- lifetime `'a` defined here
LL |     assert_send::<&'a str>(); //~ ERROR does not fulfill the required lifetime
   |     ^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'a` must outlive `'static`

error: unsatisfied lifetime constraints
  --> $DIR/regions-bounded-by-trait-requiring-static.rs:40:5
   |
LL | fn param_not_ok2<'a>(_: &'a isize) {
   |                  -- lifetime `'a` defined here
LL |     assert_send::<&'a [isize]>(); //~ ERROR does not fulfill the required lifetime
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'a` must outlive `'static`

error: unsatisfied lifetime constraints
  --> $DIR/regions-bounded-by-trait-requiring-static.rs:54:5
   |
LL | fn box_with_region_not_ok<'a>() {
   |                           -- lifetime `'a` defined here
LL |     assert_send::<Box<&'a isize>>(); //~ ERROR does not fulfill the required lifetime
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'a` must outlive `'static`

error: unsatisfied lifetime constraints
  --> $DIR/regions-bounded-by-trait-requiring-static.rs:65:5
   |
LL | fn unsafe_ok2<'a>(_: &'a isize) {
   |               -- lifetime `'a` defined here
LL |     assert_send::<*const &'a isize>(); //~ ERROR does not fulfill the required lifetime
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'a` must outlive `'static`

error: unsatisfied lifetime constraints
  --> $DIR/regions-bounded-by-trait-requiring-static.rs:69:5
   |
LL | fn unsafe_ok3<'a>(_: &'a isize) {
   |               -- lifetime `'a` defined here
LL |     assert_send::<*mut &'a isize>(); //~ ERROR does not fulfill the required lifetime
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'a` must outlive `'static`

error: aborting due to 6 previous errors

