warning: not reporting region error due to nll
  --> $DIR/regions-proc-bound-capture.rs:19:14
   |
LL |     Box::new(move|| { *x }) //~ ERROR explicit lifetime required in the type of `x` [E0621]
   |              ^^^^^^^^^^^^^

error[E0621]: explicit lifetime required in the type of `x`
  --> $DIR/regions-proc-bound-capture.rs:19:5
   |
LL | fn static_proc(x: &isize) -> Box<FnMut()->(isize) + 'static> {
   |                   ------ help: add explicit lifetime `'static` to the type of `x`: `&'static isize`
LL |     // This is illegal, because the region bound on `proc` is 'static.
LL |     Box::new(move|| { *x }) //~ ERROR explicit lifetime required in the type of `x` [E0621]
   |     ^^^^^^^^^^^^^^^^^^^^^^^ lifetime `'static` required

error: aborting due to previous error

For more information about this error, try `rustc --explain E0621`.
