warning: not reporting region error due to nll
  --> $DIR/region-multiple-lifetime-bounds-on-fns-where-clause.rs:19:10
   |
LL |     *x = *y; //~ ERROR E0623
   |          ^^

warning: not reporting region error due to nll
  --> $DIR/region-multiple-lifetime-bounds-on-fns-where-clause.rs:20:10
   |
LL |     *z = *y; //~ ERROR E0623
   |          ^^

warning: not reporting region error due to nll
  --> $DIR/region-multiple-lifetime-bounds-on-fns-where-clause.rs:26:5
   |
LL |     a(x, y, z); //~ ERROR 26:7: 26:8: lifetime mismatch [E0623]
   |     ^

error[E0308]: mismatched types
  --> $DIR/region-multiple-lifetime-bounds-on-fns-where-clause.rs:32:56
   |
LL |     let _: fn(&mut &isize, &mut &isize, &mut &isize) = a; //~ ERROR mismatched types
   |                                                        ^ expected concrete lifetime, found bound lifetime parameter
   |
   = note: expected type `for<'r, 's, 't0, 't1, 't2, 't3> fn(&'r mut &'s isize, &'t0 mut &'t1 isize, &'t2 mut &'t3 isize)`
              found type `for<'r, 's, 't0> fn(&'r mut &isize, &'s mut &isize, &'t0 mut &isize) {a::<'_, '_, '_>}`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
