warning: not reporting region error due to nll
  --> $DIR/regions-creating-enums4.rs:17:5
   |
LL |     ast::add(x, y) //~ ERROR cannot infer
   |     ^^^^^^^^

error: unsatisfied lifetime constraints
  --> $DIR/regions-creating-enums4.rs:17:14
   |
LL | fn mk_add_bad2<'a,'b>(x: &'a ast<'a>, y: &'a ast<'a>, z: &ast) -> ast<'b> {
   |                -- -- lifetime `'b` defined here
   |                |
   |                lifetime `'a` defined here
LL |     ast::add(x, y) //~ ERROR cannot infer
   |              ^ requires that `'a` must outlive `'b`

error: aborting due to previous error

