error: unsatisfied lifetime constraints
  --> $DIR/regions-free-region-ordering-caller.rs:18:12
   |
LL | fn call2<'a, 'b>(a: &'a usize, b: &'b usize) {
   |          --  -- lifetime `'b` defined here
   |          |
   |          lifetime `'a` defined here
LL |     let z: Option<&'b &'a usize> = None;//~ ERROR E0623
   |            ^^^^^^^^^^^^^^^^^^^^^ type annotation requires that `'a` must outlive `'b`

error: unsatisfied lifetime constraints
  --> $DIR/regions-free-region-ordering-caller.rs:23:12
   |
LL | fn call3<'a, 'b>(a: &'a usize, b: &'b usize) {
   |          --  -- lifetime `'b` defined here
   |          |
   |          lifetime `'a` defined here
LL |     let y: Paramd<'a> = Paramd { x: a };
LL |     let z: Option<&'b Paramd<'a>> = None;//~ ERROR E0623
   |            ^^^^^^^^^^^^^^^^^^^^^^ type annotation requires that `'a` must outlive `'b`

error: unsatisfied lifetime constraints
  --> $DIR/regions-free-region-ordering-caller.rs:27:12
   |
LL | fn call4<'a, 'b>(a: &'a usize, b: &'b usize) {
   |          --  -- lifetime `'b` defined here
   |          |
   |          lifetime `'a` defined here
LL |     let z: Option<&'a &'b usize> = None;//~ ERROR E0623
   |            ^^^^^^^^^^^^^^^^^^^^^ type annotation requires that `'b` must outlive `'a`

error: aborting due to 3 previous errors

