warning: not reporting region error due to nll
  --> $DIR/region-lifetime-bounds-on-fns-where-clause.rs:18:10
   |
LL |     *x = *y; //~ ERROR E0623
   |          ^^

warning: not reporting region error due to nll
  --> $DIR/region-lifetime-bounds-on-fns-where-clause.rs:24:5
   |
LL |     a(x, y); //~ ERROR 24:7: 24:8: lifetime mismatch [E0623]
   |     ^

error[E0308]: mismatched types
  --> $DIR/region-lifetime-bounds-on-fns-where-clause.rs:30:43
   |
LL |     let _: fn(&mut &isize, &mut &isize) = a; //~ ERROR mismatched types
   |                                           ^ expected concrete lifetime, found bound lifetime parameter
   |
   = note: expected type `for<'r, 's, 't0, 't1> fn(&'r mut &'s isize, &'t0 mut &'t1 isize)`
              found type `for<'r, 's> fn(&'r mut &isize, &'s mut &isize) {a::<'_, '_>}`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
