error[E0491]: in type `&'a WithAssoc<TheType<'b>>`, reference has a longer lifetime than the data it references
  --> $DIR/regions-outlives-projection-container.rs:50:12
   |
LL |     let _: &'a WithAssoc<TheType<'b>> = loop { }; //~ ERROR reference has a longer lifetime
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the pointer is valid for the lifetime 'a as defined on the function body at 44:15
  --> $DIR/regions-outlives-projection-container.rs:44:15
   |
LL | fn with_assoc<'a,'b>() {
   |               ^^
note: but the referenced data is only valid for the lifetime 'b as defined on the function body at 44:18
  --> $DIR/regions-outlives-projection-container.rs:44:18
   |
LL | fn with_assoc<'a,'b>() {
   |                  ^^

error[E0491]: in type `&'a WithoutAssoc<TheType<'b>>`, reference has a longer lifetime than the data it references
  --> $DIR/regions-outlives-projection-container.rs:67:12
   |
LL |     let _: &'a WithoutAssoc<TheType<'b>> = loop { }; //~ ERROR reference has a longer lifetime
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the pointer is valid for the lifetime 'a as defined on the function body at 63:18
  --> $DIR/regions-outlives-projection-container.rs:63:18
   |
LL | fn without_assoc<'a,'b>() {
   |                  ^^
note: but the referenced data is only valid for the lifetime 'b as defined on the function body at 63:21
  --> $DIR/regions-outlives-projection-container.rs:63:21
   |
LL | fn without_assoc<'a,'b>() {
   |                     ^^

error[E0491]: in type `&'a WithAssoc<TheType<'b>>`, reference has a longer lifetime than the data it references
  --> $DIR/regions-outlives-projection-container.rs:75:12
   |
LL |     call::<&'a WithAssoc<TheType<'b>>>();
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the pointer is valid for the lifetime 'a as defined on the function body at 70:20
  --> $DIR/regions-outlives-projection-container.rs:70:20
   |
LL | fn call_with_assoc<'a,'b>() {
   |                    ^^
note: but the referenced data is only valid for the lifetime 'b as defined on the function body at 70:23
  --> $DIR/regions-outlives-projection-container.rs:70:23
   |
LL | fn call_with_assoc<'a,'b>() {
   |                       ^^

error[E0491]: in type `&'a WithoutAssoc<TheType<'b>>`, reference has a longer lifetime than the data it references
  --> $DIR/regions-outlives-projection-container.rs:82:12
   |
LL |     call::<&'a WithoutAssoc<TheType<'b>>>(); //~ ERROR reference has a longer lifetime
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the pointer is valid for the lifetime 'a as defined on the function body at 79:23
  --> $DIR/regions-outlives-projection-container.rs:79:23
   |
LL | fn call_without_assoc<'a,'b>() {
   |                       ^^
note: but the referenced data is only valid for the lifetime 'b as defined on the function body at 79:26
  --> $DIR/regions-outlives-projection-container.rs:79:26
   |
LL | fn call_without_assoc<'a,'b>() {
   |                          ^^

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0491`.
