error[E0502]: cannot borrow `v` as immutable because it is also borrowed as mutable
  --> $DIR/regions-adjusted-lvalue-op.rs:24:16
   |
LL |     v[0].oh_no(&v); //~ ERROR cannot borrow `v` as immutable because
   |     -----------^^-
   |     |          |
   |     |          immutable borrow occurs here
   |     mutable borrow occurs here
   |     borrow later used here

error[E0502]: cannot borrow `v` as immutable because it is also borrowed as mutable
  --> $DIR/regions-adjusted-lvalue-op.rs:25:16
   |
LL |     (*v).oh_no(&v); //~ ERROR cannot borrow `v` as immutable because
   |     -----------^^-
   |     | |        |
   |     | |        immutable borrow occurs here
   |     | mutable borrow occurs here
   |     borrow later used here

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0502`.
