warning: not reporting region error due to nll
  --> $DIR/regions-creating-enums3.rs:17:5
   |
LL |     ast::add(x, y) //~ ERROR 17:5: 17:19: lifetime mismatch [E0623]
   |     ^^^^^^^^

error: unsatisfied lifetime constraints
  --> $DIR/regions-creating-enums3.rs:17:14
   |
LL | fn mk_add_bad1<'a,'b>(x: &'a ast<'a>, y: &'b ast<'b>) -> ast<'a> {
   |                -- -- lifetime `'b` defined here
   |                |
   |                lifetime `'a` defined here
LL |     ast::add(x, y) //~ ERROR 17:5: 17:19: lifetime mismatch [E0623]
   |              ^ requires that `'b` must outlive `'a`

error: aborting due to previous error

