warning: not reporting region error due to nll
  --> $DIR/regions-reborrow-from-shorter-mut-ref.rs:16:5
   |
LL |     &mut **p //~ ERROR 16:5: 16:13: lifetime mismatch [E0623]
   |     ^^^^^^^^

error: unsatisfied lifetime constraints
  --> $DIR/regions-reborrow-from-shorter-mut-ref.rs:16:5
   |
LL | fn copy_borrowed_ptr<'a, 'b>(p: &'a mut &'b mut isize) -> &'b mut isize {
   |                      --  -- lifetime `'b` defined here
   |                      |
   |                      lifetime `'a` defined here
LL |     &mut **p //~ ERROR 16:5: 16:13: lifetime mismatch [E0623]
   |     ^^^^^^^^ function was supposed to return data with lifetime `'b` but it is returning data with lifetime `'a`

error: aborting due to previous error

