warning: not reporting region error due to nll
  --> $DIR/regions-infer-bound-from-trait.rs:43:5
   |
LL |     check_bound(x, a) //~ ERROR parameter type `A` may not live long enough
   |     ^^^^^^^^^^^

warning: not reporting region error due to nll
  --> $DIR/regions-infer-bound-from-trait.rs:47:5
   |
LL |     check_bound(x, a) //~ ERROR parameter type `A` may not live long enough
   |     ^^^^^^^^^^^

error[E0309]: the parameter type `A` may not live long enough
  --> $DIR/regions-infer-bound-from-trait.rs:43:5
   |
LL |     check_bound(x, a) //~ ERROR parameter type `A` may not live long enough
   |     ^^^^^^^^^^^^^^^^^
   |
   = help: consider adding an explicit lifetime bound `A: 'a`...

error[E0309]: the parameter type `A` may not live long enough
  --> $DIR/regions-infer-bound-from-trait.rs:47:5
   |
LL |     check_bound(x, a) //~ ERROR parameter type `A` may not live long enough
   |     ^^^^^^^^^^^^^^^^^
   |
   = help: consider adding an explicit lifetime bound `A: 'a`...

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0309`.
