warning: not reporting region error due to nll
  --> $DIR/regions-reborrow-from-shorter-mut-ref-mut-ref.rs:14:5
   |
LL |     &mut ***p //~ ERROR 14:5: 14:14: lifetime mismatch [E0623]
   |     ^^^^^^^^^

error: unsatisfied lifetime constraints
  --> $DIR/regions-reborrow-from-shorter-mut-ref-mut-ref.rs:13:85
   |
LL |   fn copy_borrowed_ptr<'a, 'b, 'c>(p: &'a mut &'b mut &'c mut isize) -> &'b mut isize {
   |  ______________________--__--_________________________________________________________^
   | |                      |   |
   | |                      |   lifetime `'b` defined here
   | |                      lifetime `'a` defined here
LL | |     &mut ***p //~ ERROR 14:5: 14:14: lifetime mismatch [E0623]
LL | | }
   | |_^ requires that `'a` must outlive `'b`

error: aborting due to previous error

