warning: not reporting region error due to nll
  --> $DIR/match-ref-mut-let-invariance.rs:21:9
   |
LL |         x //~ ERROR mismatched types
   |         ^

error: unsatisfied lifetime constraints
  --> $DIR/match-ref-mut-let-invariance.rs:21:9
   |
LL | impl<'b> S<'b> {
   |      -- lifetime `'b` defined here
LL |     fn bar<'a>(&'a mut self) -> &'a mut &'a i32 {
   |            -- lifetime `'a` defined here
LL |         let ref mut x = self.0;
LL |         x //~ ERROR mismatched types
   |         ^ returning this value requires that `'a` must outlive `'b`

error: aborting due to previous error

